// @wix/ecom_shipping-rates/build/cjs/src/interfaces-ecom-v1-shipping-rates-provider.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const address_2 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('ECOM_SHIPPING_RATES', [
    {
        name: 'getShippingRates',
        primaryHttpMappingPath: '/v1/getRates',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: address_1.transformSDKAddressToRESTAddress,
                        paths: [{ path: 'shippingRates.logistics.pickupDetails.address' }],
                    },
                    {
                        transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                        paths: [
                            { path: 'shippingRates.logistics.deliveryTimeSlot.from' },
                            { path: 'shippingRates.logistics.deliveryTimeSlot.to' },
                        ],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: address_2.transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.shippingDestination' },
                            { path: 'request.shippingOrigin' },
                        ],
                    },
                    {
                        transformFn: float_1.transformRESTFloatToSDKFloat,
                        paths: [{ path: 'request.lineItems.physicalProperties.weight' }],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
