// @wix/ecom_shipping-rates/build/cjs/src/service-plugins-error-classes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericShippingRatesWixError = exports.InvalidAddressWixError = exports.InvalidPostalCodeWixError = exports.MissingPostalCodeWixError = void 0;
/**
 * Missing postal code error.

Throw this error when a postal code is not provided in the request.
This error tells Wix that a valid postal code is required to calculate the shipping rate.
You can optionally provide error data to help debug the issue.
Error data you provide appears in the monitoring log but is not exposed to the user.
This error may block the site visitor from proceeding to checkout, so throw it only if there is no other way to proceed.
 */
class MissingPostalCodeWixError extends Error {
    constructor(data = {}) {
        super('MissingPostalCode');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'MISSING_POSTAL_CODE';
        this.name = 'MissingPostalCode';
        this.errorSchemaName =
            'com.wix.ecom.shipping_rates.v1.MissingPostalCodeError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'MissingPostalCode',
            applicationCode: 'MISSING_POSTAL_CODE',
        };
        this.data = data;
    }
}
exports.MissingPostalCodeWixError = MissingPostalCodeWixError;
/** @hidden */
MissingPostalCodeWixError.__type = 'wix_spi_error';
/**
 * Invalid postal code error.

Throw this error when an invalid postal code is provided in the request.
This error tells Wix that a valid postal code is required to calculate the shipping rate.
You can optionally provide error data to help debug the issue.
Error data you provide appears in the monitoring log but is not exposed to the user.
This error may block the site visitor from proceeding to checkout, so throw it only if there is no other way to proceed.
 */
class InvalidPostalCodeWixError extends Error {
    constructor(data = {}) {
        super('InvalidPostalCode');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'INVALID_POSTAL_CODE';
        this.name = 'InvalidPostalCode';
        this.errorSchemaName =
            'com.wix.ecom.shipping_rates.v1.InvalidPostalCodeError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'InvalidPostalCode',
            applicationCode: 'INVALID_POSTAL_CODE',
        };
        this.data = data;
    }
}
exports.InvalidPostalCodeWixError = InvalidPostalCodeWixError;
/** @hidden */
InvalidPostalCodeWixError.__type = 'wix_spi_error';
/**
 * Invalid address error.

Throw this error when one or more fields in the address provided are invalid.
This error tells Wix that more address details are needed to calculate the shipping rate.
Wix may then prompt the user to provide more detailed address information.
The error data must include the invalid fields in the address. For example: `ShippingDestination.subdivision`.
 */
class InvalidAddressWixError extends Error {
    constructor(data = {}) {
        super('InvalidAddress');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'PARTIAL_ADDRESS';
        this.name = 'InvalidAddress';
        this.errorSchemaName = 'com.wix.ecom.shipping_rates.v1.InvalidAddressError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'InvalidAddress',
            applicationCode: 'PARTIAL_ADDRESS',
        };
        this.data = data;
    }
}
exports.InvalidAddressWixError = InvalidAddressWixError;
/** @hidden */
InvalidAddressWixError.__type = 'wix_spi_error';
/**
 * Generic shipping rate calculation error.

Throw this error for cases the other errors don't cover.
You can optionally provide error data to help debug the issue.
Error data you provide appears in the monitoring log but is not exposed to the user.
This error may block the site visitor from proceeding to checkout, so throw it only if there is no other way to proceed.
 */
class GenericShippingRatesWixError extends Error {
    constructor(data = {}) {
        super('GenericShippingRatesError');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'GENERIC_ERROR';
        this.name = 'GenericShippingRatesError';
        this.errorSchemaName =
            'com.wix.ecom.shipping_rates.v1.GenericShippingRatesError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'GenericShippingRatesError',
            applicationCode: 'GENERIC_ERROR',
        };
        this.data = data;
    }
}
exports.GenericShippingRatesWixError = GenericShippingRatesWixError;
/** @hidden */
GenericShippingRatesWixError.__type = 'wix_spi_error';
