// @wix/ecom_cart/build/es/src/ecom-v1-cart-cart.meta.js
import * as ambassadorWixEcomV1Cart from './ecom-v1-cart-cart.http.js';
export function createCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.createCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCart() {
    const payload = { cartInfo: { id: ':cartInfoId' } };
    const getRequestOptions = ambassadorWixEcomV1Cart.updateCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/carts/{cartInfo.id}',
        pathParams: { cartInfoId: 'cartInfoId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCart() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.getCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/carts/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addToCart() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.addToCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/add-to-cart',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeLineItems() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.removeLineItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/remove-line-items',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createCheckout() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.createCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/create-checkout',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeCoupon() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.removeCoupon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/remove-coupon',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateLineItemsQuantity() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.updateLineItemsQuantity(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/update-line-items-quantity',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function estimateTotals() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.estimateTotals(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/estimate-totals',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCart() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.deleteCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/carts/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
