// @wix/ecom_discounts-custom-trigger/build/cjs/src/ecom-custom-trigger-v1-custom-trigger-discounts-custom-trigger.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTriggers = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/custom-triggers-service',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/discount-custom-triggers',
                destPath: '/v1/discount-custom-triggers',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/discount-custom-triggers',
                destPath: '/v1/discount-custom-triggers',
            },
            {
                srcPath: '/v1/discount-custom-triggers',
                destPath: '/v1/discount-custom-triggers',
            },
        ],
    };
    return (0, rest_modules_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_discounts-custom-trigger';
/** List all custom triggers that are available on a given site */
function listTriggers(payload) {
    function __listTriggers({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.custom_trigger.v1.custom_trigger',
            method: 'POST',
            methodFqn: 'com.wix.ecom.custom_trigger.v1.CustomTriggersService.ListTriggers',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomCustomTriggerV1CustomTriggersServiceUrl({
                protoPath: '/v1/discount-custom-triggers/list',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __listTriggers;
}
exports.listTriggers = listTriggers;
