// @wix/ecom_order-transactions/build/es/src/ecom-v1-order-transactions-order-transactions.public.js
import { addPayments as universalAddPayments, bulkUpdatePaymentStatuses as universalBulkUpdatePaymentStatuses, listTransactionsForMultipleOrders as universalListTransactionsForMultipleOrders, listTransactionsForSingleOrder as universalListTransactionsForSingleOrder, updatePaymentStatus as universalUpdatePaymentStatus, } from './ecom-v1-order-transactions-order-transactions.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function listTransactionsForSingleOrder(httpClient) {
    return (orderId) => universalListTransactionsForSingleOrder(orderId, 
    // @ts-ignore
    { httpClient });
}
export function listTransactionsForMultipleOrders(httpClient) {
    return (orderIds) => universalListTransactionsForMultipleOrders(orderIds, 
    // @ts-ignore
    { httpClient });
}
export function addPayments(httpClient) {
    return (orderId, payments) => universalAddPayments(orderId, payments, 
    // @ts-ignore
    { httpClient });
}
export function updatePaymentStatus(httpClient) {
    return (identifiers, options) => universalUpdatePaymentStatus(identifiers, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkUpdatePaymentStatuses(httpClient) {
    return (paymentAndOrderIds, options) => universalBulkUpdatePaymentStatuses(paymentAndOrderIds, options, 
    // @ts-ignore
    { httpClient });
}
export { ActionType, ActivityType, AuthorizationCaptureStatus, AuthorizationVoidStatus, ChannelType, EnteredByIdentityType, FulfillmentStatus, IdentityType, InvoiceSource, LineItemType, ManuallyRefundableReason, MediaItemType, MembershipPaymentStatus, NonRefundableReason, PaymentStatus, Reason, RefundStatus, RefundableStatus, RestockType, SortOrder, SubscriptionFrequency, TransactionStatus, VatType, WebhookIdentityType, WeightUnit, } from './ecom-v1-order-transactions-order-transactions.universal.js';
