// @wix/ecom_order-transactions/build/es/src/ecom-v1-order-transactions-order-transactions.meta.js
import * as ambassadorWixEcomV1OrderTransactions from './ecom-v1-order-transactions-order-transactions.http.js';
export function listTransactionsForSingleOrder() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.listTransactionsForSingleOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/payments/orders/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listTransactionsForMultipleOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.listTransactionsForMultipleOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments/list-by-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addPayments() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.addPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments/orders/{orderId}/add-payment',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updatePaymentStatus() {
    const payload = { paymentId: ':paymentId', orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.updatePaymentStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments/{paymentId}/orders/{orderId}/update-payment-transaction-status',
        pathParams: { paymentId: 'paymentId', orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdatePaymentStatuses() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.bulkUpdatePaymentStatuses(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/payments/update-payment-transaction-status',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
