// @wix/ecom_order-transactions/build/es/src/ecom-v1-order-transactions-order-transactions.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveComWixEcomOrdersPaymentsV1PaymentsUrl(opts) {
    const domainToMappings = {
        _: [
            {
                srcPath: '/_api/ecom-payments',
                destPath: '',
            },
            {
                srcPath: '/_api/payments',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/bulk/payments',
                destPath: '/v1/bulk/payments',
            },
            {
                srcPath: '/ecom/v1/bulk/invoices',
                destPath: '/v1/bulk/invoices',
            },
            {
                srcPath: '/ecom/v1/payments',
                destPath: '/v1/payments',
            },
            {
                srcPath: '/ecom/v1/invoices',
                destPath: '/v1/invoices',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/ecom-payments',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_order-transactions';
/**
 * Retrieves information about payments and refunds associated with a specified order.
 *
 *
 * The `listTransactionsForSingleOrder()` function returns a Promise that resolves when the specified order's transaction records are retrieved.
 */
export function listTransactionsForSingleOrder(payload) {
    function __listTransactionsForSingleOrder({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_transactions',
            method: 'GET',
            methodFqn: 'com.wix.ecom.orders.payments.v1.Payments.ListTransactionsForSingleOrder',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomOrdersPaymentsV1PaymentsUrl({
                protoPath: '/v1/payments/orders/{orderId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderTransactions.payments.createdDate' },
                        { path: 'orderTransactions.payments.updatedDate' },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate',
                        },
                        { path: 'orderTransactions.refunds.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listTransactionsForSingleOrder;
}
/**
 * Retrieves information about payments and refunds associated with all specified orders.
 *
 *
 * The `listTransactionsForMultipleOrders()` function returns a Promise that resolves when the specified orders' transaction records are retrieved.
 */
export function listTransactionsForMultipleOrders(payload) {
    function __listTransactionsForMultipleOrders({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_transactions',
            method: 'POST',
            methodFqn: 'com.wix.ecom.orders.payments.v1.Payments.ListTransactionsForMultipleOrders',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomOrdersPaymentsV1PaymentsUrl({
                protoPath: '/v1/payments/list-by-ids',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderTransactions.payments.createdDate' },
                        { path: 'orderTransactions.payments.updatedDate' },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate',
                        },
                        { path: 'orderTransactions.refunds.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listTransactionsForMultipleOrders;
}
/**
 * Adds up to 50 payment records to an order.
 *
 *
 * The `addPayments()` function returns a Promise that resolves when the payment records are added to an order.
 *
 * > **Note:** This does **NOT** perform the actual charging - the order is only updated with records of the payments.
 */
export function addPayments(payload) {
    function __addPayments({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'payments.createdDate' },
                    { path: 'payments.updatedDate' },
                    {
                        path: 'payments.regularPaymentDetails.authorizationDetails.authorizedDate',
                    },
                    {
                        path: 'payments.regularPaymentDetails.authorizationDetails.captures.createdDate',
                    },
                    {
                        path: 'payments.regularPaymentDetails.authorizationDetails.void.voidedDate',
                    },
                    {
                        path: 'payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_transactions',
            method: 'POST',
            methodFqn: 'com.wix.ecom.orders.payments.v1.Payments.AddPayments',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomOrdersPaymentsV1PaymentsUrl({
                protoPath: '/v1/payments/orders/{orderId}/add-payment',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderTransactions.payments.createdDate' },
                        { path: 'orderTransactions.payments.updatedDate' },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate',
                        },
                        { path: 'orderTransactions.refunds.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __addPayments;
}
/**
 * Updates the status of an order's payment.
 *
 *
 * The `updatePaymentStatus()` function returns a Promise that resolves when the payment status is updated.
 */
export function updatePaymentStatus(payload) {
    function __updatePaymentStatus({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'statusUpdatedDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_transactions',
            method: 'POST',
            methodFqn: 'com.wix.ecom.orders.payments.v1.Payments.UpdatePaymentStatus',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomOrdersPaymentsV1PaymentsUrl({
                protoPath: '/v1/payments/{paymentId}/orders/{orderId}/update-payment-transaction-status',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderTransactions.payments.createdDate' },
                        { path: 'orderTransactions.payments.updatedDate' },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.authorizedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.captures.createdDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.void.voidedDate',
                        },
                        {
                            path: 'orderTransactions.payments.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate',
                        },
                        { path: 'orderTransactions.refunds.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updatePaymentStatus;
}
/**
 * Updates multiple order payments with a specified status.
 *
 *
 * The `bulkUpdatePaymentStatus()` function returns a Promise that resolves when the payment statuses are updated.
 */
export function bulkUpdatePaymentStatuses(payload) {
    function __bulkUpdatePaymentStatuses({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'statusUpdatedDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_transactions',
            method: 'POST',
            methodFqn: 'com.wix.ecom.orders.payments.v1.Payments.BulkUpdatePaymentStatuses',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomOrdersPaymentsV1PaymentsUrl({
                protoPath: '/v1/bulk/payments/update-payment-transaction-status',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.item.createdDate' },
                        { path: 'results.item.updatedDate' },
                        {
                            path: 'results.item.regularPaymentDetails.authorizationDetails.authorizedDate',
                        },
                        {
                            path: 'results.item.regularPaymentDetails.authorizationDetails.captures.createdDate',
                        },
                        {
                            path: 'results.item.regularPaymentDetails.authorizationDetails.void.voidedDate',
                        },
                        {
                            path: 'results.item.regularPaymentDetails.authorizationDetails.scheduledAction.executionDate',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkUpdatePaymentStatuses;
}
