// @wix/ecom_abandoned-checkouts/build/cjs/src/ecom-v1-abandoned-checkout-abandoned-checkouts.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.VatType = exports.Status = exports.SortOrder = exports.Mode = exports.Identity = exports.CommonSortOrder = exports.ActivityType = exports.onAbandonedCheckoutRecovered = exports.redirectToCheckout = exports.searchAbandonedCheckouts = exports.queryAbandonedCheckouts = exports.deleteAbandonedCheckout = exports.getAbandonedCheckout = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_1 = require("./ecom-v1-abandoned-checkout-abandoned-checkouts.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function getAbandonedCheckout(httpClient) {
    return (abandonedCheckoutId) => (0, ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_1.getAbandonedCheckout)(abandonedCheckoutId, 
    // @ts-ignore
    { httpClient });
}
exports.getAbandonedCheckout = getAbandonedCheckout;
function deleteAbandonedCheckout(httpClient) {
    return (abandonedCheckoutId) => (0, ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_1.deleteAbandonedCheckout)(abandonedCheckoutId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteAbandonedCheckout = deleteAbandonedCheckout;
function queryAbandonedCheckouts(httpClient) {
    return () => (0, ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_1.queryAbandonedCheckouts)(
    // @ts-ignore
    { httpClient });
}
exports.queryAbandonedCheckouts = queryAbandonedCheckouts;
function searchAbandonedCheckouts(httpClient) {
    return (options) => (0, ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_1.searchAbandonedCheckouts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.searchAbandonedCheckouts = searchAbandonedCheckouts;
function redirectToCheckout(httpClient) {
    return (abandonedCheckoutId, metasiteId) => (0, ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_1.redirectToCheckout)(abandonedCheckoutId, metasiteId, 
    // @ts-ignore
    { httpClient });
}
exports.redirectToCheckout = redirectToCheckout;
exports.onAbandonedCheckoutRecovered = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.abandoned_checkout_recovered', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.abandonedCheckout.createdDate' },
            { path: 'data.abandonedCheckout.updatedDate' },
            { path: 'data.abandonedCheckout.checkoutRecoveredDate' },
            { path: 'data.abandonedCheckout.activities.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2 = require("./ecom-v1-abandoned-checkout-abandoned-checkouts.universal.js");
Object.defineProperty(exports, "ActivityType", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.ActivityType; } });
Object.defineProperty(exports, "CommonSortOrder", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.CommonSortOrder; } });
Object.defineProperty(exports, "Identity", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.Identity; } });
Object.defineProperty(exports, "Mode", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.Mode; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.Status; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.VatType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_abandoned_checkout_abandoned_checkouts_universal_js_2.WebhookIdentityType; } });
