// @wix/ecom_abandoned-checkouts/build/cjs/src/ecom-v1-abandoned-checkout-abandoned-checkouts.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.redirectToCheckout = exports.searchAbandonedCheckouts = exports.queryAbandonedCheckouts = exports.deleteAbandonedCheckout = exports.getAbandonedCheckout = void 0;
const ambassadorWixEcomV1AbandonedCheckout = __importStar(require("./ecom-v1-abandoned-checkout-abandoned-checkouts.http.js"));
function getAbandonedCheckout() {
    const payload = { abandonedCheckoutId: ':abandonedCheckoutId' };
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.getAbandonedCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/abandoned-checkout/{abandonedCheckoutId}',
        pathParams: { abandonedCheckoutId: 'abandonedCheckoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getAbandonedCheckout = getAbandonedCheckout;
function deleteAbandonedCheckout() {
    const payload = { abandonedCheckoutId: ':abandonedCheckoutId' };
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.deleteAbandonedCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/abandoned-checkout/{abandonedCheckoutId}',
        pathParams: { abandonedCheckoutId: 'abandonedCheckoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteAbandonedCheckout = deleteAbandonedCheckout;
function queryAbandonedCheckouts() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.queryAbandonedCheckouts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/abandoned-checkout/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryAbandonedCheckouts = queryAbandonedCheckouts;
function searchAbandonedCheckouts() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.searchAbandonedCheckouts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/abandoned-checkout/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchAbandonedCheckouts = searchAbandonedCheckouts;
function redirectToCheckout() {
    const payload = { abandonedCheckoutId: ':abandonedCheckoutId' };
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.redirectToCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout',
        pathParams: { abandonedCheckoutId: 'abandonedCheckoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.redirectToCheckout = redirectToCheckout;
