// @wix/ecom_shippo-configurations/build/cjs/src/ecom-v1-shippo-configuration-shippo-configurations.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateExtendedFields = exports.queryShippoConfigurations = exports.deleteShippoConfiguration = exports.updateShippoConfiguration = exports.getShippoConfiguration = exports.createShippoConfiguration = void 0;
const ambassadorWixEcomV1ShippoConfiguration = __importStar(require("./ecom-v1-shippo-configuration-shippo-configurations.http.js"));
function createShippoConfiguration() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.createShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shippo-configurations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createShippoConfiguration = createShippoConfiguration;
function getShippoConfiguration() {
    const payload = { shippoConfigurationId: ':shippoConfigurationId' };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.getShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shippo-configurations/{shippoConfigurationId}',
        pathParams: { shippoConfigurationId: 'shippoConfigurationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getShippoConfiguration = getShippoConfiguration;
function updateShippoConfiguration() {
    const payload = {
        shippoConfiguration: { id: ':shippoConfigurationId' },
    };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.updateShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/shippo-configurations/{shippoConfiguration.id}',
        pathParams: { shippoConfigurationId: 'shippoConfigurationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateShippoConfiguration = updateShippoConfiguration;
function deleteShippoConfiguration() {
    const payload = { shippoConfigurationId: ':shippoConfigurationId' };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.deleteShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/shippo-configurations/{shippoConfigurationId}',
        pathParams: { shippoConfigurationId: 'shippoConfigurationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteShippoConfiguration = deleteShippoConfiguration;
function queryShippoConfigurations() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.queryShippoConfigurations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shippo-configurations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryShippoConfigurations = queryShippoConfigurations;
function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shippo-configurations/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateExtendedFields = updateExtendedFields;
