// @wix/ecom_shippo-configurations/build/cjs/src/ecom-v1-shippo-configuration-shippo-configurations.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.PackageType = exports.InternationalService = exports.CalculationType = exports.DomesticService = void 0;
var DomesticService;
(function (DomesticService) {
    DomesticService["UNKNOWN_DOMESTIC_SERVICE"] = "UNKNOWN_DOMESTIC_SERVICE";
    DomesticService["GROUND_ADVANTAGE"] = "GROUND_ADVANTAGE";
    DomesticService["PRIORITY_MAIL"] = "PRIORITY_MAIL";
    DomesticService["PRIORITY_MAIL_EXPRESS"] = "PRIORITY_MAIL_EXPRESS";
})(DomesticService || (exports.DomesticService = DomesticService = {}));
var CalculationType;
(function (CalculationType) {
    CalculationType["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
    CalculationType["FIXED"] = "FIXED";
    CalculationType["PERCENTAGE"] = "PERCENTAGE";
})(CalculationType || (exports.CalculationType = CalculationType = {}));
var InternationalService;
(function (InternationalService) {
    InternationalService["UNKNOWN_INTERNATIONAL_SERVICE"] = "UNKNOWN_INTERNATIONAL_SERVICE";
    InternationalService["FIRST_CLASS_PACKAGE_INTERNATIONAL"] = "FIRST_CLASS_PACKAGE_INTERNATIONAL";
    InternationalService["PRIORITY_MAIL_INTERNATIONAL"] = "PRIORITY_MAIL_INTERNATIONAL";
    InternationalService["PRIORITY_MAIL_EXPRESS_INTERNATIONAL"] = "PRIORITY_MAIL_EXPRESS_INTERNATIONAL";
})(InternationalService || (exports.InternationalService = InternationalService = {}));
var PackageType;
(function (PackageType) {
    PackageType["UNKNOWN_PACKAGE_TYPE"] = "UNKNOWN_PACKAGE_TYPE";
    PackageType["CUSTOM"] = "CUSTOM";
    PackageType["FLAT_RATE_ENVELOPE"] = "FLAT_RATE_ENVELOPE";
    PackageType["PADDED_FLAT_RATE_ENVELOPE"] = "PADDED_FLAT_RATE_ENVELOPE";
    PackageType["SMALL_FLAT_RATE_BOX"] = "SMALL_FLAT_RATE_BOX";
    PackageType["MEDIUM_FLAT_RATE_BOX_1"] = "MEDIUM_FLAT_RATE_BOX_1";
    PackageType["MEDIUM_FLAT_RATE_BOX_2"] = "MEDIUM_FLAT_RATE_BOX_2";
    PackageType["LARGE_FLAT_RATE_BOX"] = "LARGE_FLAT_RATE_BOX";
})(PackageType || (exports.PackageType = PackageType = {}));
var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (exports.SortOrder = SortOrder = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
