// @wix/ecom_shippo-configurations/build/cjs/src/ecom-v1-shippo-configuration-shippo-configurations.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.PackageType = exports.InternationalService = exports.DomesticService = exports.CalculationType = exports.onShippoConfigurationUpdated = exports.onShippoConfigurationDeleted = exports.onShippoConfigurationCreated = exports.updateExtendedFields = exports.queryShippoConfigurations = exports.deleteShippoConfiguration = exports.updateShippoConfiguration = exports.getShippoConfiguration = exports.createShippoConfiguration = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_shippo_configuration_shippo_configurations_universal_js_1 = require("./ecom-v1-shippo-configuration-shippo-configurations.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createShippoConfiguration(httpClient) {
    return (shippoConfiguration, options) => (0, ecom_v1_shippo_configuration_shippo_configurations_universal_js_1.createShippoConfiguration)(shippoConfiguration, options, 
    // @ts-ignore
    { httpClient });
}
exports.createShippoConfiguration = createShippoConfiguration;
function getShippoConfiguration(httpClient) {
    return (shippoConfigurationId) => (0, ecom_v1_shippo_configuration_shippo_configurations_universal_js_1.getShippoConfiguration)(shippoConfigurationId, 
    // @ts-ignore
    { httpClient });
}
exports.getShippoConfiguration = getShippoConfiguration;
function updateShippoConfiguration(httpClient) {
    return (_id, shippoConfiguration) => (0, ecom_v1_shippo_configuration_shippo_configurations_universal_js_1.updateShippoConfiguration)(_id, shippoConfiguration, 
    // @ts-ignore
    { httpClient });
}
exports.updateShippoConfiguration = updateShippoConfiguration;
function deleteShippoConfiguration(httpClient) {
    return (shippoConfigurationId) => (0, ecom_v1_shippo_configuration_shippo_configurations_universal_js_1.deleteShippoConfiguration)(shippoConfigurationId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteShippoConfiguration = deleteShippoConfiguration;
function queryShippoConfigurations(httpClient) {
    return () => (0, ecom_v1_shippo_configuration_shippo_configurations_universal_js_1.queryShippoConfigurations)(
    // @ts-ignore
    { httpClient });
}
exports.queryShippoConfigurations = queryShippoConfigurations;
function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => (0, ecom_v1_shippo_configuration_shippo_configurations_universal_js_1.updateExtendedFields)(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateExtendedFields = updateExtendedFields;
exports.onShippoConfigurationCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.shippo_configuration_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'entity.shipFromAddress' }],
    },
])))();
exports.onShippoConfigurationDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.shippo_configuration_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'entity.shipFromAddress' }],
    },
])))();
exports.onShippoConfigurationUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.shippo_configuration_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'entity.shipFromAddress' }],
    },
])))();
var ecom_v1_shippo_configuration_shippo_configurations_universal_js_2 = require("./ecom-v1-shippo-configuration-shippo-configurations.universal.js");
Object.defineProperty(exports, "CalculationType", { enumerable: true, get: function () { return ecom_v1_shippo_configuration_shippo_configurations_universal_js_2.CalculationType; } });
Object.defineProperty(exports, "DomesticService", { enumerable: true, get: function () { return ecom_v1_shippo_configuration_shippo_configurations_universal_js_2.DomesticService; } });
Object.defineProperty(exports, "InternationalService", { enumerable: true, get: function () { return ecom_v1_shippo_configuration_shippo_configurations_universal_js_2.InternationalService; } });
Object.defineProperty(exports, "PackageType", { enumerable: true, get: function () { return ecom_v1_shippo_configuration_shippo_configurations_universal_js_2.PackageType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_shippo_configuration_shippo_configurations_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_shippo_configuration_shippo_configurations_universal_js_2.WebhookIdentityType; } });
