// @wix/ecom_custom-triggers/build/cjs/src/interfaces-ecommerce-v1-custom-trigger.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('ECOM_DISCOUNTS_TRIGGER', [
    {
        name: 'listTriggers',
        primaryHttpMappingPath: '/v1/list-triggers',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'getEligibleTriggers',
        primaryHttpMappingPath: '/v1/get-eligible-triggers',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
