// @wix/ecom_gift-vouchers-provider/build/cjs/src/service-plugins-error-classes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlreadyVoidedWixError = exports.TransactionNotFoundWixError = exports.CurrencyNotSupportedWixError = exports.AlreadyRedeemedWixError = exports.InsufficientFundsWixError = exports.GiftCardExpiredWixError = exports.GiftCardDisabledWixError = exports.GiftCardNotFoundWixError = void 0;
/**
 * Gift card not found
 */
class GiftCardNotFoundWixError extends Error {
    constructor() {
        super('GiftCardNotFound');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'GIFT_CARD_NOT_FOUND';
        this.name = 'GiftCardNotFound';
        this.errorType = 'APPLICATION';
    }
}
exports.GiftCardNotFoundWixError = GiftCardNotFoundWixError;
/** @hidden */
GiftCardNotFoundWixError.__type = 'wix_spi_error';
/**
 * Gift card is disabled
 */
class GiftCardDisabledWixError extends Error {
    constructor() {
        super('GiftCardDisabled');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'GIFT_CARD_DISABLED';
        this.name = 'GiftCardDisabled';
        this.errorType = 'APPLICATION';
    }
}
exports.GiftCardDisabledWixError = GiftCardDisabledWixError;
/** @hidden */
GiftCardDisabledWixError.__type = 'wix_spi_error';
/**
 * Gift card was expired
 */
class GiftCardExpiredWixError extends Error {
    constructor() {
        super('GiftCardExpired');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'GIFT_CARD_EXPIRED';
        this.name = 'GiftCardExpired';
        this.errorType = 'APPLICATION';
    }
}
exports.GiftCardExpiredWixError = GiftCardExpiredWixError;
/** @hidden */
GiftCardExpiredWixError.__type = 'wix_spi_error';
/**
 * Insufficient funds for redeem
 */
class InsufficientFundsWixError extends Error {
    constructor() {
        super('InsufficientFunds');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'INSUFFICIENT_FUNDS';
        this.name = 'InsufficientFunds';
        this.errorType = 'APPLICATION';
    }
}
exports.InsufficientFundsWixError = InsufficientFundsWixError;
/** @hidden */
InsufficientFundsWixError.__type = 'wix_spi_error';
/**
 * Gift card already redeemed for the given order
 */
class AlreadyRedeemedWixError extends Error {
    constructor() {
        super('AlreadyRedeemed');
        this.httpCode = 409;
        this.statusCode = 'ALREADY_EXISTS';
        this.applicationCode = 'ALREADY_REDEEMED';
        this.name = 'AlreadyRedeemed';
        this.errorType = 'APPLICATION';
    }
}
exports.AlreadyRedeemedWixError = AlreadyRedeemedWixError;
/** @hidden */
AlreadyRedeemedWixError.__type = 'wix_spi_error';
/**
 * Currency code not supported
 */
class CurrencyNotSupportedWixError extends Error {
    constructor() {
        super('CurrencyNotSupported');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'CURRENCY_NOT_SUPPORTED';
        this.name = 'CurrencyNotSupported';
        this.errorType = 'APPLICATION';
    }
}
exports.CurrencyNotSupportedWixError = CurrencyNotSupportedWixError;
/** @hidden */
CurrencyNotSupportedWixError.__type = 'wix_spi_error';
/**
 * Transaction not found.
 */
class TransactionNotFoundWixError extends Error {
    constructor() {
        super('TransactionNotFound');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'TRANSACTION_NOT_FOUND';
        this.name = 'TransactionNotFound';
        this.errorType = 'APPLICATION';
    }
}
exports.TransactionNotFoundWixError = TransactionNotFoundWixError;
/** @hidden */
TransactionNotFoundWixError.__type = 'wix_spi_error';
/**
 * Transaction already voided
 */
class AlreadyVoidedWixError extends Error {
    constructor() {
        super('AlreadyVoided');
        this.httpCode = 409;
        this.statusCode = 'ALREADY_EXISTS';
        this.applicationCode = 'ALREADY_VOIDED';
        this.name = 'AlreadyVoided';
        this.errorType = 'APPLICATION';
    }
}
exports.AlreadyVoidedWixError = AlreadyVoidedWixError;
/** @hidden */
AlreadyVoidedWixError.__type = 'wix_spi_error';
