// @wix/ecom_order-payment-requests/build/es/src/ecom-v1-order-payment-request-order-payment-requests.public.js
import { createOrderPaymentRequest as universalCreateOrderPaymentRequest, deleteOrderPaymentRequest as universalDeleteOrderPaymentRequest, getOrderPaymentRequest as universalGetOrderPaymentRequest, getOrderPaymentRequestUrl as universalGetOrderPaymentRequestUrl, queryOrderPaymentRequests as universalQueryOrderPaymentRequests, updateExtendedFields as universalUpdateExtendedFields, updateOrderPaymentRequest as universalUpdateOrderPaymentRequest, } from './ecom-v1-order-payment-request-order-payment-requests.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function createOrderPaymentRequest(httpClient) {
    return (options) => universalCreateOrderPaymentRequest(options, 
    // @ts-ignore
    { httpClient });
}
export function getOrderPaymentRequest(httpClient) {
    return (orderPaymentRequestId) => universalGetOrderPaymentRequest(orderPaymentRequestId, 
    // @ts-ignore
    { httpClient });
}
export function updateOrderPaymentRequest(httpClient) {
    return (_id, orderPaymentRequest) => universalUpdateOrderPaymentRequest(_id, orderPaymentRequest, 
    // @ts-ignore
    { httpClient });
}
export function deleteOrderPaymentRequest(httpClient) {
    return (orderPaymentRequestId) => universalDeleteOrderPaymentRequest(orderPaymentRequestId, 
    // @ts-ignore
    { httpClient });
}
export function queryOrderPaymentRequests(httpClient) {
    return () => universalQueryOrderPaymentRequests(
    // @ts-ignore
    { httpClient });
}
export function getOrderPaymentRequestUrl(httpClient) {
    return (orderPaymentRequestId) => universalGetOrderPaymentRequestUrl(orderPaymentRequestId, 
    // @ts-ignore
    { httpClient });
}
export function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => universalUpdateExtendedFields(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
export { SortOrder, Status, WebhookIdentityType, } from './ecom-v1-order-payment-request-order-payment-requests.universal.js';
