// @wix/ecom_order-payment-requests/build/es/src/ecom-v1-order-payment-request-order-payment-requests.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformSDKFieldMaskToRESTFieldMask } from '@wix/sdk-runtime/transformations/field-mask';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/order-payment-request',
                destPath: '',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
        _: [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_order-payment-requests';
/** Creates a order payment request. */
export function createOrderPaymentRequest(payload) {
    function __createOrderPaymentRequest({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'orderPaymentRequest.expirationDate' },
                    { path: 'orderPaymentRequest.createdDate' },
                    { path: 'orderPaymentRequest.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.CreateOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({ protoPath: '/v1/order-payment-requests', data: serializedData, host }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createOrderPaymentRequest;
}
/** Retrieves a order payment request. */
export function getOrderPaymentRequest(payload) {
    function __getOrderPaymentRequest({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'GET',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.GetOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{orderPaymentRequestId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getOrderPaymentRequest;
}
/**
 * Updates a order payment request.
 *
 * Please note that only `UNPAID` payment requests can be updated.
 */
export function updateOrderPaymentRequest(payload) {
    function __updateOrderPaymentRequest({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'orderPaymentRequest.expirationDate' },
                    { path: 'orderPaymentRequest.createdDate' },
                    { path: 'orderPaymentRequest.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'PATCH',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.UpdateOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{orderPaymentRequest.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateOrderPaymentRequest;
}
/**
 * Deletes a order payment request.
 *
 * Please note that only `UNPAID` payment requests can be deleted.
 */
export function deleteOrderPaymentRequest(payload) {
    function __deleteOrderPaymentRequest({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'DELETE',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.DeleteOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{orderPaymentRequestId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __deleteOrderPaymentRequest;
}
/**
 * Retrieves a list of Payment Requests, given the provided [paging, filtering, and sorting][1].
 *
 * Query Payment Requests runs with these defaults, which you can override:
 * - `createdDate` is sorted in DESC order
 * - `cursorPaging.limit` is 100
 *
 * To learn about working with _Query_ endpoints, see
 * [API Query Language](https://dev.wix.com/docs/rest/articles/getting-started/api-query-language), and
 * [Sorting and Paging](https://dev.wix.com/docs/rest/articles/getting-started/sorting-and-paging).
 */
export function queryOrderPaymentRequests(payload) {
    function __queryOrderPaymentRequests({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.QueryOrderPaymentRequests',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({ protoPath: '/v1/order-payment-requests/query', data: payload, host }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequests.expirationDate' },
                        { path: 'orderPaymentRequests.createdDate' },
                        { path: 'orderPaymentRequests.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryOrderPaymentRequests;
}
/** Retrieves the order payment request page URL of a specified order payment request. */
export function getOrderPaymentRequestUrl(payload) {
    function __getOrderPaymentRequestUrl({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.GetOrderPaymentRequestURL',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({ protoPath: '/v1/order-payment-requests/url', data: payload, host }),
            data: payload,
        };
        return metadata;
    }
    return __getOrderPaymentRequestUrl;
}
/** Updates extended fields of a order payment request */
export function updateExtendedFields(payload) {
    function __updateExtendedFields({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.UpdateExtendedFields',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{id}/update-extended-fields',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateExtendedFields;
}
