// @wix/ecom_order-payment-requests/build/es/src/ecom-v1-order-payment-request-order-payment-requests.meta.js
import * as ambassadorWixEcomV1OrderPaymentRequest from './ecom-v1-order-payment-request-order-payment-requests.http.js';
export function createOrderPaymentRequest() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.createOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOrderPaymentRequest() {
    const payload = { orderPaymentRequestId: ':orderPaymentRequestId' };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.getOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/order-payment-requests/{orderPaymentRequestId}',
        pathParams: { orderPaymentRequestId: 'orderPaymentRequestId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateOrderPaymentRequest() {
    const payload = {
        orderPaymentRequest: { id: ':orderPaymentRequestId' },
    };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.updateOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/order-payment-requests/{orderPaymentRequest.id}',
        pathParams: { orderPaymentRequestId: 'orderPaymentRequestId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteOrderPaymentRequest() {
    const payload = { orderPaymentRequestId: ':orderPaymentRequestId' };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.deleteOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/order-payment-requests/{orderPaymentRequestId}',
        pathParams: { orderPaymentRequestId: 'orderPaymentRequestId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryOrderPaymentRequests() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.queryOrderPaymentRequests(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOrderPaymentRequestUrl() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.getOrderPaymentRequestUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests/url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
