// @wix/ecom_draft-orders/build/cjs/src/ecom-v1-draft-order-draft-orders.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setTaxExemption = exports.queryDraftOrders = exports.deleteDraftOrder = exports.createOrderFromDraft = exports.commitDraftOrder = exports.getOrderDraftabilityStatus = exports.getDraftOrder = exports.setBillingInfo = exports.setRecipientInfo = exports.setBuyerInfo = exports.setShippingInfo = exports.deleteCustomAdditionalFees = exports.createCustomAdditionalFees = exports.setAdditionalFees = exports.deleteCustomDiscounts = exports.createCustomDiscounts = exports.setDiscounts = exports.updateLineItems = exports.addLineItemsToDraftOrder = exports.createEmptyDraftOrder = exports.createDraftOrder = void 0;
const ambassadorWixEcomV1DraftOrder = __importStar(require("./ecom-v1-draft-order-draft-orders.http.js"));
function createDraftOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDraftOrder = createDraftOrder;
function createEmptyDraftOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createEmptyDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/create-empty-draft-order',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createEmptyDraftOrder = createEmptyDraftOrder;
function addLineItemsToDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.addLineItemsToDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/add-line-items',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addLineItemsToDraftOrder = addLineItemsToDraftOrder;
function updateLineItems() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.updateLineItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/line-items',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateLineItems = updateLineItems;
function setDiscounts() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setDiscounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/set-discounts',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setDiscounts = setDiscounts;
function createCustomDiscounts() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createCustomDiscounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/create-custom-discounts',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCustomDiscounts = createCustomDiscounts;
function deleteCustomDiscounts() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.deleteCustomDiscounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/delete-custom-discounts',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCustomDiscounts = deleteCustomDiscounts;
function setAdditionalFees() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setAdditionalFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/set-additional-fees',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setAdditionalFees = setAdditionalFees;
function createCustomAdditionalFees() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createCustomAdditionalFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/create-additional-fees',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCustomAdditionalFees = createCustomAdditionalFees;
function deleteCustomAdditionalFees() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.deleteCustomAdditionalFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/delete-custom-additional-fees',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCustomAdditionalFees = deleteCustomAdditionalFees;
function setShippingInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setShippingInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-shipping-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setShippingInfo = setShippingInfo;
function setBuyerInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setBuyerInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-buyer-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setBuyerInfo = setBuyerInfo;
function setRecipientInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setRecipientInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-recipient-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setRecipientInfo = setRecipientInfo;
function setBillingInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setBillingInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-billing-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setBillingInfo = setBillingInfo;
function getDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.getDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/draft-orders/{draftOrderId}',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getDraftOrder = getDraftOrder;
function getOrderDraftabilityStatus() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.getOrderDraftabilityStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/draft-orders/{orderId}/draftability',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrderDraftabilityStatus = getOrderDraftabilityStatus;
function commitDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.commitDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/commit',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.commitDraftOrder = commitDraftOrder;
function createOrderFromDraft() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createOrderFromDraft(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/create-order',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createOrderFromDraft = createOrderFromDraft;
function deleteDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.deleteDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/draft-orders/{draftOrderId}',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteDraftOrder = deleteDraftOrder;
function queryDraftOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.queryDraftOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryDraftOrders = queryDraftOrders;
function setTaxExemption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setTaxExemption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-tax-exemption',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setTaxExemption = setTaxExemption;
