// @wix/ecom_draft-orders/build/cjs/src/ecom-v1-draft-order-draft-orders.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeightUnit = exports.WebhookIdentityType = exports.VatType = exports.TaxableAddressType = exports.SubscriptionFrequency = exports.SourceType = exports.SortOrder = exports.RuleType = exports.PickupMethod = exports.PickupDetailsPickupMethod = exports.PaymentStatus = exports.PaymentOptionType = exports.OrderStatus = exports.NonDraftableReason = exports.LineItemQuantityChangeType = exports.JurisdictionType = exports.ItemTypeItemType = exports.FulfillmentStatus = exports.FileType = exports.EditingStatus = exports.DiscountType = exports.DiscountReason = exports.DescriptionLineType = exports.ChannelType = exports.AttributionSource = exports.AppliedDiscountDiscountType = exports.ActivityType = exports.setTaxExemption = exports.queryDraftOrders = exports.deleteDraftOrder = exports.createOrderFromDraft = exports.commitDraftOrder = exports.getOrderDraftabilityStatus = exports.getDraftOrder = exports.setBillingInfo = exports.setRecipientInfo = exports.setBuyerInfo = exports.setShippingInfo = exports.deleteCustomAdditionalFees = exports.createCustomAdditionalFees = exports.setAdditionalFees = exports.deleteCustomDiscounts = exports.createCustomDiscounts = exports.setDiscounts = exports.updateLineItems = exports.addLineItemsToDraftOrder = exports.createEmptyDraftOrder = exports.createDraftOrder = exports.__metadata = void 0;
const ecom_v1_draft_order_draft_orders_universal_js_1 = require("./ecom-v1-draft-order-draft-orders.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createDraftOrder(httpClient) {
    return (orderId) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.createDraftOrder)(orderId, 
    // @ts-ignore
    { httpClient });
}
exports.createDraftOrder = createDraftOrder;
function createEmptyDraftOrder(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.createEmptyDraftOrder)(
    // @ts-ignore
    { httpClient });
}
exports.createEmptyDraftOrder = createEmptyDraftOrder;
function addLineItemsToDraftOrder(httpClient) {
    return (draftOrderId, options) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.addLineItemsToDraftOrder)(draftOrderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.addLineItemsToDraftOrder = addLineItemsToDraftOrder;
function updateLineItems(httpClient) {
    return (draftOrderId, options) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.updateLineItems)(draftOrderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateLineItems = updateLineItems;
function setDiscounts(httpClient) {
    return (draftOrderId, discounts) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.setDiscounts)(draftOrderId, discounts, 
    // @ts-ignore
    { httpClient });
}
exports.setDiscounts = setDiscounts;
function createCustomDiscounts(httpClient) {
    return (draftOrderId, options) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.createCustomDiscounts)(draftOrderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.createCustomDiscounts = createCustomDiscounts;
function deleteCustomDiscounts(httpClient) {
    return (draftOrderId, discountIds) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.deleteCustomDiscounts)(draftOrderId, discountIds, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCustomDiscounts = deleteCustomDiscounts;
function setAdditionalFees(httpClient) {
    return (draftOrderId, additionalFees) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.setAdditionalFees)(draftOrderId, additionalFees, 
    // @ts-ignore
    { httpClient });
}
exports.setAdditionalFees = setAdditionalFees;
function createCustomAdditionalFees(httpClient) {
    return (draftOrderId, options) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.createCustomAdditionalFees)(draftOrderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.createCustomAdditionalFees = createCustomAdditionalFees;
function deleteCustomAdditionalFees(httpClient) {
    return (draftOrderId, customAdditionalFees) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.deleteCustomAdditionalFees)(draftOrderId, customAdditionalFees, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCustomAdditionalFees = deleteCustomAdditionalFees;
function setShippingInfo(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.setShippingInfo)(
    // @ts-ignore
    { httpClient });
}
exports.setShippingInfo = setShippingInfo;
function setBuyerInfo(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.setBuyerInfo)(
    // @ts-ignore
    { httpClient });
}
exports.setBuyerInfo = setBuyerInfo;
function setRecipientInfo(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.setRecipientInfo)(
    // @ts-ignore
    { httpClient });
}
exports.setRecipientInfo = setRecipientInfo;
function setBillingInfo(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.setBillingInfo)(
    // @ts-ignore
    { httpClient });
}
exports.setBillingInfo = setBillingInfo;
function getDraftOrder(httpClient) {
    return (draftOrderId) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.getDraftOrder)(draftOrderId, 
    // @ts-ignore
    { httpClient });
}
exports.getDraftOrder = getDraftOrder;
function getOrderDraftabilityStatus(httpClient) {
    return (orderId) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.getOrderDraftabilityStatus)(orderId, 
    // @ts-ignore
    { httpClient });
}
exports.getOrderDraftabilityStatus = getOrderDraftabilityStatus;
function commitDraftOrder(httpClient) {
    return (draftOrderId, options) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.commitDraftOrder)(draftOrderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.commitDraftOrder = commitDraftOrder;
function createOrderFromDraft(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.createOrderFromDraft)(
    // @ts-ignore
    { httpClient });
}
exports.createOrderFromDraft = createOrderFromDraft;
function deleteDraftOrder(httpClient) {
    return (draftOrderId) => (0, ecom_v1_draft_order_draft_orders_universal_js_1.deleteDraftOrder)(draftOrderId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteDraftOrder = deleteDraftOrder;
function queryDraftOrders(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.queryDraftOrders)(
    // @ts-ignore
    { httpClient });
}
exports.queryDraftOrders = queryDraftOrders;
function setTaxExemption(httpClient) {
    return () => (0, ecom_v1_draft_order_draft_orders_universal_js_1.setTaxExemption)(
    // @ts-ignore
    { httpClient });
}
exports.setTaxExemption = setTaxExemption;
var ecom_v1_draft_order_draft_orders_universal_js_2 = require("./ecom-v1-draft-order-draft-orders.universal.js");
Object.defineProperty(exports, "ActivityType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.ActivityType; } });
Object.defineProperty(exports, "AppliedDiscountDiscountType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.AppliedDiscountDiscountType; } });
Object.defineProperty(exports, "AttributionSource", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.AttributionSource; } });
Object.defineProperty(exports, "ChannelType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.ChannelType; } });
Object.defineProperty(exports, "DescriptionLineType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.DescriptionLineType; } });
Object.defineProperty(exports, "DiscountReason", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.DiscountReason; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "EditingStatus", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.EditingStatus; } });
Object.defineProperty(exports, "FileType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.FileType; } });
Object.defineProperty(exports, "FulfillmentStatus", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.FulfillmentStatus; } });
Object.defineProperty(exports, "ItemTypeItemType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.ItemTypeItemType; } });
Object.defineProperty(exports, "JurisdictionType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.JurisdictionType; } });
Object.defineProperty(exports, "LineItemQuantityChangeType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.LineItemQuantityChangeType; } });
Object.defineProperty(exports, "NonDraftableReason", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.NonDraftableReason; } });
Object.defineProperty(exports, "OrderStatus", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.OrderStatus; } });
Object.defineProperty(exports, "PaymentOptionType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.PaymentOptionType; } });
Object.defineProperty(exports, "PaymentStatus", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.PaymentStatus; } });
Object.defineProperty(exports, "PickupDetailsPickupMethod", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.PickupDetailsPickupMethod; } });
Object.defineProperty(exports, "PickupMethod", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.PickupMethod; } });
Object.defineProperty(exports, "RuleType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.RuleType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "SourceType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.SourceType; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "TaxableAddressType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.TaxableAddressType; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.VatType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return ecom_v1_draft_order_draft_orders_universal_js_2.WeightUnit; } });
