// @wix/ecom_gift-vouchers/build/cjs/src/ecom-v1-gift-card-gift-vouchers.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.voidTransaction = exports.redeemGiftCard = exports.getGiftCard = void 0;
const ambassadorWixEcomV1GiftCard = __importStar(require("./ecom-v1-gift-card-gift-vouchers.http.js"));
function getGiftCard() {
    const payload = { code: ':code' };
    const getRequestOptions = ambassadorWixEcomV1GiftCard.getGiftCard(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/gift-cards/{code}',
        pathParams: { code: 'code' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getGiftCard = getGiftCard;
function redeemGiftCard() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1GiftCard.redeemGiftCard(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/gift-cards/redeem',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.redeemGiftCard = redeemGiftCard;
function voidTransaction() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1GiftCard.voidTransaction(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/gift-cards/void',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.voidTransaction = voidTransaction;
