// @wix/ecom_validations/build/es/src/interfaces-ecommerce-v1-validations-provider.public.js
import { ServicePluginDefinition } from '@wix/sdk-types';
import { transformRESTAddressToSDKAddress } from '@wix/sdk-runtime/transformations/address';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
export const provideHandlers = ServicePluginDefinition('ECOM_VALIDATIONS', [
    {
        name: 'getValidationViolations',
        primaryHttpMappingPath: '/v1/get-violations',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = transformPaths(payload, [
                    {
                        transformFn: transformRESTFloatToSDKFloat,
                        paths: [
                            {
                                path: 'request.validationInfo.lineItems.physicalProperties.weight',
                            },
                        ],
                    },
                    {
                        transformFn: transformRESTTimestampToSDKTimestamp,
                        paths: [
                            {
                                path: 'request.validationInfo.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
                            },
                        ],
                    },
                    {
                        transformFn: transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.validationInfo.billingInfo.address' },
                            { path: 'request.validationInfo.shippingAddress.address' },
                        ],
                    },
                ]);
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
]);
