// @wix/ecom_shipping-options/build/es/src/ecom-v1-shipping-option-shipping-options.meta.js
import * as ambassadorWixEcomV1ShippingOption from './ecom-v1-shipping-option-shipping-options.http.js';
export function createShippingOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.createShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shipping-options',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getShippingOption() {
    const payload = { shippingOptionId: ':shippingOptionId' };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.getShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shipping-options/{shippingOptionId}',
        pathParams: { shippingOptionId: 'shippingOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateShippingOption() {
    const payload = { shippingOption: { id: ':shippingOptionId' } };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.updateShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/shipping-options/{shippingOption.id}',
        pathParams: { shippingOptionId: 'shippingOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteShippingOption() {
    const payload = { shippingOptionId: ':shippingOptionId' };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.deleteShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/shipping-options/{shippingOptionId}',
        pathParams: { shippingOptionId: 'shippingOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryShippingOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.queryShippingOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shipping-options/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shipping-options/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
