// @wix/ecom_order-fulfillments/build/es/src/ecom-v1-fulfillments-order-fulfillments.meta.js
import * as ambassadorWixEcomV1Fulfillments from './ecom-v1-fulfillments-order-fulfillments.http.js';
export function listFulfillmentsForSingleOrder() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.listFulfillmentsForSingleOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/fulfillments/orders/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listFulfillmentsForMultipleOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.listFulfillmentsForMultipleOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/fulfillments/list-by-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createFulfillment() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.createFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/fulfillments/orders/{orderId}/create-fulfillment',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateFulfillment() {
    const payload = {
        fulfillment: { id: ':fulfillmentId' },
        orderId: ':orderId',
    };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.updateFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/fulfillments/{fulfillment.id}/orders/{orderId}',
        pathParams: { fulfillmentId: 'fulfillmentId', orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteFulfillment() {
    const payload = {
        fulfillmentId: ':fulfillmentId',
        orderId: ':orderId',
    };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.deleteFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/fulfillments/{fulfillmentId}/orders/{orderId}',
        pathParams: { fulfillmentId: 'fulfillmentId', orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateFulfillments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.bulkCreateFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/fulfillments/orders/bulk/create-fulfillments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
