// @wix/ecom_discounts/build/es/src/service-plugins-types.js
export var TargetType;
(function (TargetType) {
    TargetType["UNKNOWN_TARGET"] = "UNKNOWN_TARGET";
    TargetType["GLOBAL"] = "GLOBAL";
    TargetType["SHIPPING"] = "SHIPPING";
    TargetType["LINE_ITEMS"] = "LINE_ITEMS";
})(TargetType || (TargetType = {}));
export var Type;
(function (Type) {
    Type["UNKNOWN_RATE"] = "UNKNOWN_RATE";
    Type["FIXED_AMOUNT"] = "FIXED_AMOUNT";
    Type["PERCENTAGE"] = "PERCENTAGE";
})(Type || (Type = {}));
export var DiscountSource;
(function (DiscountSource) {
    DiscountSource["UNKNOWN_SOURCE"] = "UNKNOWN_SOURCE";
    DiscountSource["COUPON"] = "COUPON";
    DiscountSource["AUTOMATIC"] = "AUTOMATIC";
    DiscountSource["CUSTOM"] = "CUSTOM";
})(DiscountSource || (DiscountSource = {}));
export var IdentityType;
(function (IdentityType) {
    IdentityType["UNKNOWN"] = "UNKNOWN";
    IdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    IdentityType["MEMBER"] = "MEMBER";
    IdentityType["WIX_USER"] = "WIX_USER";
    IdentityType["APP"] = "APP";
})(IdentityType || (IdentityType = {}));
