// @wix/ecom_discounts/build/es/src/service-plugins-error-classes.js
export class CouponLimitPerCustomerExceededWixError extends Error {
    constructor() {
        super('CouponLimitPerCustomerExceeded');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED';
        this.name = 'CouponLimitPerCustomerExceeded';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponLimitPerCustomerExceeded',
            applicationCode: 'ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED',
        };
    }
}
/** @hidden */
CouponLimitPerCustomerExceededWixError.__type = 'wix_spi_error';
export class CouponIsDisabledWixError extends Error {
    constructor() {
        super('CouponIsDisabled');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_IS_DISABLED';
        this.name = 'CouponIsDisabled';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponIsDisabled',
            applicationCode: 'ERROR_COUPON_IS_DISABLED',
        };
    }
}
/** @hidden */
CouponIsDisabledWixError.__type = 'wix_spi_error';
export class CouponUsageExceededWixError extends Error {
    constructor() {
        super('CouponUsageExceeded');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_USAGE_EXCEEDED';
        this.name = 'CouponUsageExceeded';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponUsageExceeded',
            applicationCode: 'ERROR_COUPON_USAGE_EXCEEDED',
        };
    }
}
/** @hidden */
CouponUsageExceededWixError.__type = 'wix_spi_error';
export class CouponHasExpiredWixError extends Error {
    constructor() {
        super('CouponHasExpired');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_HAS_EXPIRED';
        this.name = 'CouponHasExpired';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponHasExpired',
            applicationCode: 'ERROR_COUPON_HAS_EXPIRED',
        };
    }
}
/** @hidden */
CouponHasExpiredWixError.__type = 'wix_spi_error';
export class CouponIsNotActiveYetWixError extends Error {
    constructor() {
        super('CouponIsNotActiveYet');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_IS_NOT_ACTIVE_YET';
        this.name = 'CouponIsNotActiveYet';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponIsNotActiveYet',
            applicationCode: 'ERROR_COUPON_IS_NOT_ACTIVE_YET',
        };
    }
}
/** @hidden */
CouponIsNotActiveYetWixError.__type = 'wix_spi_error';
export class CouponDoesNotExistWixError extends Error {
    constructor() {
        super('CouponDoesNotExist');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'ERROR_COUPON_DOES_NOT_EXIST';
        this.name = 'CouponDoesNotExist';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponDoesNotExist',
            applicationCode: 'ERROR_COUPON_DOES_NOT_EXIST',
        };
    }
}
/** @hidden */
CouponDoesNotExistWixError.__type = 'wix_spi_error';
