// @wix/ecom_subscription-contracts/build/cjs/src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.querySubscriptionContracts = exports.getSubscriptionContract = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl(opts) {
    const domainToMappings = {};
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_subscription-contracts';
/** Retrieves a SubscriptionContract. */
function getSubscriptionContract(payload) {
    function __getSubscriptionContract({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.subscription_contracts.v1.subscription_contract',
            method: 'GET',
            methodFqn: 'wix.ecom.subscription_contracts.v1.SubscriptionContracts.GetSubscriptionContract',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
                protoPath: '/v1/subscription-contracts/{subscriptionContractId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'subscriptionContract.createdDate' },
                        { path: 'subscriptionContract.updatedDate' },
                        { path: 'subscriptionContract.subscriptionSettings.startDate' },
                        {
                            path: 'subscriptionContract.lineItems.image.urlExpirationDate',
                        },
                        {
                            path: 'subscriptionContract.lineItems.digitalFile.expirationDate',
                        },
                        {
                            path: 'subscriptionContract.lineItems.subscriptionInfo.subscriptionSettings.startDate',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.deliverByDate',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.from',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.to',
                        },
                    ],
                },
                {
                    transformFn: float_1.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'subscriptionContract.lineItems.image.focalPoint.x' },
                        { path: 'subscriptionContract.lineItems.image.focalPoint.y' },
                        {
                            path: 'subscriptionContract.lineItems.physicalProperties.weight',
                        },
                        {
                            path: 'subscriptionContract.billingInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.billingInfo.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContract.recipientInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.recipientInfo.address.geocode.longitude',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getSubscriptionContract;
}
exports.getSubscriptionContract = getSubscriptionContract;
/**
 * Retrieves a list of SubscriptionContracts, given the provided [paging, filtering, and sorting][1].
 *
 * Up to 300 SubscriptionContracts can be returned per request.
 *
 * To learn how to query SubscriptionContracts, see [API Query Language][2].
 *
 * [1]: https://dev.wix.com/api/rest/getting-started/sorting-and-paging
 * [2]: https://dev.wix.com/api/rest/getting-started/api-query-language
 */
function querySubscriptionContracts(payload) {
    function __querySubscriptionContracts({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.subscription_contracts.v1.subscription_contract',
            method: 'POST',
            methodFqn: 'wix.ecom.subscription_contracts.v1.SubscriptionContracts.QuerySubscriptionContracts',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
                protoPath: '/v1/subscription-contracts/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'subscriptionContracts.createdDate' },
                        { path: 'subscriptionContracts.updatedDate' },
                        { path: 'subscriptionContracts.subscriptionSettings.startDate' },
                        {
                            path: 'subscriptionContracts.lineItems.image.urlExpirationDate',
                        },
                        {
                            path: 'subscriptionContracts.lineItems.digitalFile.expirationDate',
                        },
                        {
                            path: 'subscriptionContracts.lineItems.subscriptionInfo.subscriptionSettings.startDate',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.deliverByDate',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.from',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.to',
                        },
                    ],
                },
                {
                    transformFn: float_1.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'subscriptionContracts.lineItems.image.focalPoint.x' },
                        { path: 'subscriptionContracts.lineItems.image.focalPoint.y' },
                        {
                            path: 'subscriptionContracts.lineItems.physicalProperties.weight',
                        },
                        {
                            path: 'subscriptionContracts.billingInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.billingInfo.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContracts.recipientInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.recipientInfo.address.geocode.longitude',
                        },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'POST',
                    url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
                        protoPath: '/v1/subscription-contracts/query',
                        data: payload,
                        host,
                    }),
                    data: payload,
                },
            ],
        };
        return metadata;
    }
    return __querySubscriptionContracts;
}
exports.querySubscriptionContracts = querySubscriptionContracts;
