// @wix/ecom_subscription-contracts/build/cjs/src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeightUnit = exports.WebhookIdentityType = exports.VatType = exports.V2SubscriptionFrequency = exports.TaxableAddressType = exports.SubscriptionFrequency = exports.SortOrder = exports.PickupMethod = exports.PaymentOptionType = exports.JurisdictionType = exports.ItemTypeItemType = exports.DiscountType = exports.DiscountReason = exports.DescriptionLineType = exports.ChannelType = exports.onSubscriptionContractUpdated = exports.onSubscriptionContractDeleted = exports.onSubscriptionContractCreated = exports.querySubscriptionContracts = exports.getSubscriptionContract = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_1 = require("./ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function getSubscriptionContract(httpClient) {
    return (subscriptionContractId) => (0, ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_1.getSubscriptionContract)(subscriptionContractId, 
    // @ts-ignore
    { httpClient });
}
exports.getSubscriptionContract = getSubscriptionContract;
function querySubscriptionContracts(httpClient) {
    return () => (0, ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_1.querySubscriptionContracts)(
    // @ts-ignore
    { httpClient });
}
exports.querySubscriptionContracts = querySubscriptionContracts;
exports.onSubscriptionContractCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.subscription_contracts.v1.subscription_contract_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.subscriptionSettings.startDate' },
            { path: 'entity.lineItems.digitalFile.expirationDate' },
            {
                path: 'entity.lineItems.subscriptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.shippingInfo.logistics.deliverByDate' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.from' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.to' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.lineItems.image' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'entity.lineItems.physicalProperties.weight' }],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.billingInfo.address' },
            {
                path: 'entity.shippingInfo.logistics.shippingDestination.address',
            },
            { path: 'entity.shippingInfo.logistics.pickupDetails.address' },
            { path: 'entity.recipientInfo.address' },
        ],
    },
])))();
exports.onSubscriptionContractDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.subscription_contracts.v1.subscription_contract_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'undefined.subscriptionSettings.startDate' },
            { path: 'undefined.lineItems.digitalFile.expirationDate' },
            {
                path: 'undefined.lineItems.subscriptionInfo.subscriptionSettings.startDate',
            },
            { path: 'undefined.shippingInfo.logistics.deliverByDate' },
            { path: 'undefined.shippingInfo.logistics.deliveryTimeSlot.from' },
            { path: 'undefined.shippingInfo.logistics.deliveryTimeSlot.to' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'undefined.lineItems.image' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'undefined.lineItems.physicalProperties.weight' }],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'undefined.billingInfo.address' },
            {
                path: 'undefined.shippingInfo.logistics.shippingDestination.address',
            },
            { path: 'undefined.shippingInfo.logistics.pickupDetails.address' },
            { path: 'undefined.recipientInfo.address' },
        ],
    },
])))();
exports.onSubscriptionContractUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.subscription_contracts.v1.subscription_contract_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.subscriptionSettings.startDate' },
            { path: 'entity.lineItems.digitalFile.expirationDate' },
            {
                path: 'entity.lineItems.subscriptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.shippingInfo.logistics.deliverByDate' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.from' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.to' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.lineItems.image' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'entity.lineItems.physicalProperties.weight' }],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.billingInfo.address' },
            {
                path: 'entity.shippingInfo.logistics.shippingDestination.address',
            },
            { path: 'entity.shippingInfo.logistics.pickupDetails.address' },
            { path: 'entity.recipientInfo.address' },
        ],
    },
])))();
var ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2 = require("./ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.universal.js");
Object.defineProperty(exports, "ChannelType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.ChannelType; } });
Object.defineProperty(exports, "DescriptionLineType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.DescriptionLineType; } });
Object.defineProperty(exports, "DiscountReason", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.DiscountReason; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "ItemTypeItemType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.ItemTypeItemType; } });
Object.defineProperty(exports, "JurisdictionType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.JurisdictionType; } });
Object.defineProperty(exports, "PaymentOptionType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.PaymentOptionType; } });
Object.defineProperty(exports, "PickupMethod", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.PickupMethod; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "TaxableAddressType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.TaxableAddressType; } });
Object.defineProperty(exports, "V2SubscriptionFrequency", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.V2SubscriptionFrequency; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.VatType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return ecom_subscription_contracts_v1_subscription_contract_subscription_contracts_universal_js_2.WeightUnit; } });
