// @wix/image-kit/dist/cjs/helpers/responsiveImageUtils.js
"use strict";

exports.__esModule = true;
exports.getResponsiveImageProps = void 0;
var _max = require("../api/max");
const getResponsiveImageProps = (imageProps, targetWidth, targetHeight) => {
  var _srcset$dpr;
  const {
    displayMode,
    uri,
    width,
    height,
    name,
    crop,
    focalPoint,
    alignType,
    quality,
    upscaleMethod,
    hasAnimation,
    allowAnimatedTransform,
    encoding
  } = imageProps;
  const {
    srcset,
    css,
    uri: src
  } = (0, _max.getData)(displayMode, {
    id: uri,
    width,
    height,
    name,
    crop,
    focalPoint
  }, {
    width: targetWidth,
    height: targetHeight,
    alignment: alignType
  }, {
    focalPoint,
    name,
    quality: quality == null ? void 0 : quality.quality,
    upscaleMethod,
    hasAnimation,
    allowAnimatedTransform,
    useSrcset: true,
    encoding
  });
  const srcsetWithPrefix = srcset == null || (_srcset$dpr = srcset.dpr) == null ? void 0 : _srcset$dpr.map(s => /^[a-z]+:/.test(s) ? s : `${_max.STATIC_MEDIA_URL}${s}`);
  const fallbackSrc = `${_max.STATIC_MEDIA_URL}${src}`;
  const srcsetString = srcsetWithPrefix == null ? void 0 : srcsetWithPrefix.join(', ');
  return {
    fallbackSrc,
    srcset: srcsetString,
    css
  };
};
exports.getResponsiveImageProps = getResponsiveImageProps;
