// @wix/ecom_catalog/build/cjs/src/interfaces-ecommerce-v1-catalog-provider.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('ECOM_CATALOG', [
    {
        name: 'getCatalogItems',
        primaryHttpMappingPath: '/get-catalog-items',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: page_url_v2_1.transformSDKPageURLV2ToRESTPageURLV2,
                        paths: [{ path: 'catalogItems.data.url' }],
                    },
                    {
                        transformFn: image_1.transformSDKImageToRESTImage,
                        paths: [{ path: 'catalogItems.data.media' }],
                    },
                    {
                        transformFn: float_1.transformSDKFloatToRESTFloat,
                        paths: [{ path: 'catalogItems.data.physicalProperties.weight' }],
                    },
                    {
                        transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                        paths: [
                            {
                                path: 'catalogItems.data.subscriptionOptionInfo.subscriptionSettings.startDate',
                            },
                            { path: 'catalogItems.data.serviceProperties.scheduledDate' },
                        ],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
