// @wix/ecom_back-in-stock-settings/build/es/src/ecom-v1-back-in-stock-settings-back-in-stock-settings.meta.js
import * as ambassadorWixEcomV1BackInStockSettings from './ecom-v1-back-in-stock-settings-back-in-stock-settings.http.js';
export function startCollectingRequests() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1BackInStockSettings.startCollectingRequests(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/back-in-stock-notification-requests/settings/start-collecting',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function stopCollectingRequests() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1BackInStockSettings.stopCollectingRequests(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/back-in-stock-notification-requests/settings/stop-collecting',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1BackInStockSettings.getSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/back-in-stock-notification-requests/settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
