// @wix/ecom_back-in-stock-settings/build/cjs/src/ecom-v1-back-in-stock-settings-back-in-stock-settings.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSettings = exports.stopCollectingRequests = exports.startCollectingRequests = void 0;
const ambassadorWixEcomV1BackInStockSettings = __importStar(require("./ecom-v1-back-in-stock-settings-back-in-stock-settings.http.js"));
function startCollectingRequests() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1BackInStockSettings.startCollectingRequests(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/back-in-stock-notification-requests/settings/start-collecting',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.startCollectingRequests = startCollectingRequests;
function stopCollectingRequests() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1BackInStockSettings.stopCollectingRequests(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/back-in-stock-notification-requests/settings/stop-collecting',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.stopCollectingRequests = stopCollectingRequests;
function getSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1BackInStockSettings.getSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/back-in-stock-notification-requests/settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getSettings = getSettings;
