// @wix/ecom_custom-triggers/build/es/src/interfaces-ecommerce-v1-custom-trigger.public.js
import { ServicePluginDefinition } from '@wix/sdk-types';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
export const provideHandlers = ServicePluginDefinition('ECOM_DISCOUNTS_TRIGGER', [
    {
        name: 'listTriggers',
        primaryHttpMappingPath: '/v1/list-triggers',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
    {
        name: 'getEligibleTriggers',
        primaryHttpMappingPath: '/v1/get-eligible-triggers',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
]);
