// @wix/ecom_discount-rules/build/es/src/ecom-discounts-v1-discount-rule-discount-rules.meta.js
import * as ambassadorWixEcomDiscountsV1DiscountRule from './ecom-discounts-v1-discount-rule-discount-rules.http.js';
export function createDiscountRule() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.createDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/discount-rules',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getDiscountRule() {
    const payload = { discountRuleId: ':discountRuleId' };
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.getDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/discount-rules/{discountRuleId}',
        pathParams: { discountRuleId: 'discountRuleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDiscountRule() {
    const payload = { discountRule: { id: ':discountRuleId' } };
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.updateDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/discount-rules/{discountRule.id}',
        pathParams: { discountRuleId: 'discountRuleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteDiscountRule() {
    const payload = { discountRuleId: ':discountRuleId' };
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.deleteDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/discount-rules/{discountRuleId}',
        pathParams: { discountRuleId: 'discountRuleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryDiscountRules() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.queryDiscountRules(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/discount-rules/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
