// @wix/ecom_orders/build/es/src/ecom-v1-order-orders.meta.js
import * as ambassadorWixEcomV1Order from './ecom-v1-order-orders.http.js';
export function preparePaymentCollection() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.preparePaymentCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments-collector/orders/{ecomOrderId}/prepare-payment-collection',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getPaymentCollectabilityStatus() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.getPaymentCollectabilityStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/payments-collector/orders/{ecomOrderId}/payment-collectability-status',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function voidAuthorizedPayments() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.voidAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments-collector/orders/{ecomOrderId}/void-authorized-payments',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function captureAuthorizedPayments() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.captureAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments-collector/orders/{ecomOrderId}/capture-authorized-payments',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Order.getOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.searchOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/orders/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.createOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateOrder() {
    const payload = { order: { id: ':orderId' } };
    const getRequestOptions = ambassadorWixEcomV1Order.updateOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/orders/{order.id}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.bulkUpdateOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/orders/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cancelOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Order.cancelOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/orders/{id}/cancel',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateOrderTags() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.bulkUpdateOrderTags(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/orders/update-tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
