// @wix/ecom_draft-orders/build/es/src/ecom-v1-draft-order-draft-orders.meta.js
import * as ambassadorWixEcomV1DraftOrder from './ecom-v1-draft-order-draft-orders.http.js';
export function createDraftOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createEmptyDraftOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createEmptyDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/create-empty-draft-order',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addLineItemsToDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.addLineItemsToDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/add-line-items',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateLineItems() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.updateLineItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/line-items',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setDiscounts() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setDiscounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/set-discounts',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createCustomDiscounts() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createCustomDiscounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/create-custom-discounts',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCustomDiscounts() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.deleteCustomDiscounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/delete-custom-discounts',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setAdditionalFees() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setAdditionalFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/set-additional-fees',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createCustomAdditionalFees() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createCustomAdditionalFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/create-additional-fees',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCustomAdditionalFees() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.deleteCustomAdditionalFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/delete-custom-additional-fees',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setShippingInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setShippingInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-shipping-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setBuyerInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setBuyerInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-buyer-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setRecipientInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setRecipientInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-recipient-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setBillingInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setBillingInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-billing-info',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.getDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/draft-orders/{draftOrderId}',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getOrderDraftabilityStatus() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.getOrderDraftabilityStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/draft-orders/{orderId}/draftability',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function commitDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.commitDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/commit',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createOrderFromDraft() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.createOrderFromDraft(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/{draftOrderId}/create-order',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteDraftOrder() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.deleteDraftOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/draft-orders/{draftOrderId}',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryDraftOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.queryDraftOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/draft-orders/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setTaxExemption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DraftOrder.setTaxExemption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v1/draft-orders/set-tax-exemption',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
