// @wix/identity_authentication/build/es/src/iam-authentication-v1-authentication-authentication.public.js
import { changePassword as universalChangePassword, loginCallback as universalLoginCallback, loginV2 as universalLoginV2, loginWithIdpConnection as universalLoginWithIdpConnection, loginWithIdpConnectionTokenParams as universalLoginWithIdpConnectionTokenParams, logout as universalLogout, registerV2 as universalRegisterV2, signOn as universalSignOn, verify as universalVerify, } from './iam-authentication-v1-authentication-authentication.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/identity' };
export function registerV2(httpClient) {
    return (loginId, options) => universalRegisterV2(loginId, options, 
    // @ts-ignore
    { httpClient });
}
export function loginV2(httpClient) {
    return (loginId, options) => universalLoginV2(loginId, options, 
    // @ts-ignore
    { httpClient });
}
export function changePassword(httpClient) {
    return (newPassword) => universalChangePassword(newPassword, 
    // @ts-ignore
    { httpClient });
}
export function loginWithIdpConnection(httpClient) {
    return (identifiers, tenantType, options) => universalLoginWithIdpConnection(identifiers, tenantType, options, 
    // @ts-ignore
    { httpClient });
}
export function loginCallback(httpClient) {
    return (options) => universalLoginCallback(options, 
    // @ts-ignore
    { httpClient });
}
export function loginWithIdpConnectionTokenParams(httpClient) {
    return (options) => universalLoginWithIdpConnectionTokenParams(options, 
    // @ts-ignore
    { httpClient });
}
export function signOn(httpClient) {
    return (loginId, options) => universalSignOn(loginId, options, 
    // @ts-ignore
    { httpClient });
}
export function logout(httpClient) {
    return (options) => universalLogout(options, 
    // @ts-ignore
    { httpClient });
}
export function verify(httpClient) {
    return (factorType, options) => universalVerify(factorType, options, 
    // @ts-ignore
    { httpClient });
}
export { AddressTag, EmailTag, FactorType, PhoneTag, PrivacyStatus, Reason, StateType, Status, StatusName, TenantType, } from './iam-authentication-v1-authentication-authentication.universal.js';
