// @wix/identity_authentication/build/es/src/iam-authentication-v1-authentication-authentication.meta.js
import * as ambassadorWixIamAuthenticationV1Authentication from './iam-authentication-v1-authentication-authentication.http.js';
export function registerV2() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.registerV2(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/register',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function loginV2() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginV2(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/login',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function changePassword() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.changePassword(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/change-password',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function loginWithIdpConnection() {
    const payload = {
        tenantId: ':tenantId',
        idpConnectionId: ':idpConnectionId',
    };
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginWithIdpConnection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/sso/login/{tenantId}/{idpConnectionId}',
        pathParams: { tenantId: 'tenantId', idpConnectionId: 'idpConnectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function loginCallback() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginCallback(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/callback',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function loginWithIdpConnectionTokenParams() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginWithIdpConnectionTokenParams(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/sso/login-with-token-params',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function signOn() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.signOn(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/sign-on',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function logout() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.logout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/logout',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function verify() {
    const payload = { factorType: ':factorType' };
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.verify(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/{factorType}/verify',
        pathParams: { factorType: 'factorType' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
