// @wix/ecom_currencies/build/cjs/src/ecommerce-currency-converter-v1-currency-rate-currencies.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getConversionRate = exports.convertCurrency = exports.listCurrencies = void 0;
const ambassadorWixEcommerceCurrencyConverterV1CurrencyRate = __importStar(require("./ecommerce-currency-converter-v1-currency-rate-currencies.http.js"));
function listCurrencies() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcommerceCurrencyConverterV1CurrencyRate.listCurrencies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/currencies',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listCurrencies = listCurrencies;
function convertCurrency() {
    const payload = { from: ':from', to: ':to' };
    const getRequestOptions = ambassadorWixEcommerceCurrencyConverterV1CurrencyRate.convertCurrency(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/currencies/amounts/{from}/convert/{to}',
        pathParams: { from: 'from', to: 'to' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.convertCurrency = convertCurrency;
function getConversionRate() {
    const payload = { from: ':from', to: ':to' };
    const getRequestOptions = ambassadorWixEcommerceCurrencyConverterV1CurrencyRate.conversionRate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/currencies/rate/{from}/convert/{to}',
        pathParams: { from: 'from', to: 'to' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getConversionRate = getConversionRate;
