// @wix/ecom_currencies/build/cjs/src/ecommerce-currency-converter-v1-currency-rate-currencies.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.conversionRate = exports.convertCurrency = exports.listCurrencies = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/currency-converter',
                destPath: '',
            },
        ],
        'cronulla-jobs._base_domain_': [
            {
                srcPath: '/currency-converter-bo',
                destPath: '',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
        'currency-converter.wixapps.net': [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/currency_converter',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/currency-converter',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_currencies';
/** Returns an array of currencies. The array lists all currencies for which Wix supports conversion and their symbols. */
function listCurrencies(payload) {
    function __listCurrencies({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecommerce.currency_converter.v1.currency_rate',
            method: 'GET',
            methodFqn: 'com.wixpress.currency.converter.api.v1.CurrencyConverter.ListCurrencies',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
                protoPath: '/v1/currencies',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __listCurrencies;
}
exports.listCurrencies = listCurrencies;
/**
 * Returns an array of amounts converted from the original (`from`) currency to the target (`to`) currency and the timestamp for the conversion rate used.
 *
 *
 * Use the `convertCurrency()` function to convert an array of one or more amounts between two currencies. The `convertCurrency()` function returns an array of converted amounts and the timestamp for the conversion rate used.
 *
 * > **Note**:  The currency codes used must exist in the array of supported currencies returned by the [`listCurrencies()`](#listcurrencies) function.
 *
 */
function convertCurrency(payload) {
    function __convertCurrency({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecommerce.currency_converter.v1.currency_rate',
            method: 'POST',
            methodFqn: 'com.wixpress.currency.converter.api.v1.CurrencyConverter.ConvertCurrency',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
                protoPath: '/v1/currencies/amounts/{from}/convert/{to}',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'rateTimestamp' }],
                },
            ]),
        };
        return metadata;
    }
    return __convertCurrency;
}
exports.convertCurrency = convertCurrency;
/** Returns the conversion rate between 2 currencies. */
function conversionRate(payload) {
    function __conversionRate({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecommerce.currency_converter.v1.currency_rate',
            method: 'GET',
            methodFqn: 'com.wixpress.currency.converter.api.v1.CurrencyConverter.ConversionRate',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressCurrencyConverterApiV1CurrencyConverterUrl({
                protoPath: '/v1/currencies/rate/{from}/convert/{to}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'rateTimestamp' }],
                },
            ]),
        };
        return metadata;
    }
    return __conversionRate;
}
exports.conversionRate = conversionRate;
