// @wix/ecom_order-transactions/build/cjs/src/ecom-v1-order-transactions-order-transactions.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkUpdatePaymentStatuses = exports.updatePaymentStatus = exports.addPayments = exports.listTransactionsForMultipleOrders = exports.listTransactionsForSingleOrder = void 0;
const ambassadorWixEcomV1OrderTransactions = __importStar(require("./ecom-v1-order-transactions-order-transactions.http.js"));
function listTransactionsForSingleOrder() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.listTransactionsForSingleOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/payments/orders/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listTransactionsForSingleOrder = listTransactionsForSingleOrder;
function listTransactionsForMultipleOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.listTransactionsForMultipleOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments/list-by-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listTransactionsForMultipleOrders = listTransactionsForMultipleOrders;
function addPayments() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.addPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments/orders/{orderId}/add-payment',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addPayments = addPayments;
function updatePaymentStatus() {
    const payload = { paymentId: ':paymentId', orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.updatePaymentStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments/{paymentId}/orders/{orderId}/update-payment-transaction-status',
        pathParams: { paymentId: 'paymentId', orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updatePaymentStatus = updatePaymentStatus;
function bulkUpdatePaymentStatuses() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderTransactions.bulkUpdatePaymentStatuses(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/payments/update-payment-transaction-status',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdatePaymentStatuses = bulkUpdatePaymentStatuses;
