// @wix/ecom_tippable-staff/build/cjs/src/tips-staff-v1-staff-tippable-staff.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listTippableStaff = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixTipsStaffV1TipStaffServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'bo._base_domain_': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'wixbo.ai': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'wix-bo.com': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/tips-staff',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_tippable-staff';
/**
 * Retrieves a list of up to 100 staff who are eligible to receive tips, given
 * the provided filtering.
 */
function listTippableStaff(payload) {
    function __listTippableStaff({ host }) {
        const metadata = {
            entityFqdn: 'wix.tips.staff.v1.staff',
            method: 'GET',
            methodFqn: 'wix.tips.staff.v1.TipStaffService.ListTippableStaff',
            packageName: PACKAGE_NAME,
            url: resolveWixTipsStaffV1TipStaffServiceUrl({
                protoPath: '/v1/eligible-staff',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __listTippableStaff;
}
exports.listTippableStaff = listTippableStaff;
