// @wix/ecom_current-cart/build/cjs/src/ecom-v1-cart-current-cart.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteCurrentCart = exports.estimateCurrentCartTotals = exports.updateCurrentCartLineItemQuantity = exports.removeCouponFromCurrentCart = exports.createCheckoutFromCurrentCart = exports.removeLineItemsFromCurrentCart = exports.addToCurrentCart = exports.updateCurrentCart = exports.getCurrentCart = void 0;
const ambassadorWixEcomV1Cart = __importStar(require("./ecom-v1-cart-current-cart.http.js"));
function getCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.currentCartGetCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/carts/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCurrentCart = getCurrentCart;
function updateCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.updateCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/carts/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCurrentCart = updateCurrentCart;
function addToCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.addToCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/add-to-cart',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addToCurrentCart = addToCurrentCart;
function removeLineItemsFromCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.removeLineItemsFromCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/remove-line-items',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeLineItemsFromCurrentCart = removeLineItemsFromCurrentCart;
function createCheckoutFromCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.createCheckoutFromCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/create-checkout',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCheckoutFromCurrentCart = createCheckoutFromCurrentCart;
function removeCouponFromCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.removeCouponFromCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/remove-coupon',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeCouponFromCurrentCart = removeCouponFromCurrentCart;
function updateCurrentCartLineItemQuantity() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.updateCurrentCartLineItemQuantity(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/update-line-items-quantity',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCurrentCartLineItemQuantity = updateCurrentCartLineItemQuantity;
function estimateCurrentCartTotals() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.estimateCurrentCartTotals(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/estimate-totals',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.estimateCurrentCartTotals = estimateCurrentCartTotals;
function deleteCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.deleteCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/carts/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCurrentCart = deleteCurrentCart;
