// @wix/ecom_current-cart/build/cjs/src/ecom-v1-cart-current-cart.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeightUnit = exports.WebhookIdentityType = exports.VatType = exports.TaxableAddressType = exports.SuggestedFix = exports.SubscriptionFrequency = exports.Severity = exports.RuleType = exports.RateType = exports.PickupMethod = exports.PaymentOptionType = exports.NameInOther = exports.NameInLineItem = exports.ManualCalculationReason = exports.JurisdictionType = exports.ItemTypeItemType = exports.ItemAvailabilityStatus = exports.FileType = exports.FallbackReason = exports.DiscountType = exports.DescriptionLineType = exports.ChargeType = exports.ChannelType = exports.AppliedDiscountDiscountType = exports.AdditionalFeeSource = exports.onCartCreated = exports.onCartUpdated = exports.onCartDeleted = exports.deleteCurrentCart = exports.estimateCurrentCartTotals = exports.updateCurrentCartLineItemQuantity = exports.removeCouponFromCurrentCart = exports.createCheckoutFromCurrentCart = exports.removeLineItemsFromCurrentCart = exports.addToCurrentCart = exports.updateCurrentCart = exports.getCurrentCart = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_cart_current_cart_universal_js_1 = require("./ecom-v1-cart-current-cart.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function getCurrentCart(httpClient) {
    return () => (0, ecom_v1_cart_current_cart_universal_js_1.getCurrentCart)(
    // @ts-ignore
    { httpClient });
}
exports.getCurrentCart = getCurrentCart;
function updateCurrentCart(httpClient) {
    return (options) => (0, ecom_v1_cart_current_cart_universal_js_1.updateCurrentCart)(options, 
    // @ts-ignore
    { httpClient });
}
exports.updateCurrentCart = updateCurrentCart;
function addToCurrentCart(httpClient) {
    return (options) => (0, ecom_v1_cart_current_cart_universal_js_1.addToCurrentCart)(options, 
    // @ts-ignore
    { httpClient });
}
exports.addToCurrentCart = addToCurrentCart;
function removeLineItemsFromCurrentCart(httpClient) {
    return (lineItemIds) => (0, ecom_v1_cart_current_cart_universal_js_1.removeLineItemsFromCurrentCart)(lineItemIds, 
    // @ts-ignore
    { httpClient });
}
exports.removeLineItemsFromCurrentCart = removeLineItemsFromCurrentCart;
function createCheckoutFromCurrentCart(httpClient) {
    return (options) => (0, ecom_v1_cart_current_cart_universal_js_1.createCheckoutFromCurrentCart)(options, 
    // @ts-ignore
    { httpClient });
}
exports.createCheckoutFromCurrentCart = createCheckoutFromCurrentCart;
function removeCouponFromCurrentCart(httpClient) {
    return () => (0, ecom_v1_cart_current_cart_universal_js_1.removeCouponFromCurrentCart)(
    // @ts-ignore
    { httpClient });
}
exports.removeCouponFromCurrentCart = removeCouponFromCurrentCart;
function updateCurrentCartLineItemQuantity(httpClient) {
    return (lineItems) => (0, ecom_v1_cart_current_cart_universal_js_1.updateCurrentCartLineItemQuantity)(lineItems, 
    // @ts-ignore
    { httpClient });
}
exports.updateCurrentCartLineItemQuantity = updateCurrentCartLineItemQuantity;
function estimateCurrentCartTotals(httpClient) {
    return (options) => (0, ecom_v1_cart_current_cart_universal_js_1.estimateCurrentCartTotals)(options, 
    // @ts-ignore
    { httpClient });
}
exports.estimateCurrentCartTotals = estimateCurrentCartTotals;
function deleteCurrentCart(httpClient) {
    return () => (0, ecom_v1_cart_current_cart_universal_js_1.deleteCurrentCart)(
    // @ts-ignore
    { httpClient });
}
exports.deleteCurrentCart = deleteCurrentCart;
exports.onCartDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.cart_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            {
                path: 'undefined.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'undefined.lineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'undefined.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'undefined.lineItems.image' },
            { path: 'undefined.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'undefined.lineItems.physicalProperties.weight' },
            {
                path: 'undefined.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'undefined.contactInfo.address' }],
    },
])))();
exports.onCartUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.cart_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            {
                path: 'entity.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.lineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.image' },
            { path: 'entity.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.lineItems.physicalProperties.weight' },
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'entity.contactInfo.address' }],
    },
])))();
exports.onCartCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.cart_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            {
                path: 'entity.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.lineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.image' },
            { path: 'entity.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.lineItems.physicalProperties.weight' },
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'entity.contactInfo.address' }],
    },
])))();
var ecom_v1_cart_current_cart_universal_js_2 = require("./ecom-v1-cart-current-cart.universal.js");
Object.defineProperty(exports, "AdditionalFeeSource", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.AdditionalFeeSource; } });
Object.defineProperty(exports, "AppliedDiscountDiscountType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.AppliedDiscountDiscountType; } });
Object.defineProperty(exports, "ChannelType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.ChannelType; } });
Object.defineProperty(exports, "ChargeType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.ChargeType; } });
Object.defineProperty(exports, "DescriptionLineType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.DescriptionLineType; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "FallbackReason", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.FallbackReason; } });
Object.defineProperty(exports, "FileType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.FileType; } });
Object.defineProperty(exports, "ItemAvailabilityStatus", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.ItemAvailabilityStatus; } });
Object.defineProperty(exports, "ItemTypeItemType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.ItemTypeItemType; } });
Object.defineProperty(exports, "JurisdictionType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.JurisdictionType; } });
Object.defineProperty(exports, "ManualCalculationReason", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.ManualCalculationReason; } });
Object.defineProperty(exports, "NameInLineItem", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.NameInLineItem; } });
Object.defineProperty(exports, "NameInOther", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.NameInOther; } });
Object.defineProperty(exports, "PaymentOptionType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.PaymentOptionType; } });
Object.defineProperty(exports, "PickupMethod", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.PickupMethod; } });
Object.defineProperty(exports, "RateType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.RateType; } });
Object.defineProperty(exports, "RuleType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.RuleType; } });
Object.defineProperty(exports, "Severity", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.Severity; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "SuggestedFix", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.SuggestedFix; } });
Object.defineProperty(exports, "TaxableAddressType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.TaxableAddressType; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.VatType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return ecom_v1_cart_current_cart_universal_js_2.WeightUnit; } });
