// @wix/ecom_cart/build/cjs/src/ecom-v1-cart-cart.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteCart = exports.estimateTotals = exports.updateLineItemsQuantity = exports.removeCoupon = exports.createCheckout = exports.removeLineItems = exports.addToCart = exports.getCart = exports.updateCart = exports.createCart = void 0;
const ambassadorWixEcomV1Cart = __importStar(require("./ecom-v1-cart-cart.http.js"));
function createCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.createCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCart = createCart;
function updateCart() {
    const payload = { cartInfo: { id: ':cartInfoId' } };
    const getRequestOptions = ambassadorWixEcomV1Cart.updateCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/carts/{cartInfo.id}',
        pathParams: { cartInfoId: 'cartInfoId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCart = updateCart;
function getCart() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.getCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/carts/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCart = getCart;
function addToCart() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.addToCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/add-to-cart',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addToCart = addToCart;
function removeLineItems() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.removeLineItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/remove-line-items',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeLineItems = removeLineItems;
function createCheckout() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.createCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/create-checkout',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCheckout = createCheckout;
function removeCoupon() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.removeCoupon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/remove-coupon',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeCoupon = removeCoupon;
function updateLineItemsQuantity() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.updateLineItemsQuantity(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/update-line-items-quantity',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateLineItemsQuantity = updateLineItemsQuantity;
function estimateTotals() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.estimateTotals(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/{id}/estimate-totals',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.estimateTotals = estimateTotals;
function deleteCart() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Cart.deleteCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/carts/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCart = deleteCart;
