// @wix/ecom_cart/build/cjs/src/ecom-v1-cart-cart.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeightUnit = exports.WebhookIdentityType = exports.VatType = exports.TaxableAddressType = exports.SuggestedFix = exports.SubscriptionFrequency = exports.Severity = exports.RuleType = exports.RateType = exports.PickupMethod = exports.PaymentOptionType = exports.NameInOther = exports.NameInLineItem = exports.ManualCalculationReason = exports.JurisdictionType = exports.ItemTypeItemType = exports.ItemAvailabilityStatus = exports.FileType = exports.FallbackReason = exports.DiscountType = exports.DescriptionLineType = exports.ChargeType = exports.ChannelType = exports.AppliedDiscountDiscountType = exports.AdditionalFeeSource = exports.onCartCreated = exports.onCartUpdated = exports.onCartDeleted = exports.deleteCart = exports.estimateTotals = exports.updateLineItemsQuantity = exports.removeCoupon = exports.createCheckout = exports.removeLineItems = exports.addToCart = exports.getCart = exports.updateCart = exports.createCart = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_cart_cart_universal_js_1 = require("./ecom-v1-cart-cart.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createCart(httpClient) {
    return (options) => (0, ecom_v1_cart_cart_universal_js_1.createCart)(options, 
    // @ts-ignore
    { httpClient });
}
exports.createCart = createCart;
function updateCart(httpClient) {
    return (_id, options) => (0, ecom_v1_cart_cart_universal_js_1.updateCart)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateCart = updateCart;
function getCart(httpClient) {
    return (_id) => (0, ecom_v1_cart_cart_universal_js_1.getCart)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getCart = getCart;
function addToCart(httpClient) {
    return (_id, options) => (0, ecom_v1_cart_cart_universal_js_1.addToCart)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.addToCart = addToCart;
function removeLineItems(httpClient) {
    return (_id, lineItemIds) => (0, ecom_v1_cart_cart_universal_js_1.removeLineItems)(_id, lineItemIds, 
    // @ts-ignore
    { httpClient });
}
exports.removeLineItems = removeLineItems;
function createCheckout(httpClient) {
    return (_id, options) => (0, ecom_v1_cart_cart_universal_js_1.createCheckout)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.createCheckout = createCheckout;
function removeCoupon(httpClient) {
    return (_id) => (0, ecom_v1_cart_cart_universal_js_1.removeCoupon)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.removeCoupon = removeCoupon;
function updateLineItemsQuantity(httpClient) {
    return (_id, lineItems) => (0, ecom_v1_cart_cart_universal_js_1.updateLineItemsQuantity)(_id, lineItems, 
    // @ts-ignore
    { httpClient });
}
exports.updateLineItemsQuantity = updateLineItemsQuantity;
function estimateTotals(httpClient) {
    return (_id, options) => (0, ecom_v1_cart_cart_universal_js_1.estimateTotals)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.estimateTotals = estimateTotals;
function deleteCart(httpClient) {
    return (_id) => (0, ecom_v1_cart_cart_universal_js_1.deleteCart)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCart = deleteCart;
exports.onCartDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.cart_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            {
                path: 'undefined.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'undefined.lineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'undefined.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'undefined.lineItems.image' },
            { path: 'undefined.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'undefined.lineItems.physicalProperties.weight' },
            {
                path: 'undefined.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'undefined.contactInfo.address' }],
    },
])))();
exports.onCartUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.cart_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            {
                path: 'entity.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.lineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.image' },
            { path: 'entity.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.lineItems.physicalProperties.weight' },
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'entity.contactInfo.address' }],
    },
])))();
exports.onCartCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.cart_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            {
                path: 'entity.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.lineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.image' },
            { path: 'entity.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.lineItems.physicalProperties.weight' },
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [{ path: 'entity.contactInfo.address' }],
    },
])))();
var ecom_v1_cart_cart_universal_js_2 = require("./ecom-v1-cart-cart.universal.js");
Object.defineProperty(exports, "AdditionalFeeSource", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.AdditionalFeeSource; } });
Object.defineProperty(exports, "AppliedDiscountDiscountType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.AppliedDiscountDiscountType; } });
Object.defineProperty(exports, "ChannelType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.ChannelType; } });
Object.defineProperty(exports, "ChargeType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.ChargeType; } });
Object.defineProperty(exports, "DescriptionLineType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.DescriptionLineType; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "FallbackReason", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.FallbackReason; } });
Object.defineProperty(exports, "FileType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.FileType; } });
Object.defineProperty(exports, "ItemAvailabilityStatus", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.ItemAvailabilityStatus; } });
Object.defineProperty(exports, "ItemTypeItemType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.ItemTypeItemType; } });
Object.defineProperty(exports, "JurisdictionType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.JurisdictionType; } });
Object.defineProperty(exports, "ManualCalculationReason", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.ManualCalculationReason; } });
Object.defineProperty(exports, "NameInLineItem", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.NameInLineItem; } });
Object.defineProperty(exports, "NameInOther", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.NameInOther; } });
Object.defineProperty(exports, "PaymentOptionType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.PaymentOptionType; } });
Object.defineProperty(exports, "PickupMethod", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.PickupMethod; } });
Object.defineProperty(exports, "RateType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.RateType; } });
Object.defineProperty(exports, "RuleType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.RuleType; } });
Object.defineProperty(exports, "Severity", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.Severity; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "SuggestedFix", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.SuggestedFix; } });
Object.defineProperty(exports, "TaxableAddressType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.TaxableAddressType; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.VatType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return ecom_v1_cart_cart_universal_js_2.WeightUnit; } });
