// @wix/ecom_delivery-profile/build/es/src/ecom-v1-delivery-profile-delivery-profile.public.js
import { addDeliveryCarrier as universalAddDeliveryCarrier, addDeliveryRegion as universalAddDeliveryRegion, createDeliveryProfile as universalCreateDeliveryProfile, deleteDeliveryProfile as universalDeleteDeliveryProfile, getDeliveryProfile as universalGetDeliveryProfile, listDeliveryCarrierDetails as universalListDeliveryCarrierDetails, listDeliveryCarriers as universalListDeliveryCarriers, listDeliveryDestinations as universalListDeliveryDestinations, listInstalledDeliveryCarriers as universalListInstalledDeliveryCarriers, queryDeliveryProfiles as universalQueryDeliveryProfiles, removeDeliveryCarrier as universalRemoveDeliveryCarrier, removeDeliveryRegion as universalRemoveDeliveryRegion, updateCarrierSettings as universalUpdateCarrierSettings, updateDeliveryCarrier as universalUpdateDeliveryCarrier, updateDeliveryProfile as universalUpdateDeliveryProfile, updateDeliveryRegion as universalUpdateDeliveryRegion, updateExtendedFields as universalUpdateExtendedFields, } from './ecom-v1-delivery-profile-delivery-profile.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function createDeliveryProfile(httpClient) {
    return (deliveryProfile) => universalCreateDeliveryProfile(deliveryProfile, 
    // @ts-ignore
    { httpClient });
}
export function getDeliveryProfile(httpClient) {
    return (deliveryProfileId) => universalGetDeliveryProfile(deliveryProfileId, 
    // @ts-ignore
    { httpClient });
}
export function updateDeliveryProfile(httpClient) {
    return (_id, deliveryProfile) => universalUpdateDeliveryProfile(_id, deliveryProfile, 
    // @ts-ignore
    { httpClient });
}
export function deleteDeliveryProfile(httpClient) {
    return (deliveryProfileId) => universalDeleteDeliveryProfile(deliveryProfileId, 
    // @ts-ignore
    { httpClient });
}
export function queryDeliveryProfiles(httpClient) {
    return () => universalQueryDeliveryProfiles(
    // @ts-ignore
    { httpClient });
}
export function addDeliveryRegion(httpClient) {
    return (deliveryProfileId, deliveryRegion, options) => universalAddDeliveryRegion(deliveryProfileId, deliveryRegion, options, 
    // @ts-ignore
    { httpClient });
}
export function updateDeliveryRegion(httpClient) {
    return (identifiers, deliveryRegion, options) => universalUpdateDeliveryRegion(identifiers, deliveryRegion, options, 
    // @ts-ignore
    { httpClient });
}
export function removeDeliveryRegion(httpClient) {
    return (identifiers, options) => universalRemoveDeliveryRegion(identifiers, options, 
    // @ts-ignore
    { httpClient });
}
export function addDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => universalAddDeliveryCarrier(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function removeDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => universalRemoveDeliveryCarrier(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => universalUpdateDeliveryCarrier(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function listDeliveryCarrierDetails(httpClient) {
    return () => universalListDeliveryCarrierDetails(
    // @ts-ignore
    { httpClient });
}
export function listInstalledDeliveryCarriers(httpClient) {
    return () => universalListInstalledDeliveryCarriers(
    // @ts-ignore
    { httpClient });
}
export function listDeliveryDestinations(httpClient) {
    return () => universalListDeliveryDestinations(
    // @ts-ignore
    { httpClient });
}
export function listDeliveryCarriers(httpClient) {
    return (deliveryProfileId, options) => universalListDeliveryCarriers(deliveryProfileId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateCarrierSettings(httpClient) {
    return (carrierId, options) => universalUpdateCarrierSettings(carrierId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => universalUpdateExtendedFields(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
export { ChargeType, DeleteStatus, DestinationScope, Namespace, SiteCreatedContext, SortOrder, State, WebhookIdentityType, } from './ecom-v1-delivery-profile-delivery-profile.universal.js';
