// @wix/ecom_delivery-profile/build/es/src/ecom-v1-delivery-profile-delivery-profile.meta.js
import * as ambassadorWixEcomV1DeliveryProfile from './ecom-v1-delivery-profile-delivery-profile.http.js';
export function createDeliveryProfile() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.createDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getDeliveryProfile() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.getDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/{deliveryProfileId}',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDeliveryProfile() {
    const payload = { deliveryProfile: { id: ':deliveryProfileId' } };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/delivery-profiles/{deliveryProfile.id}',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteDeliveryProfile() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.deleteDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/delivery-profiles/{deliveryProfileId}',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryDeliveryProfiles() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.queryDeliveryProfiles(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addDeliveryRegion() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.addDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDeliveryRegion() {
    const payload = {
        deliveryProfileId: ':deliveryProfileId',
        deliveryRegion: { id: ':deliveryRegionId' },
    };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegion.id}',
        pathParams: {
            deliveryProfileId: 'deliveryProfileId',
            deliveryRegionId: 'deliveryRegionId',
        },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeDeliveryRegion() {
    const payload = {
        deliveryProfileId: ':deliveryProfileId',
        deliveryRegionId: ':deliveryRegionId',
    };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.removeDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegionId}',
        pathParams: {
            deliveryProfileId: 'deliveryProfileId',
            deliveryRegionId: 'deliveryRegionId',
        },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addDeliveryCarrier() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.addDeliveryCarrier(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/add-delivery-carrier',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeDeliveryCarrier() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.removeDeliveryCarrier(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/remove-delivery-carrier',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDeliveryCarrier() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateDeliveryCarrier(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/delivery-profiles/update-delivery-carrier',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDeliveryCarrierDetails() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listDeliveryCarrierDetails(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-carriers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listInstalledDeliveryCarriers() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listInstalledDeliveryCarriers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/installed-carriers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDeliveryDestinations() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listDeliveryDestinations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/list-delivery-destinations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDeliveryCarriers() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listDeliveryCarriers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-carriers',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCarrierSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateCarrierSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-carriers/update-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
