// @wix/ecom_checkout-settings/build/cjs/src/ecom-v1-checkout-settings-checkout-settings.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateCheckoutSettings = exports.getCheckoutSettings = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const float_2 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomCheckoutSettingsApiV1CheckoutSettingsServiceUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/checkout-settings',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/checkout-settings',
                destPath: '/v1/checkout-settings',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/checkout-settings',
                destPath: '/v1/checkout-settings',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_checkout-settings';
/**
 * Retrieves the sites' checkout settings.
 *
 *
 * The `getCheckoutSettings()` function returns a Promise that resolves to checkout settings.
 */
function getCheckoutSettings(payload) {
    function __getCheckoutSettings({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.checkout_settings',
            method: 'GET',
            methodFqn: 'wix.ecom.checkout_settings.api.v1.CheckoutSettingsService.GetCheckoutSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomCheckoutSettingsApiV1CheckoutSettingsServiceUrl({
                protoPath: '/v1/checkout-settings',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'checkoutSettings.createdDate' },
                        { path: 'checkoutSettings.updatedDate' },
                        {
                            path: 'checkoutSettings.checkoutBrand.header.logo.image.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'checkoutSettings.checkoutBrand.header.logo.image.focalPoint.x',
                        },
                        {
                            path: 'checkoutSettings.checkoutBrand.header.logo.image.focalPoint.y',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getCheckoutSettings;
}
exports.getCheckoutSettings = getCheckoutSettings;
/**
 * Updates the sites' checkout settings.
 *
 *
 * The `updateCheckoutSettings()` function returns a Promise that resolves to the newly updated checkout settings.
 */
function updateCheckoutSettings(payload) {
    function __updateCheckoutSettings({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'checkoutSettings.createdDate' },
                    { path: 'checkoutSettings.updatedDate' },
                    {
                        path: 'checkoutSettings.checkoutBrand.header.logo.image.urlExpirationDate',
                    },
                ],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    {
                        path: 'checkoutSettings.checkoutBrand.header.logo.image.focalPoint.x',
                    },
                    {
                        path: 'checkoutSettings.checkoutBrand.header.logo.image.focalPoint.y',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.checkout_settings',
            method: 'PATCH',
            methodFqn: 'wix.ecom.checkout_settings.api.v1.CheckoutSettingsService.UpdateCheckoutSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomCheckoutSettingsApiV1CheckoutSettingsServiceUrl({
                protoPath: '/v1/checkout-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'checkoutSettings.createdDate' },
                        { path: 'checkoutSettings.updatedDate' },
                        {
                            path: 'checkoutSettings.checkoutBrand.header.logo.image.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'checkoutSettings.checkoutBrand.header.logo.image.focalPoint.x',
                        },
                        {
                            path: 'checkoutSettings.checkoutBrand.header.logo.image.focalPoint.y',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateCheckoutSettings;
}
exports.updateCheckoutSettings = updateCheckoutSettings;
