// @wix/ecom_checkout-settings/build/cjs/src/ecom-v1-checkout-settings-checkout-settings.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.LogoSize = exports.Alignment = exports.onCheckoutSettingsUpdated = exports.updateCheckoutSettings = exports.getCheckoutSettings = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_checkout_settings_checkout_settings_universal_js_1 = require("./ecom-v1-checkout-settings-checkout-settings.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function getCheckoutSettings(httpClient) {
    return () => (0, ecom_v1_checkout_settings_checkout_settings_universal_js_1.getCheckoutSettings)(
    // @ts-ignore
    { httpClient });
}
exports.getCheckoutSettings = getCheckoutSettings;
function updateCheckoutSettings(httpClient) {
    return (checkoutSettings) => (0, ecom_v1_checkout_settings_checkout_settings_universal_js_1.updateCheckoutSettings)(checkoutSettings, 
    // @ts-ignore
    { httpClient });
}
exports.updateCheckoutSettings = updateCheckoutSettings;
exports.onCheckoutSettingsUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_settings_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.checkoutBrand.header.logo.image' }],
    },
])))();
var ecom_v1_checkout_settings_checkout_settings_universal_js_2 = require("./ecom-v1-checkout-settings-checkout-settings.universal.js");
Object.defineProperty(exports, "Alignment", { enumerable: true, get: function () { return ecom_v1_checkout_settings_checkout_settings_universal_js_2.Alignment; } });
Object.defineProperty(exports, "LogoSize", { enumerable: true, get: function () { return ecom_v1_checkout_settings_checkout_settings_universal_js_2.LogoSize; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_checkout_settings_checkout_settings_universal_js_2.WebhookIdentityType; } });
