// @wix/ecom_memberships/build/es/src/service-plugins-types.js
export var MembershipErrorType;
(function (MembershipErrorType) {
    MembershipErrorType["UNKNOWN"] = "UNKNOWN";
    /** Membership not found. */
    MembershipErrorType["MEMBERSHIP_NOT_FOUND"] = "MEMBERSHIP_NOT_FOUND";
    /** The selection points to a line item ID that wasn't provided. */
    MembershipErrorType["LINE_ITEM_NOT_FOUND"] = "LINE_ITEM_NOT_FOUND";
    /** The membership cannot be used for the specific line items provided. */
    MembershipErrorType["MEMBERSHIP_CANNOT_BE_USED_FOR_LINE_ITEMS"] = "MEMBERSHIP_CANNOT_BE_USED_FOR_LINE_ITEMS";
    /** The membership can be used for each individual line item, but combining all of them is invalid. */
    MembershipErrorType["MEMBERSHIP_SELECTION_INVALID"] = "MEMBERSHIP_SELECTION_INVALID";
})(MembershipErrorType || (MembershipErrorType = {}));
export var MembershipCannotBeChargedType;
(function (MembershipCannotBeChargedType) {
    /** Membership is out of credits. For example, 5 are required, but only 4 remain. */
    MembershipCannotBeChargedType["OUT_OF_CREDITS"] = "OUT_OF_CREDITS";
    /** Membership has not become active yet. */
    MembershipCannotBeChargedType["NOT_STARTED_YET"] = "NOT_STARTED_YET";
    /** Membership has expired or ended. */
    MembershipCannotBeChargedType["ENDED"] = "ENDED";
    /** Not applicable for multiple participants. */
    MembershipCannotBeChargedType["NOT_APPLICABLE_FOR_MULTIPLE_PARTICIPANTS"] = "NOT_APPLICABLE_FOR_MULTIPLE_PARTICIPANTS";
})(MembershipCannotBeChargedType || (MembershipCannotBeChargedType = {}));
export var IdentityType;
(function (IdentityType) {
    IdentityType["UNKNOWN"] = "UNKNOWN";
    IdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    IdentityType["MEMBER"] = "MEMBER";
    IdentityType["WIX_USER"] = "WIX_USER";
    IdentityType["APP"] = "APP";
})(IdentityType || (IdentityType = {}));
