// @wix/ecom_local-delivery-options/build/cjs/src/ecom-v1-local-delivery-option-local-delivery-options.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.LogicalOperator = exports.LengthUnit = exports.DayOfWeek = exports.ConditionType = exports.removeDeliveryRegion = exports.addDeliveryRegion = exports.bulkDeleteLocalDeliveryOption = exports.bulkUpdateLocalDeliveryOption = exports.bulkCreateLocalDeliveryOption = exports.deleteLocalDeliveryOption = exports.updateLocalDeliveryOption = exports.queryLocalDeliveryOptions = exports.listLocalDeliveryOptions = exports.getLocalDeliveryOption = exports.createLocalDeliveryOption = exports.__metadata = void 0;
const ecom_v1_local_delivery_option_local_delivery_options_universal_js_1 = require("./ecom-v1-local-delivery-option-local-delivery-options.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createLocalDeliveryOption(httpClient) {
    return (localDeliveryOption) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.createLocalDeliveryOption)(localDeliveryOption, 
    // @ts-ignore
    { httpClient });
}
exports.createLocalDeliveryOption = createLocalDeliveryOption;
function getLocalDeliveryOption(httpClient) {
    return (localDeliveryOptionId, options) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.getLocalDeliveryOption)(localDeliveryOptionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getLocalDeliveryOption = getLocalDeliveryOption;
function listLocalDeliveryOptions(httpClient) {
    return (externalId) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.listLocalDeliveryOptions)(externalId, 
    // @ts-ignore
    { httpClient });
}
exports.listLocalDeliveryOptions = listLocalDeliveryOptions;
function queryLocalDeliveryOptions(httpClient) {
    return () => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.queryLocalDeliveryOptions)(
    // @ts-ignore
    { httpClient });
}
exports.queryLocalDeliveryOptions = queryLocalDeliveryOptions;
function updateLocalDeliveryOption(httpClient) {
    return (_id, localDeliveryOption) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.updateLocalDeliveryOption)(_id, localDeliveryOption, 
    // @ts-ignore
    { httpClient });
}
exports.updateLocalDeliveryOption = updateLocalDeliveryOption;
function deleteLocalDeliveryOption(httpClient) {
    return (localDeliveryOptionId, options) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.deleteLocalDeliveryOption)(localDeliveryOptionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.deleteLocalDeliveryOption = deleteLocalDeliveryOption;
function bulkCreateLocalDeliveryOption(httpClient) {
    return (localDeliveryOptions) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.bulkCreateLocalDeliveryOption)(localDeliveryOptions, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateLocalDeliveryOption = bulkCreateLocalDeliveryOption;
function bulkUpdateLocalDeliveryOption(httpClient) {
    return (localDeliveryOptions) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.bulkUpdateLocalDeliveryOption)(localDeliveryOptions, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateLocalDeliveryOption = bulkUpdateLocalDeliveryOption;
function bulkDeleteLocalDeliveryOption(httpClient) {
    return (ids) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.bulkDeleteLocalDeliveryOption)(ids, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteLocalDeliveryOption = bulkDeleteLocalDeliveryOption;
function addDeliveryRegion(httpClient) {
    return (deliveryRegionId, options) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.addDeliveryRegion)(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.addDeliveryRegion = addDeliveryRegion;
function removeDeliveryRegion(httpClient) {
    return (deliveryRegionId, options) => (0, ecom_v1_local_delivery_option_local_delivery_options_universal_js_1.removeDeliveryRegion)(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.removeDeliveryRegion = removeDeliveryRegion;
var ecom_v1_local_delivery_option_local_delivery_options_universal_js_2 = require("./ecom-v1-local-delivery-option-local-delivery-options.universal.js");
Object.defineProperty(exports, "ConditionType", { enumerable: true, get: function () { return ecom_v1_local_delivery_option_local_delivery_options_universal_js_2.ConditionType; } });
Object.defineProperty(exports, "DayOfWeek", { enumerable: true, get: function () { return ecom_v1_local_delivery_option_local_delivery_options_universal_js_2.DayOfWeek; } });
Object.defineProperty(exports, "LengthUnit", { enumerable: true, get: function () { return ecom_v1_local_delivery_option_local_delivery_options_universal_js_2.LengthUnit; } });
Object.defineProperty(exports, "LogicalOperator", { enumerable: true, get: function () { return ecom_v1_local_delivery_option_local_delivery_options_universal_js_2.LogicalOperator; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_local_delivery_option_local_delivery_options_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_local_delivery_option_local_delivery_options_universal_js_2.WebhookIdentityType; } });
