// @wix/ecom_local-delivery-options/build/cjs/src/ecom-v1-local-delivery-option-local-delivery-options.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeDeliveryRegion = exports.addDeliveryRegion = exports.bulkDeleteLocalDeliveryOption = exports.bulkUpdateLocalDeliveryOption = exports.bulkCreateLocalDeliveryOption = exports.deleteLocalDeliveryOption = exports.updateLocalDeliveryOption = exports.queryLocalDeliveryOptions = exports.listLocalDeliveryOptions = exports.getLocalDeliveryOption = exports.createLocalDeliveryOption = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const float_2 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/wix-ecommerce-shipments-local-delivery-web',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/stores/v1/localDeliveryOptions',
                destPath: '/v1/localDeliveryOptions',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/local-delivery-server',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/local-delivery-server',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_local-delivery-options';
function createLocalDeliveryOption(payload) {
    function __createLocalDeliveryOption({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'localDeliveryOption.minimumFreeDeliveryOrderAmount' },
                    { path: 'localDeliveryOption.radiusConfig.address.geocode.latitude' },
                    {
                        path: 'localDeliveryOption.radiusConfig.address.geocode.longitude',
                    },
                    { path: 'localDeliveryOption.customAreaConfig.geocodes.latitude' },
                    { path: 'localDeliveryOption.customAreaConfig.geocodes.longitude' },
                    { path: 'localDeliveryOption.radiusConfig.radius' },
                ],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'localDeliveryOption.createdDate' },
                    { path: 'localDeliveryOption.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'POST',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.CreateLocalDeliveryOption',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/local-delivery-options',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOption.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOption.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOption.createdDate' },
                        { path: 'localDeliveryOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createLocalDeliveryOption;
}
exports.createLocalDeliveryOption = createLocalDeliveryOption;
function getLocalDeliveryOption(payload) {
    function __getLocalDeliveryOption({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'GET',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.GetLocalDeliveryOption',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/local-delivery-options/{localDeliveryOptionId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOption.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOption.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOption.createdDate' },
                        { path: 'localDeliveryOption.updatedDate' },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'GET',
                    url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                        protoPath: '/v1/local-delivery-options/{localDeliveryOptionId}',
                        data: payload,
                        host,
                    }),
                    params: (0, rest_modules_1.toURLSearchParams)(payload),
                },
            ],
        };
        return metadata;
    }
    return __getLocalDeliveryOption;
}
exports.getLocalDeliveryOption = getLocalDeliveryOption;
/** @deprecated method is deprecated due to the fact that external_id is deprecated, it has been replaced with ListLocalDeliveryOptionsByDeliveryRegion(), and will be removed on 2024-11-01. */
function listLocalDeliveryOptions(payload) {
    function __listLocalDeliveryOptions({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'GET',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.ListLocalDeliveryOptions',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/localDeliveryOptions',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOptions.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOptions.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOptions.createdDate' },
                        { path: 'localDeliveryOptions.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listLocalDeliveryOptions;
}
exports.listLocalDeliveryOptions = listLocalDeliveryOptions;
/**
 * Retrieves a list of local delivery options, given the provided paging, filtering, and sorting.
 *
 * For field support for filters and sorting, see [Local Delivery: Supported Filters and Sorting](https://dev.wix.com/docs/rest/business-management/payments/tax/tax-regions/supported-filters-and-sorting).
 *
 * To learn about working with _Query_ endpoints, see
 * [API Query Language](https://dev.wix.com/api/rest/getting-started/api-query-language),
 * [Sorting and Paging](https://dev.wix.com/api/rest/getting-started/sorting-and-paging),
 * and [Field Projection](https://dev.wix.com/api/rest/getting-started/field-projection).
 */
function queryLocalDeliveryOptions(payload) {
    function __queryLocalDeliveryOptions({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'POST',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.QueryLocalDeliveryOptions',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/local-delivery-options/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOptions.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOptions.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOptions.createdDate' },
                        { path: 'localDeliveryOptions.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryLocalDeliveryOptions;
}
exports.queryLocalDeliveryOptions = queryLocalDeliveryOptions;
function updateLocalDeliveryOption(payload) {
    function __updateLocalDeliveryOption({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'localDeliveryOption.minimumFreeDeliveryOrderAmount' },
                    { path: 'localDeliveryOption.radiusConfig.address.geocode.latitude' },
                    {
                        path: 'localDeliveryOption.radiusConfig.address.geocode.longitude',
                    },
                    { path: 'localDeliveryOption.customAreaConfig.geocodes.latitude' },
                    { path: 'localDeliveryOption.customAreaConfig.geocodes.longitude' },
                    { path: 'localDeliveryOption.radiusConfig.radius' },
                ],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'localDeliveryOption.createdDate' },
                    { path: 'localDeliveryOption.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'PATCH',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.UpdateLocalDeliveryOption',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/local-delivery-options/{localDeliveryOption.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOption.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOption.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOption.createdDate' },
                        { path: 'localDeliveryOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateLocalDeliveryOption;
}
exports.updateLocalDeliveryOption = updateLocalDeliveryOption;
function deleteLocalDeliveryOption(payload) {
    function __deleteLocalDeliveryOption({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'DELETE',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.DeleteLocalDeliveryOption',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/local-delivery-options/{localDeliveryOptionId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteLocalDeliveryOption;
}
exports.deleteLocalDeliveryOption = deleteLocalDeliveryOption;
function bulkCreateLocalDeliveryOption(payload) {
    function __bulkCreateLocalDeliveryOption({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'localDeliveryOptions.minimumFreeDeliveryOrderAmount' },
                    {
                        path: 'localDeliveryOptions.radiusConfig.address.geocode.latitude',
                    },
                    {
                        path: 'localDeliveryOptions.radiusConfig.address.geocode.longitude',
                    },
                    { path: 'localDeliveryOptions.customAreaConfig.geocodes.latitude' },
                    { path: 'localDeliveryOptions.customAreaConfig.geocodes.longitude' },
                    { path: 'localDeliveryOptions.radiusConfig.radius' },
                ],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'localDeliveryOptions.createdDate' },
                    { path: 'localDeliveryOptions.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'POST',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkCreateLocalDeliveryOption',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/bulk/local-delivery-options/create',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOptions.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOptions.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOptions.createdDate' },
                        { path: 'localDeliveryOptions.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkCreateLocalDeliveryOption;
}
exports.bulkCreateLocalDeliveryOption = bulkCreateLocalDeliveryOption;
function bulkUpdateLocalDeliveryOption(payload) {
    function __bulkUpdateLocalDeliveryOption({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'localDeliveryOptions.minimumFreeDeliveryOrderAmount' },
                    {
                        path: 'localDeliveryOptions.radiusConfig.address.geocode.latitude',
                    },
                    {
                        path: 'localDeliveryOptions.radiusConfig.address.geocode.longitude',
                    },
                    { path: 'localDeliveryOptions.customAreaConfig.geocodes.latitude' },
                    { path: 'localDeliveryOptions.customAreaConfig.geocodes.longitude' },
                    { path: 'localDeliveryOptions.radiusConfig.radius' },
                ],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'localDeliveryOptions.createdDate' },
                    { path: 'localDeliveryOptions.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'PATCH',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkUpdateLocalDeliveryOption',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/bulk/local-delivery-options/update',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOptions.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOptions.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOptions.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOptions.createdDate' },
                        { path: 'localDeliveryOptions.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkUpdateLocalDeliveryOption;
}
exports.bulkUpdateLocalDeliveryOption = bulkUpdateLocalDeliveryOption;
function bulkDeleteLocalDeliveryOption(payload) {
    function __bulkDeleteLocalDeliveryOption({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'DELETE',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.BulkDeleteLocalDeliveryOption',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/bulk/local-delivery-options/delete',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __bulkDeleteLocalDeliveryOption;
}
exports.bulkDeleteLocalDeliveryOption = bulkDeleteLocalDeliveryOption;
function addDeliveryRegion(payload) {
    function __addDeliveryRegion({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'POST',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.AddDeliveryRegion',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/local-delivery-options/add-delivery-region',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOption.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOption.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOption.createdDate' },
                        { path: 'localDeliveryOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __addDeliveryRegion;
}
exports.addDeliveryRegion = addDeliveryRegion;
function removeDeliveryRegion(payload) {
    function __removeDeliveryRegion({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.local_delivery_option',
            method: 'POST',
            methodFqn: 'com.wix.ecom.shipments.localdelivery.LocalDelivery.RemoveDeliveryRegion',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomShipmentsLocaldeliveryLocalDeliveryUrl({
                protoPath: '/v1/local-delivery-options/remove-delivery-region',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'localDeliveryOption.minimumFreeDeliveryOrderAmount' },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.latitude',
                        },
                        {
                            path: 'localDeliveryOption.radiusConfig.address.geocode.longitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.latitude',
                        },
                        {
                            path: 'localDeliveryOption.customAreaConfig.geocodes.longitude',
                        },
                        { path: 'localDeliveryOption.radiusConfig.radius' },
                    ],
                },
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'localDeliveryOption.createdDate' },
                        { path: 'localDeliveryOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __removeDeliveryRegion;
}
exports.removeDeliveryRegion = removeDeliveryRegion;
