// @wix/ecom_local-delivery-options/build/cjs/src/ecom-v1-local-delivery-option-local-delivery-options.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeDeliveryRegion = exports.addDeliveryRegion = exports.bulkDeleteLocalDeliveryOption = exports.bulkUpdateLocalDeliveryOption = exports.bulkCreateLocalDeliveryOption = exports.deleteLocalDeliveryOption = exports.updateLocalDeliveryOption = exports.queryLocalDeliveryOptions = exports.listLocalDeliveryOptions = exports.getLocalDeliveryOption = exports.createLocalDeliveryOption = void 0;
const ambassadorWixEcomV1LocalDeliveryOption = __importStar(require("./ecom-v1-local-delivery-option-local-delivery-options.http.js"));
function createLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.createLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createLocalDeliveryOption = createLocalDeliveryOption;
function getLocalDeliveryOption() {
    const payload = { localDeliveryOptionId: ':localDeliveryOptionId' };
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.getLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/local-delivery-options/{localDeliveryOptionId}',
        pathParams: { localDeliveryOptionId: 'localDeliveryOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getLocalDeliveryOption = getLocalDeliveryOption;
function listLocalDeliveryOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.listLocalDeliveryOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/localDeliveryOptions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listLocalDeliveryOptions = listLocalDeliveryOptions;
function queryLocalDeliveryOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.queryLocalDeliveryOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryLocalDeliveryOptions = queryLocalDeliveryOptions;
function updateLocalDeliveryOption() {
    const payload = {
        localDeliveryOption: { id: ':localDeliveryOptionId' },
    };
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.updateLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/local-delivery-options/{localDeliveryOption.id}',
        pathParams: { localDeliveryOptionId: 'localDeliveryOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateLocalDeliveryOption = updateLocalDeliveryOption;
function deleteLocalDeliveryOption() {
    const payload = { localDeliveryOptionId: ':localDeliveryOptionId' };
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.deleteLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/local-delivery-options/{localDeliveryOptionId}',
        pathParams: { localDeliveryOptionId: 'localDeliveryOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteLocalDeliveryOption = deleteLocalDeliveryOption;
function bulkCreateLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.bulkCreateLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/local-delivery-options/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateLocalDeliveryOption = bulkCreateLocalDeliveryOption;
function bulkUpdateLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.bulkUpdateLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/bulk/local-delivery-options/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateLocalDeliveryOption = bulkUpdateLocalDeliveryOption;
function bulkDeleteLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.bulkDeleteLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/bulk/local-delivery-options/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteLocalDeliveryOption = bulkDeleteLocalDeliveryOption;
function addDeliveryRegion() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.addDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options/add-delivery-region',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addDeliveryRegion = addDeliveryRegion;
function removeDeliveryRegion() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.removeDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options/remove-delivery-region',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeDeliveryRegion = removeDeliveryRegion;
