// @wix/ecom_shipping-rates/build/es/src/interfaces-ecom-v1-shipping-rates-provider.public.js
import { ServicePluginDefinition } from '@wix/sdk-types';
import { transformSDKAddressToRESTAddress } from '@wix/sdk-runtime/transformations/address';
import { transformRESTAddressToSDKAddress } from '@wix/sdk-runtime/transformations/address';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
export const provideHandlers = ServicePluginDefinition('ECOM_SHIPPING_RATES', [
    {
        name: 'getShippingRates',
        primaryHttpMappingPath: '/v1/getRates',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = transformPaths(payload, [
                    {
                        transformFn: transformSDKAddressToRESTAddress,
                        paths: [{ path: 'shippingRates.logistics.pickupDetails.address' }],
                    },
                    {
                        transformFn: transformSDKTimestampToRESTTimestamp,
                        paths: [
                            { path: 'shippingRates.logistics.deliveryTimeSlot.from' },
                            { path: 'shippingRates.logistics.deliveryTimeSlot.to' },
                        ],
                    },
                ]);
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = transformPaths(payload, [
                    {
                        transformFn: transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.shippingDestination' },
                            { path: 'request.shippingOrigin' },
                        ],
                    },
                    {
                        transformFn: transformRESTFloatToSDKFloat,
                        paths: [{ path: 'request.lineItems.physicalProperties.weight' }],
                    },
                ]);
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
]);
