// @wix/ecom_abandoned-checkouts/build/es/src/ecom-v1-abandoned-checkout-abandoned-checkouts.meta.js
import * as ambassadorWixEcomV1AbandonedCheckout from './ecom-v1-abandoned-checkout-abandoned-checkouts.http.js';
export function getAbandonedCheckout() {
    const payload = { abandonedCheckoutId: ':abandonedCheckoutId' };
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.getAbandonedCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/abandoned-checkout/{abandonedCheckoutId}',
        pathParams: { abandonedCheckoutId: 'abandonedCheckoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteAbandonedCheckout() {
    const payload = { abandonedCheckoutId: ':abandonedCheckoutId' };
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.deleteAbandonedCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/abandoned-checkout/{abandonedCheckoutId}',
        pathParams: { abandonedCheckoutId: 'abandonedCheckoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryAbandonedCheckouts() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.queryAbandonedCheckouts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/abandoned-checkout/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchAbandonedCheckouts() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.searchAbandonedCheckouts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/abandoned-checkout/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function redirectToCheckout() {
    const payload = { abandonedCheckoutId: ':abandonedCheckoutId' };
    const getRequestOptions = ambassadorWixEcomV1AbandonedCheckout.redirectToCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout',
        pathParams: { abandonedCheckoutId: 'abandonedCheckoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
