// @wix/ecom_abandoned-checkouts/build/es/src/ecom-v1-abandoned-checkout-abandoned-checkouts.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformRESTBytesToSDKBytes } from '@wix/sdk-runtime/transformations/bytes';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/abandoned-checkout',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/abandoned-checkout',
                destPath: '/v1/abandoned-checkout',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_abandoned-checkouts';
/** Retrieves an abandoned checkout. */
export function getAbandonedCheckout(payload) {
    function __getAbandonedCheckout({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.abandoned_checkout',
            method: 'GET',
            methodFqn: 'wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.GetAbandonedCheckout',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
                protoPath: '/v1/abandoned-checkout/{abandonedCheckoutId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'abandonedCheckout.createdDate' },
                        { path: 'abandonedCheckout.updatedDate' },
                        { path: 'abandonedCheckout.checkoutRecoveredDate' },
                        { path: 'abandonedCheckout.activities.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getAbandonedCheckout;
}
/** Delete an AbandonedCheckout */
export function deleteAbandonedCheckout(payload) {
    function __deleteAbandonedCheckout({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.abandoned_checkout',
            method: 'DELETE',
            methodFqn: 'wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.DeleteAbandonedCheckout',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
                protoPath: '/v1/abandoned-checkout/{abandonedCheckoutId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __deleteAbandonedCheckout;
}
/**
 * Creates a query to retrieve a list of abandoned checkouts.
 *
 * The `queryAbandonedCheckouts()` function builds a query to retrieve a list of abandoned checkouts and returns a `ResultsQueryBuilder` object.
 *
 * The returned object contains the query definition, which is typically used to run the query using the `find()` function.
 *
 * You can refine the query by chaining `ResultsQueryBuilder` functions onto the query. `ResultsQueryBuilder` functions enable you to sort, filter, and control the results `queryAbandonedCheckouts()` returns.
 */
export function queryAbandonedCheckouts(payload) {
    function __queryAbandonedCheckouts({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.abandoned_checkout',
            method: 'POST',
            methodFqn: 'wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.QueryAbandonedCheckouts',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
                protoPath: '/v1/abandoned-checkout/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.createdDate' },
                        { path: 'results.updatedDate' },
                        { path: 'results.checkoutRecoveredDate' },
                        { path: 'results.activities.createdDate' },
                        { path: 'abandonedCheckouts.createdDate' },
                        { path: 'abandonedCheckouts.updatedDate' },
                        { path: 'abandonedCheckouts.checkoutRecoveredDate' },
                        { path: 'abandonedCheckouts.activities.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryAbandonedCheckouts;
}
/** Retrieves a list of abandoned checkouts, given the provided paging, filtering, searchDetails and sorting. Up to 100 abandoned checkouts can be returned per request. */
export function searchAbandonedCheckouts(payload) {
    function __searchAbandonedCheckouts({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.abandoned_checkout',
            method: 'POST',
            methodFqn: 'wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.SearchAbandonedCheckouts',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
                protoPath: '/v1/abandoned-checkout/search',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'abandonedCheckouts.createdDate' },
                        { path: 'abandonedCheckouts.updatedDate' },
                        { path: 'abandonedCheckouts.checkoutRecoveredDate' },
                        { path: 'abandonedCheckouts.activities.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __searchAbandonedCheckouts;
}
/** Redirects the abandoned checkout to the checkout page. */
export function redirectToCheckout(payload) {
    function __redirectToCheckout({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.abandoned_checkout',
            method: 'GET',
            methodFqn: 'wix.ecom.abandoned_checkout.api.v1.AbandonedCheckoutService.RedirectToCheckout',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomAbandonedCheckoutApiV1AbandonedCheckoutServiceUrl({
                protoPath: '/v1/abandoned-checkout/{abandonedCheckoutId}/redirect-to-checkout',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTBytesToSDKBytes,
                    paths: [{ path: 'body' }],
                },
            ]),
        };
        return metadata;
    }
    return __redirectToCheckout;
}
