// @wix/ecom_currencies/build/es/src/ecommerce-currency-converter-v1-currency-rate-currencies.meta.js
import * as ambassadorWixEcommerceCurrencyConverterV1CurrencyRate from './ecommerce-currency-converter-v1-currency-rate-currencies.http.js';
export function listCurrencies() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcommerceCurrencyConverterV1CurrencyRate.listCurrencies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/currencies',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function convertCurrency() {
    const payload = { from: ':from', to: ':to' };
    const getRequestOptions = ambassadorWixEcommerceCurrencyConverterV1CurrencyRate.convertCurrency(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/currencies/amounts/{from}/convert/{to}',
        pathParams: { from: 'from', to: 'to' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getConversionRate() {
    const payload = { from: ':from', to: ':to' };
    const getRequestOptions = ambassadorWixEcommerceCurrencyConverterV1CurrencyRate.conversionRate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/currencies/rate/{from}/convert/{to}',
        pathParams: { from: 'from', to: 'to' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
