// @wix/ecom_gift-vouchers-provider/build/es/src/interfaces-ecom-v1-gift-card-provider-entity.public.js
import { ServicePluginDefinition } from '@wix/sdk-types';
import { transformSDKFloatToRESTFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
export const provideHandlers = ServicePluginDefinition('GIFT_CARDS_PROVIDER', [
    {
        name: 'getBalance',
        primaryHttpMappingPath: '/v1/balance',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = transformPaths(payload, [
                    {
                        transformFn: transformSDKFloatToRESTFloat,
                        paths: [{ path: 'balance' }],
                    },
                ]);
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
    {
        name: 'redeem',
        primaryHttpMappingPath: '/v1/redeem',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = transformPaths(payload, [
                    {
                        transformFn: transformSDKFloatToRESTFloat,
                        paths: [{ path: 'remainingBalance' }],
                    },
                ]);
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = transformPaths(payload, [
                    {
                        transformFn: transformRESTFloatToSDKFloat,
                        paths: [{ path: 'request.amount' }],
                    },
                ]);
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
    {
        name: '_void',
        primaryHttpMappingPath: '/v1/void',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = transformPaths(payload, [
                    {
                        transformFn: transformSDKFloatToRESTFloat,
                        paths: [{ path: 'remainingBalance' }],
                    },
                ]);
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
]);
