// @wix/ecom_gift-vouchers-provider/build/es/src/service-plugins-error-classes.js
/**
 * Gift card not found
 */
export class GiftCardNotFoundWixError extends Error {
    constructor() {
        super('GiftCardNotFound');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'GIFT_CARD_NOT_FOUND';
        this.name = 'GiftCardNotFound';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
GiftCardNotFoundWixError.__type = 'wix_spi_error';
/**
 * Gift card is disabled
 */
export class GiftCardDisabledWixError extends Error {
    constructor() {
        super('GiftCardDisabled');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'GIFT_CARD_DISABLED';
        this.name = 'GiftCardDisabled';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
GiftCardDisabledWixError.__type = 'wix_spi_error';
/**
 * Gift card was expired
 */
export class GiftCardExpiredWixError extends Error {
    constructor() {
        super('GiftCardExpired');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'GIFT_CARD_EXPIRED';
        this.name = 'GiftCardExpired';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
GiftCardExpiredWixError.__type = 'wix_spi_error';
/**
 * Insufficient funds for redeem
 */
export class InsufficientFundsWixError extends Error {
    constructor() {
        super('InsufficientFunds');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'INSUFFICIENT_FUNDS';
        this.name = 'InsufficientFunds';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
InsufficientFundsWixError.__type = 'wix_spi_error';
/**
 * Gift card already redeemed for the given order
 */
export class AlreadyRedeemedWixError extends Error {
    constructor() {
        super('AlreadyRedeemed');
        this.httpCode = 409;
        this.statusCode = 'ALREADY_EXISTS';
        this.applicationCode = 'ALREADY_REDEEMED';
        this.name = 'AlreadyRedeemed';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
AlreadyRedeemedWixError.__type = 'wix_spi_error';
/**
 * Currency code not supported
 */
export class CurrencyNotSupportedWixError extends Error {
    constructor() {
        super('CurrencyNotSupported');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'CURRENCY_NOT_SUPPORTED';
        this.name = 'CurrencyNotSupported';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
CurrencyNotSupportedWixError.__type = 'wix_spi_error';
/**
 * Transaction not found.
 */
export class TransactionNotFoundWixError extends Error {
    constructor() {
        super('TransactionNotFound');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'TRANSACTION_NOT_FOUND';
        this.name = 'TransactionNotFound';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
TransactionNotFoundWixError.__type = 'wix_spi_error';
/**
 * Transaction already voided
 */
export class AlreadyVoidedWixError extends Error {
    constructor() {
        super('AlreadyVoided');
        this.httpCode = 409;
        this.statusCode = 'ALREADY_EXISTS';
        this.applicationCode = 'ALREADY_VOIDED';
        this.name = 'AlreadyVoided';
        this.errorType = 'APPLICATION';
    }
}
/** @hidden */
AlreadyVoidedWixError.__type = 'wix_spi_error';
