// @wix/ecom_orders/build/cjs/src/ecom-v1-order-orders.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefundableStatus = exports.RefundStatus = exports.Reason = exports.PreviewEmailType = exports.Placement = exports.PickupMethod = exports.PaymentStatus = exports.PaymentOptionType = exports.PaymentCollectabilityStatus = exports.OrderStatus = exports.OrderApprovalStrategy = exports.NonRefundableReason = exports.Namespace = exports.MembershipPaymentStatus = exports.ManuallyRefundableReason = exports.LineItemQuantityChangeType = exports.JurisdictionType = exports.ItemTypeItemType = exports.InvoiceStatus = exports.InventoryAction = exports.FulfillmentStatus = exports.DurationUnit = exports.DiscountType = exports.DiscountReason = exports.DescriptionLineType = exports.DepositType = exports.DeltaPaymentOptionType = exports.CustomFieldGroup = exports.ChannelType = exports.AuthorizationVoidStatus = exports.AuthorizationCaptureStatus = exports.AttributionSource = exports.ActivityType = exports.ActionType = exports.onOrderCreated = exports.onOrderCanceled = exports.onOrderUpdated = exports.onOrderApproved = exports.bulkUpdateOrderTags = exports.cancelOrder = exports.bulkUpdateOrders = exports.updateOrder = exports.createOrder = exports.searchOrders = exports.getOrder = exports.captureAuthorizedPayments = exports.voidAuthorizedPayments = exports.getPaymentCollectabilityStatus = exports.preparePaymentCollection = exports.__metadata = void 0;
exports.WeightUnit = exports.WebhookIdentityType = exports.VersioningMode = exports.VatType = exports.ValueType = exports.TransactionStatus = exports.TaxableAddressType = exports.SubscriptionFrequency = exports.SubdivisionType = exports.State = exports.SourceType = exports.SortOrder = exports.ScheduledAction = exports.RestockType = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_order_orders_universal_js_1 = require("./ecom-v1-order-orders.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function preparePaymentCollection(httpClient) {
    return (ecomOrderId, amount, options) => (0, ecom_v1_order_orders_universal_js_1.preparePaymentCollection)(ecomOrderId, amount, options, 
    // @ts-ignore
    { httpClient });
}
exports.preparePaymentCollection = preparePaymentCollection;
function getPaymentCollectabilityStatus(httpClient) {
    return (ecomOrderId) => (0, ecom_v1_order_orders_universal_js_1.getPaymentCollectabilityStatus)(ecomOrderId, 
    // @ts-ignore
    { httpClient });
}
exports.getPaymentCollectabilityStatus = getPaymentCollectabilityStatus;
function voidAuthorizedPayments(httpClient) {
    return (ecomOrderId, paymentIds) => (0, ecom_v1_order_orders_universal_js_1.voidAuthorizedPayments)(ecomOrderId, paymentIds, 
    // @ts-ignore
    { httpClient });
}
exports.voidAuthorizedPayments = voidAuthorizedPayments;
function captureAuthorizedPayments(httpClient) {
    return (ecomOrderId, payments) => (0, ecom_v1_order_orders_universal_js_1.captureAuthorizedPayments)(ecomOrderId, payments, 
    // @ts-ignore
    { httpClient });
}
exports.captureAuthorizedPayments = captureAuthorizedPayments;
function getOrder(httpClient) {
    return (_id) => (0, ecom_v1_order_orders_universal_js_1.getOrder)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getOrder = getOrder;
function searchOrders(httpClient) {
    return (options) => (0, ecom_v1_order_orders_universal_js_1.searchOrders)(options, 
    // @ts-ignore
    { httpClient });
}
exports.searchOrders = searchOrders;
function createOrder(httpClient) {
    return (order, options) => (0, ecom_v1_order_orders_universal_js_1.createOrder)(order, options, 
    // @ts-ignore
    { httpClient });
}
exports.createOrder = createOrder;
function updateOrder(httpClient) {
    return (_id, order) => (0, ecom_v1_order_orders_universal_js_1.updateOrder)(_id, order, 
    // @ts-ignore
    { httpClient });
}
exports.updateOrder = updateOrder;
function bulkUpdateOrders(httpClient) {
    return (orders, options) => (0, ecom_v1_order_orders_universal_js_1.bulkUpdateOrders)(orders, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateOrders = bulkUpdateOrders;
function cancelOrder(httpClient) {
    return (_id, options) => (0, ecom_v1_order_orders_universal_js_1.cancelOrder)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.cancelOrder = cancelOrder;
function bulkUpdateOrderTags(httpClient) {
    return (orderIds, options) => (0, ecom_v1_order_orders_universal_js_1.bulkUpdateOrderTags)(orderIds, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateOrderTags = bulkUpdateOrderTags;
exports.onOrderApproved = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.order_approved', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.purchasedDate' },
            { path: 'data.order.lineItems.digitalFile.expirationDate' },
            {
                path: 'data.order.lineItems.subscriptionInfo.subscriptionSettings.startDate',
            },
            { path: 'data.order.shippingInfo.logistics.deliverByDate' },
            { path: 'data.order.shippingInfo.logistics.deliveryTimeSlot.from' },
            { path: 'data.order.shippingInfo.logistics.deliveryTimeSlot.to' },
            { path: 'data.order.activities.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'data.order.lineItems.image' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'data.order.lineItems.physicalProperties.weight' }],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'data.order.billingInfo.address' },
            {
                path: 'data.order.shippingInfo.logistics.shippingDestination.address',
            },
            { path: 'data.order.shippingInfo.logistics.pickupDetails.address' },
            { path: 'data.order.recipientInfo.address' },
        ],
    },
])))();
exports.onOrderUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.order_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.purchasedDate' },
            { path: 'entity.lineItems.digitalFile.expirationDate' },
            {
                path: 'entity.lineItems.subscriptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.shippingInfo.logistics.deliverByDate' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.from' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.to' },
            { path: 'entity.activities.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.lineItems.image' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'entity.lineItems.physicalProperties.weight' }],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.billingInfo.address' },
            {
                path: 'entity.shippingInfo.logistics.shippingDestination.address',
            },
            { path: 'entity.shippingInfo.logistics.pickupDetails.address' },
            { path: 'entity.recipientInfo.address' },
        ],
    },
])))();
exports.onOrderCanceled = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.order_canceled', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.order.createdDate' },
            { path: 'data.order.updatedDate' },
            { path: 'data.order.purchasedDate' },
            { path: 'data.order.lineItems.digitalFile.expirationDate' },
            {
                path: 'data.order.lineItems.subscriptionInfo.subscriptionSettings.startDate',
            },
            { path: 'data.order.shippingInfo.logistics.deliverByDate' },
            { path: 'data.order.shippingInfo.logistics.deliveryTimeSlot.from' },
            { path: 'data.order.shippingInfo.logistics.deliveryTimeSlot.to' },
            { path: 'data.order.activities.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'data.order.lineItems.image' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'data.order.lineItems.physicalProperties.weight' }],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'data.order.billingInfo.address' },
            {
                path: 'data.order.shippingInfo.logistics.shippingDestination.address',
            },
            { path: 'data.order.shippingInfo.logistics.pickupDetails.address' },
            { path: 'data.order.recipientInfo.address' },
        ],
    },
])))();
exports.onOrderCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.order_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.purchasedDate' },
            { path: 'entity.lineItems.digitalFile.expirationDate' },
            {
                path: 'entity.lineItems.subscriptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.shippingInfo.logistics.deliverByDate' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.from' },
            { path: 'entity.shippingInfo.logistics.deliveryTimeSlot.to' },
            { path: 'entity.activities.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.lineItems.image' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'entity.lineItems.physicalProperties.weight' }],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.billingInfo.address' },
            {
                path: 'entity.shippingInfo.logistics.shippingDestination.address',
            },
            { path: 'entity.shippingInfo.logistics.pickupDetails.address' },
            { path: 'entity.recipientInfo.address' },
        ],
    },
])))();
var ecom_v1_order_orders_universal_js_2 = require("./ecom-v1-order-orders.universal.js");
Object.defineProperty(exports, "ActionType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.ActionType; } });
Object.defineProperty(exports, "ActivityType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.ActivityType; } });
Object.defineProperty(exports, "AttributionSource", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.AttributionSource; } });
Object.defineProperty(exports, "AuthorizationCaptureStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.AuthorizationCaptureStatus; } });
Object.defineProperty(exports, "AuthorizationVoidStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.AuthorizationVoidStatus; } });
Object.defineProperty(exports, "ChannelType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.ChannelType; } });
Object.defineProperty(exports, "CustomFieldGroup", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.CustomFieldGroup; } });
Object.defineProperty(exports, "DeltaPaymentOptionType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.DeltaPaymentOptionType; } });
Object.defineProperty(exports, "DepositType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.DepositType; } });
Object.defineProperty(exports, "DescriptionLineType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.DescriptionLineType; } });
Object.defineProperty(exports, "DiscountReason", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.DiscountReason; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "DurationUnit", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.DurationUnit; } });
Object.defineProperty(exports, "FulfillmentStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.FulfillmentStatus; } });
Object.defineProperty(exports, "InventoryAction", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.InventoryAction; } });
Object.defineProperty(exports, "InvoiceStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.InvoiceStatus; } });
Object.defineProperty(exports, "ItemTypeItemType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.ItemTypeItemType; } });
Object.defineProperty(exports, "JurisdictionType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.JurisdictionType; } });
Object.defineProperty(exports, "LineItemQuantityChangeType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.LineItemQuantityChangeType; } });
Object.defineProperty(exports, "ManuallyRefundableReason", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.ManuallyRefundableReason; } });
Object.defineProperty(exports, "MembershipPaymentStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.MembershipPaymentStatus; } });
Object.defineProperty(exports, "Namespace", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.Namespace; } });
Object.defineProperty(exports, "NonRefundableReason", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.NonRefundableReason; } });
Object.defineProperty(exports, "OrderApprovalStrategy", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.OrderApprovalStrategy; } });
Object.defineProperty(exports, "OrderStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.OrderStatus; } });
Object.defineProperty(exports, "PaymentCollectabilityStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.PaymentCollectabilityStatus; } });
Object.defineProperty(exports, "PaymentOptionType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.PaymentOptionType; } });
Object.defineProperty(exports, "PaymentStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.PaymentStatus; } });
Object.defineProperty(exports, "PickupMethod", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.PickupMethod; } });
Object.defineProperty(exports, "Placement", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.Placement; } });
Object.defineProperty(exports, "PreviewEmailType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.PreviewEmailType; } });
Object.defineProperty(exports, "Reason", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.Reason; } });
Object.defineProperty(exports, "RefundStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.RefundStatus; } });
Object.defineProperty(exports, "RefundableStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.RefundableStatus; } });
Object.defineProperty(exports, "RestockType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.RestockType; } });
Object.defineProperty(exports, "ScheduledAction", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.ScheduledAction; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "SourceType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.SourceType; } });
Object.defineProperty(exports, "State", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.State; } });
Object.defineProperty(exports, "SubdivisionType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.SubdivisionType; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "TaxableAddressType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.TaxableAddressType; } });
Object.defineProperty(exports, "TransactionStatus", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.TransactionStatus; } });
Object.defineProperty(exports, "ValueType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.ValueType; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.VatType; } });
Object.defineProperty(exports, "VersioningMode", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.VersioningMode; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return ecom_v1_order_orders_universal_js_2.WeightUnit; } });
