// @wix/ecom_orders/build/cjs/src/ecom-v1-order-orders.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkUpdateOrderTags = exports.cancelOrder = exports.bulkUpdateOrders = exports.updateOrder = exports.createOrder = exports.searchOrders = exports.getOrder = exports.captureAuthorizedPayments = exports.voidAuthorizedPayments = exports.getPaymentCollectabilityStatus = exports.preparePaymentCollection = void 0;
const ambassadorWixEcomV1Order = __importStar(require("./ecom-v1-order-orders.http.js"));
function preparePaymentCollection() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.preparePaymentCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments-collector/orders/{ecomOrderId}/prepare-payment-collection',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.preparePaymentCollection = preparePaymentCollection;
function getPaymentCollectabilityStatus() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.getPaymentCollectabilityStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/payments-collector/orders/{ecomOrderId}/payment-collectability-status',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getPaymentCollectabilityStatus = getPaymentCollectabilityStatus;
function voidAuthorizedPayments() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.voidAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments-collector/orders/{ecomOrderId}/void-authorized-payments',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.voidAuthorizedPayments = voidAuthorizedPayments;
function captureAuthorizedPayments() {
    const payload = { ecomOrderId: ':ecomOrderId' };
    const getRequestOptions = ambassadorWixEcomV1Order.captureAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/payments-collector/orders/{ecomOrderId}/capture-authorized-payments',
        pathParams: { ecomOrderId: 'ecomOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.captureAuthorizedPayments = captureAuthorizedPayments;
function getOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Order.getOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/orders/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrder = getOrder;
function searchOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.searchOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/orders/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchOrders = searchOrders;
function createOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.createOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/orders',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createOrder = createOrder;
function updateOrder() {
    const payload = { order: { id: ':orderId' } };
    const getRequestOptions = ambassadorWixEcomV1Order.updateOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/orders/{order.id}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateOrder = updateOrder;
function bulkUpdateOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.bulkUpdateOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/orders/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateOrders = bulkUpdateOrders;
function cancelOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Order.cancelOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/orders/{id}/cancel',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.cancelOrder = cancelOrder;
function bulkUpdateOrderTags() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Order.bulkUpdateOrderTags(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/orders/update-tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateOrderTags = bulkUpdateOrderTags;
