// @wix/ecom_checkout-templates/build/es/src/ecom-v1-checkout-template-checkout-templates.context.js
import { createCheckoutTemplate as publicCreateCheckoutTemplate, getCheckoutTemplate as publicGetCheckoutTemplate, updateCheckoutTemplate as publicUpdateCheckoutTemplate, deleteCheckoutTemplate as publicDeleteCheckoutTemplate, queryCheckoutTemplates as publicQueryCheckoutTemplates, createCheckoutFromTemplate as publicCreateCheckoutFromTemplate, createAndRedirectToCheckout as publicCreateAndRedirectToCheckout, } from './ecom-v1-checkout-template-checkout-templates.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onCheckoutTemplateCreated as publicOnCheckoutTemplateCreated } from './ecom-v1-checkout-template-checkout-templates.public.js';
import { onCheckoutTemplateDeleted as publicOnCheckoutTemplateDeleted } from './ecom-v1-checkout-template-checkout-templates.public.js';
import { onCheckoutTemplateUpdated as publicOnCheckoutTemplateUpdated } from './ecom-v1-checkout-template-checkout-templates.public.js';
import { onCheckoutTemplateUsed as publicOnCheckoutTemplateUsed } from './ecom-v1-checkout-template-checkout-templates.public.js';
export const createCheckoutTemplate = /*#__PURE__*/ createRESTModule(publicCreateCheckoutTemplate);
export const getCheckoutTemplate = /*#__PURE__*/ createRESTModule(publicGetCheckoutTemplate);
export const updateCheckoutTemplate = /*#__PURE__*/ createRESTModule(publicUpdateCheckoutTemplate);
export const deleteCheckoutTemplate = /*#__PURE__*/ createRESTModule(publicDeleteCheckoutTemplate);
export const queryCheckoutTemplates = /*#__PURE__*/ createRESTModule(publicQueryCheckoutTemplates);
export const createCheckoutFromTemplate = /*#__PURE__*/ createRESTModule(publicCreateCheckoutFromTemplate);
export const createAndRedirectToCheckout = /*#__PURE__*/ createRESTModule(publicCreateAndRedirectToCheckout);
/**
 * Triggered when a checkout template is created.
 */
export const onCheckoutTemplateCreated = createEventModule(publicOnCheckoutTemplateCreated);
/**
 * Triggered when a checkout template is deleted.
 */
export const onCheckoutTemplateDeleted = createEventModule(publicOnCheckoutTemplateDeleted);
/**
 * Triggered when a checkout template is updated.
 */
export const onCheckoutTemplateUpdated = createEventModule(publicOnCheckoutTemplateUpdated);
/**
 * Triggered when a checkout is created from a checkout template.
 */
export const onCheckoutTemplateUsed = createEventModule(publicOnCheckoutTemplateUsed);
export { Status, DescriptionLineType, PaymentOptionType, ItemTypeItemType, SubscriptionFrequency, FileType, SortOrder, JurisdictionType, ItemAvailabilityStatus, TaxableAddressType, VatType, PickupMethod, ChargeType, RuleType, DiscountType, WeightUnit, RateType, ManualCalculationReason, FallbackReason, ChannelType, AdditionalFeeSource, Severity, NameInOther, NameInLineItem, SuggestedFix, WebhookIdentityType, } from './ecom-v1-checkout-template-checkout-templates.universal.js';
