// @wix/ecom_checkout-templates/build/es/src/ecom-v1-checkout-template-checkout-templates.meta.js
import * as ambassadorWixEcomV1CheckoutTemplate from './ecom-v1-checkout-template-checkout-templates.http.js';
export function createCheckoutTemplate() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.createCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkout-templates',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCheckoutTemplate() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.getCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkout-templates/{checkoutTemplateId}',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCheckoutTemplate() {
    const payload = { checkoutTemplate: { id: ':checkoutTemplateId' } };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.updateCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/checkout-templates/{checkoutTemplate.id}',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCheckoutTemplate() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.deleteCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/checkout-templates/{checkoutTemplateId}',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryCheckoutTemplates() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.queryCheckoutTemplates(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkout-templates/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createCheckoutFromTemplate() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.createCheckoutFromTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkout-templates/{checkoutTemplateId}/create-checkout-from-template',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createAndRedirectToCheckout() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.createAndRedirectToCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkout-templates/{checkoutTemplateId}/create-and-redirect-to-checkout',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
