// @wix/ecom_checkout/build/es/src/ecom-v1-checkout-checkout.meta.js
import * as ambassadorWixEcomV1Checkout from './ecom-v1-checkout-checkout.http.js';
export function createCheckout() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Checkout.createCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCheckout() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.getCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkouts/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCheckoutByCartId() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.getCheckoutByCartId(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkouts/by-cart-id/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCheckoutUrl() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.getCheckoutUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkouts/{id}/checkout-url',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCheckout() {
    const payload = { checkout: { id: ':checkoutId' } };
    const getRequestOptions = ambassadorWixEcomV1Checkout.updateCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/checkouts/{checkout.id}',
        pathParams: { checkoutId: 'checkoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeCoupon() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeCoupon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-coupon',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeGiftCard() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeGiftCard(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-gift-card',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeOverrideCheckoutUrl() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeOverrideCheckoutUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-override-checkout-url',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addToCheckout() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.addToCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/add-to-checkout',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeLineItems() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeLineItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-line-items',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.createOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/create-order',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function markCheckoutAsCompleted() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.markCheckoutAsCompleted(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/mark-as-completed',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateLineItemsQuantity() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.updateLineItemsQuantity(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/update-line-items-quantity',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
