// @wix/ecom_checkout/build/es/src/ecom-v1-checkout-checkout.context.js
import { createCheckout as publicCreateCheckout, getCheckout as publicGetCheckout, getCheckoutByCartId as publicGetCheckoutByCartId, getCheckoutUrl as publicGetCheckoutUrl, updateCheckout as publicUpdateCheckout, removeCoupon as publicRemoveCoupon, removeGiftCard as publicRemoveGiftCard, removeOverrideCheckoutUrl as publicRemoveOverrideCheckoutUrl, addToCheckout as publicAddToCheckout, removeLineItems as publicRemoveLineItems, createOrder as publicCreateOrder, markCheckoutAsCompleted as publicMarkCheckoutAsCompleted, updateLineItemsQuantity as publicUpdateLineItemsQuantity, } from './ecom-v1-checkout-checkout.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onCheckoutCompleted as publicOnCheckoutCompleted } from './ecom-v1-checkout-checkout.public.js';
import { onCheckoutCreated as publicOnCheckoutCreated } from './ecom-v1-checkout-checkout.public.js';
import { onCheckoutUpdated as publicOnCheckoutUpdated } from './ecom-v1-checkout-checkout.public.js';
export const createCheckout = /*#__PURE__*/ createRESTModule(publicCreateCheckout);
export const getCheckout = /*#__PURE__*/ createRESTModule(publicGetCheckout);
export const getCheckoutByCartId = /*#__PURE__*/ createRESTModule(publicGetCheckoutByCartId);
export const getCheckoutUrl = /*#__PURE__*/ createRESTModule(publicGetCheckoutUrl);
export const updateCheckout = /*#__PURE__*/ createRESTModule(publicUpdateCheckout);
export const removeCoupon = /*#__PURE__*/ createRESTModule(publicRemoveCoupon);
export const removeGiftCard = /*#__PURE__*/ createRESTModule(publicRemoveGiftCard);
export const removeOverrideCheckoutUrl = /*#__PURE__*/ createRESTModule(publicRemoveOverrideCheckoutUrl);
export const addToCheckout = /*#__PURE__*/ createRESTModule(publicAddToCheckout);
export const removeLineItems = /*#__PURE__*/ createRESTModule(publicRemoveLineItems);
export const createOrder = /*#__PURE__*/ createRESTModule(publicCreateOrder);
export const markCheckoutAsCompleted = /*#__PURE__*/ createRESTModule(publicMarkCheckoutAsCompleted);
export const updateLineItemsQuantity = /*#__PURE__*/ createRESTModule(publicUpdateLineItemsQuantity);
/**
 * Triggered when an order created from this checkout is
 * successfully paid for or when a checkout is marked as completed.
 */
export const onCheckoutCompleted = createEventModule(publicOnCheckoutCompleted);
/**
 * Triggered when a checkout is created.
 */
export const onCheckoutCreated = createEventModule(publicOnCheckoutCreated);
/**
 * Triggered when a checkout is updated.
 */
export const onCheckoutUpdated = createEventModule(publicOnCheckoutUpdated);
export { JurisdictionType, DescriptionLineType, ItemAvailabilityStatus, ItemTypeItemType, SubscriptionFrequency, FileType, PaymentOptionType, TaxableAddressType, CommonVatType, PickupMethod, ChargeType, RuleType, DiscountType, WeightUnit, RateType, ManualCalculationReason, FallbackReason, ChannelType, AdditionalFeeSource, Severity, NameInOther, NameInLineItem, SuggestedFix, GetCheckoutPaymentSettingsResponsePaymentOption, IdentityType, LineItemType, MediaItemType, VatType, DiscountDiscountType, ChannelInfoChannelType, WebhookIdentityType, } from './ecom-v1-checkout-checkout.universal.js';
