// @wix/ecom_tip-settings/build/es/src/tips-settings-v1-tip-settings-tip-settings.public.js
import { createDefaultTipSettings as universalCreateDefaultTipSettings, createTipSettings as universalCreateTipSettings, deleteTipSettings as universalDeleteTipSettings, queryTipSettings as universalQueryTipSettings, updateTipSettings as universalUpdateTipSettings, } from './tips-settings-v1-tip-settings-tip-settings.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function queryTipSettings(httpClient) {
    return () => universalQueryTipSettings(
    // @ts-ignore
    { httpClient });
}
export function updateTipSettings(httpClient) {
    return (tipSettings) => universalUpdateTipSettings(tipSettings, 
    // @ts-ignore
    { httpClient });
}
export function createTipSettings(httpClient) {
    return (tipSettings) => universalCreateTipSettings(tipSettings, 
    // @ts-ignore
    { httpClient });
}
export function createDefaultTipSettings(httpClient) {
    return (tipSettings) => universalCreateDefaultTipSettings(tipSettings, 
    // @ts-ignore
    { httpClient });
}
export function deleteTipSettings(httpClient) {
    return (tipSettingsId) => universalDeleteTipSettings(tipSettingsId, 
    // @ts-ignore
    { httpClient });
}
export { SortOrder, StaffDistributionMethod, TipType, WebhookIdentityType, } from './tips-settings-v1-tip-settings-tip-settings.universal.js';
