// @wix/ecom_tip-settings/build/es/src/tips-settings-v1-tip-settings-tip-settings.meta.js
import * as ambassadorWixTipsSettingsV1TipSettings from './tips-settings-v1-tip-settings-tip-settings.http.js';
export function queryTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.queryTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tip-settings/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.updateTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tip-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.createTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tip-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createDefaultTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.createDefaultTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tip-settings/create-default',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteTipSettings() {
    const payload = { tipSettingsId: ':tipSettingsId' };
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.deleteTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tip-settings/{tipSettingsId}',
        pathParams: { tipSettingsId: 'tipSettingsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
