// @wix/ecom_additional-fees/build/cjs/src/interfaces-ecommerce-v1-additional-fees-provider.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('ECOM_ADDITIONAL_FEES', [
    {
        name: 'calculateAdditionalFees',
        primaryHttpMappingPath: '/v1/calculate-additional-fees',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: address_1.transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.shippingAddress' },
                            {
                                path: 'request.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address',
                            },
                        ],
                    },
                    {
                        transformFn: float_1.transformRESTFloatToSDKFloat,
                        paths: [{ path: 'request.lineItems.physicalProperties.weight' }],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
