// @wix/ecom_order-fulfillments/build/cjs/src/ecom-v1-fulfillments-order-fulfillments.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkCreateFulfillments = exports.deleteFulfillment = exports.updateFulfillment = exports.createFulfillment = exports.listFulfillmentsForMultipleOrders = exports.listFulfillmentsForSingleOrder = void 0;
const ambassadorWixEcomV1Fulfillments = __importStar(require("./ecom-v1-fulfillments-order-fulfillments.http.js"));
function listFulfillmentsForSingleOrder() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.listFulfillmentsForSingleOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/fulfillments/orders/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listFulfillmentsForSingleOrder = listFulfillmentsForSingleOrder;
function listFulfillmentsForMultipleOrders() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.listFulfillmentsForMultipleOrders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/fulfillments/list-by-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listFulfillmentsForMultipleOrders = listFulfillmentsForMultipleOrders;
function createFulfillment() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.createFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/fulfillments/orders/{orderId}/create-fulfillment',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createFulfillment = createFulfillment;
function updateFulfillment() {
    const payload = {
        fulfillment: { id: ':fulfillmentId' },
        orderId: ':orderId',
    };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.updateFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/fulfillments/{fulfillment.id}/orders/{orderId}',
        pathParams: { fulfillmentId: 'fulfillmentId', orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateFulfillment = updateFulfillment;
function deleteFulfillment() {
    const payload = {
        fulfillmentId: ':fulfillmentId',
        orderId: ':orderId',
    };
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.deleteFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/fulfillments/{fulfillmentId}/orders/{orderId}',
        pathParams: { fulfillmentId: 'fulfillmentId', orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteFulfillment = deleteFulfillment;
function bulkCreateFulfillments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Fulfillments.bulkCreateFulfillment(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/fulfillments/orders/bulk/create-fulfillments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateFulfillments = bulkCreateFulfillments;
