// @wix/ecom_order-fulfillments/build/cjs/src/ecom-v1-fulfillments-order-fulfillments.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.IdentityType = exports.FulfillmentStatus = exports.onFulfillmentsUpdated = exports.bulkCreateFulfillments = exports.deleteFulfillment = exports.updateFulfillment = exports.createFulfillment = exports.listFulfillmentsForMultipleOrders = exports.listFulfillmentsForSingleOrder = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_fulfillments_order_fulfillments_universal_js_1 = require("./ecom-v1-fulfillments-order-fulfillments.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function listFulfillmentsForSingleOrder(httpClient) {
    return (orderId) => (0, ecom_v1_fulfillments_order_fulfillments_universal_js_1.listFulfillmentsForSingleOrder)(orderId, 
    // @ts-ignore
    { httpClient });
}
exports.listFulfillmentsForSingleOrder = listFulfillmentsForSingleOrder;
function listFulfillmentsForMultipleOrders(httpClient) {
    return (orderIds) => (0, ecom_v1_fulfillments_order_fulfillments_universal_js_1.listFulfillmentsForMultipleOrders)(orderIds, 
    // @ts-ignore
    { httpClient });
}
exports.listFulfillmentsForMultipleOrders = listFulfillmentsForMultipleOrders;
function createFulfillment(httpClient) {
    return (orderId, fulfillment) => (0, ecom_v1_fulfillments_order_fulfillments_universal_js_1.createFulfillment)(orderId, fulfillment, 
    // @ts-ignore
    { httpClient });
}
exports.createFulfillment = createFulfillment;
function updateFulfillment(httpClient) {
    return (identifiers, options) => (0, ecom_v1_fulfillments_order_fulfillments_universal_js_1.updateFulfillment)(identifiers, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateFulfillment = updateFulfillment;
function deleteFulfillment(httpClient) {
    return (identifiers) => (0, ecom_v1_fulfillments_order_fulfillments_universal_js_1.deleteFulfillment)(identifiers, 
    // @ts-ignore
    { httpClient });
}
exports.deleteFulfillment = deleteFulfillment;
function bulkCreateFulfillments(httpClient) {
    return (ordersWithFulfillments) => (0, ecom_v1_fulfillments_order_fulfillments_universal_js_1.bulkCreateFulfillments)(ordersWithFulfillments, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateFulfillments = bulkCreateFulfillments;
exports.onFulfillmentsUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.fulfillments_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.fulfillments.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var ecom_v1_fulfillments_order_fulfillments_universal_js_2 = require("./ecom-v1-fulfillments-order-fulfillments.universal.js");
Object.defineProperty(exports, "FulfillmentStatus", { enumerable: true, get: function () { return ecom_v1_fulfillments_order_fulfillments_universal_js_2.FulfillmentStatus; } });
Object.defineProperty(exports, "IdentityType", { enumerable: true, get: function () { return ecom_v1_fulfillments_order_fulfillments_universal_js_2.IdentityType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_fulfillments_order_fulfillments_universal_js_2.WebhookIdentityType; } });
