// @wix/ecom_pickup-locations/build/cjs/src/ecom-v1-pickup-location-pickup-locations.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkDeletePickupLocation = exports.bulkUpdatePickupLocation = exports.bulkCreatePickupLocation = exports.removeDeliveryRegion = exports.addDeliveryRegion = exports.queryPickupLocation = exports.deletePickupLocation = exports.updatePickupLocation = exports.getPickupLocation = exports.createPickupLocation = void 0;
const ambassadorWixEcomV1PickupLocation = __importStar(require("./ecom-v1-pickup-location-pickup-locations.http.js"));
function createPickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.createPickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createPickupLocation = createPickupLocation;
function getPickupLocation() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.getPickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/pickup-locations/{pickupLocationId}',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getPickupLocation = getPickupLocation;
function updatePickupLocation() {
    const payload = { pickupLocation: { id: ':pickupLocationId' } };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.updatePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/pickup-locations/{pickupLocation.id}',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updatePickupLocation = updatePickupLocation;
function deletePickupLocation() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.deletePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/pickup-locations/{pickupLocationId}',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deletePickupLocation = deletePickupLocation;
function queryPickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.queryPickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryPickupLocation = queryPickupLocation;
function addDeliveryRegion() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.addDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations/{pickupLocationId}/add-delivery-region',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addDeliveryRegion = addDeliveryRegion;
function removeDeliveryRegion() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.removeDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations/{pickupLocationId}/remove-delivery-region',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeDeliveryRegion = removeDeliveryRegion;
function bulkCreatePickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.bulkCreatePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/pickup-locations/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreatePickupLocation = bulkCreatePickupLocation;
function bulkUpdatePickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.bulkUpdatePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/bulk/pickup-locations/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdatePickupLocation = bulkUpdatePickupLocation;
function bulkDeletePickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.bulkDeletePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/bulk/pickup-locations/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeletePickupLocation = bulkDeletePickupLocation;
