// @wix/ecom_pickup-locations/build/cjs/src/ecom-v1-pickup-location-pickup-locations.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.LogicalOperator = exports.ConditionType = exports.bulkDeletePickupLocation = exports.bulkUpdatePickupLocation = exports.bulkCreatePickupLocation = exports.removeDeliveryRegion = exports.addDeliveryRegion = exports.queryPickupLocation = exports.deletePickupLocation = exports.updatePickupLocation = exports.getPickupLocation = exports.createPickupLocation = exports.__metadata = void 0;
const ecom_v1_pickup_location_pickup_locations_universal_js_1 = require("./ecom-v1-pickup-location-pickup-locations.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createPickupLocation(httpClient) {
    return (pickupLocation) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.createPickupLocation)(pickupLocation, 
    // @ts-ignore
    { httpClient });
}
exports.createPickupLocation = createPickupLocation;
function getPickupLocation(httpClient) {
    return (pickupLocationId) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.getPickupLocation)(pickupLocationId, 
    // @ts-ignore
    { httpClient });
}
exports.getPickupLocation = getPickupLocation;
function updatePickupLocation(httpClient) {
    return (_id, pickupLocation) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.updatePickupLocation)(_id, pickupLocation, 
    // @ts-ignore
    { httpClient });
}
exports.updatePickupLocation = updatePickupLocation;
function deletePickupLocation(httpClient) {
    return (pickupLocationId) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.deletePickupLocation)(pickupLocationId, 
    // @ts-ignore
    { httpClient });
}
exports.deletePickupLocation = deletePickupLocation;
function queryPickupLocation(httpClient) {
    return () => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.queryPickupLocation)(
    // @ts-ignore
    { httpClient });
}
exports.queryPickupLocation = queryPickupLocation;
function addDeliveryRegion(httpClient) {
    return (pickupLocationId, deliveryRegionId, options) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.addDeliveryRegion)(pickupLocationId, deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.addDeliveryRegion = addDeliveryRegion;
function removeDeliveryRegion(httpClient) {
    return (pickupLocationId, deliveryRegionId, options) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.removeDeliveryRegion)(pickupLocationId, deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.removeDeliveryRegion = removeDeliveryRegion;
function bulkCreatePickupLocation(httpClient) {
    return (pickupLocations) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.bulkCreatePickupLocation)(pickupLocations, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreatePickupLocation = bulkCreatePickupLocation;
function bulkUpdatePickupLocation(httpClient) {
    return (pickupLocations) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.bulkUpdatePickupLocation)(pickupLocations, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdatePickupLocation = bulkUpdatePickupLocation;
function bulkDeletePickupLocation(httpClient) {
    return (pickupLocationIds) => (0, ecom_v1_pickup_location_pickup_locations_universal_js_1.bulkDeletePickupLocation)(pickupLocationIds, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeletePickupLocation = bulkDeletePickupLocation;
var ecom_v1_pickup_location_pickup_locations_universal_js_2 = require("./ecom-v1-pickup-location-pickup-locations.universal.js");
Object.defineProperty(exports, "ConditionType", { enumerable: true, get: function () { return ecom_v1_pickup_location_pickup_locations_universal_js_2.ConditionType; } });
Object.defineProperty(exports, "LogicalOperator", { enumerable: true, get: function () { return ecom_v1_pickup_location_pickup_locations_universal_js_2.LogicalOperator; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_pickup_location_pickup_locations_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_pickup_location_pickup_locations_universal_js_2.WebhookIdentityType; } });
