// @wix/ecom_order-invoices/build/cjs/src/ecom-orders-v1-invoice-order-invoices.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listInvoicesForMultipleOrders = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomOrdersInvoicesV1InvoicesUrl(opts) {
    const domainToMappings = {
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/ep-invoices',
                destPath: '/v1/invoices',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/ep-invoices',
                destPath: '/v1/invoices',
            },
        ],
        _: [
            {
                srcPath: '/v1/invoices',
                destPath: '/v1/invoices',
            },
        ],
    };
    return (0, rest_modules_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_order-invoices';
/**
 * Retrieves the IDs of invoices associated with all specified orders.
 *
 *
 * The `listInvoicesForSingleOrder()` function returns a Promise that resolves when the specified order's transaction records are retrieved.
 */
function listInvoicesForMultipleOrders(payload) {
    function __listInvoicesForMultipleOrders({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.orders.v1.invoice',
            method: 'POST',
            methodFqn: 'com.wix.ecom.orders.invoices.v1.Invoices.ListInvoicesForMultipleOrders',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomOrdersInvoicesV1InvoicesUrl({
                protoPath: '/v1/invoices/list-by-ids',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __listInvoicesForMultipleOrders;
}
exports.listInvoicesForMultipleOrders = listInvoicesForMultipleOrders;
