// @wix/ecom_shipping-options/build/cjs/src/ecom-v1-shipping-option-shipping-options.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateExtendedFields = exports.queryShippingOptions = exports.deleteShippingOption = exports.updateShippingOption = exports.getShippingOption = exports.createShippingOption = void 0;
const ambassadorWixEcomV1ShippingOption = __importStar(require("./ecom-v1-shipping-option-shipping-options.http.js"));
function createShippingOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.createShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shipping-options',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createShippingOption = createShippingOption;
function getShippingOption() {
    const payload = { shippingOptionId: ':shippingOptionId' };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.getShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shipping-options/{shippingOptionId}',
        pathParams: { shippingOptionId: 'shippingOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getShippingOption = getShippingOption;
function updateShippingOption() {
    const payload = { shippingOption: { id: ':shippingOptionId' } };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.updateShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/shipping-options/{shippingOption.id}',
        pathParams: { shippingOptionId: 'shippingOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateShippingOption = updateShippingOption;
function deleteShippingOption() {
    const payload = { shippingOptionId: ':shippingOptionId' };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.deleteShippingOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/shipping-options/{shippingOptionId}',
        pathParams: { shippingOptionId: 'shippingOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteShippingOption = deleteShippingOption;
function queryShippingOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.queryShippingOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shipping-options/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryShippingOptions = queryShippingOptions;
function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1ShippingOption.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shipping-options/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateExtendedFields = updateExtendedFields;
