// @wix/ecom_shipping-options/build/cjs/src/ecom-v1-shipping-option-shipping-options.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateExtendedFields = exports.queryShippingOptions = exports.deleteShippingOption = exports.updateShippingOption = exports.getShippingOption = exports.createShippingOption = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomDeliveryV1ShippingOptionServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/shipping-option-service',
                destPath: '',
            },
            {
                srcPath: '/_api/shipping-option-service',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/shipping-options',
                destPath: '/v1/shipping-options',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_shipping-options';
/**
 * Creates a ShippingOption.
 *
 * The request body must include shipping option and delivery region associated to.
 */
function createShippingOption(payload) {
    function __createShippingOption({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'shippingOption.createdDate' },
                    { path: 'shippingOption.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.shipping_option',
            method: 'POST',
            methodFqn: 'wix.ecom.delivery.v1.ShippingOptionService.CreateShippingOption',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryV1ShippingOptionServiceUrl({
                protoPath: '/v1/shipping-options',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'shippingOption.createdDate' },
                        { path: 'shippingOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createShippingOption;
}
exports.createShippingOption = createShippingOption;
/** Retrieves a ShippingOption. */
function getShippingOption(payload) {
    function __getShippingOption({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.shipping_option',
            method: 'GET',
            methodFqn: 'wix.ecom.delivery.v1.ShippingOptionService.GetShippingOption',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryV1ShippingOptionServiceUrl({
                protoPath: '/v1/shipping-options/{shippingOptionId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'shippingOption.createdDate' },
                        { path: 'shippingOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getShippingOption;
}
exports.getShippingOption = getShippingOption;
/**
 * Updates a ShippingOption.
 *
 *
 * Each time the ShippingOption is updated,
 * `revision` increments by 1.
 * The current `revision` must be passed when updating the ShippingOption.
 * This ensures you're working with the latest ShippingOption
 * and prevents unintended overwrites.
 */
function updateShippingOption(payload) {
    function __updateShippingOption({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'shippingOption.createdDate' },
                    { path: 'shippingOption.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.shipping_option',
            method: 'PATCH',
            methodFqn: 'wix.ecom.delivery.v1.ShippingOptionService.UpdateShippingOption',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryV1ShippingOptionServiceUrl({
                protoPath: '/v1/shipping-options/{shippingOption.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'shippingOption.createdDate' },
                        { path: 'shippingOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateShippingOption;
}
exports.updateShippingOption = updateShippingOption;
/**
 * Deletes a ShippingOption.
 * Deleting a ShippingOption permanently removes them from the ShippingOption List.
 */
function deleteShippingOption(payload) {
    function __deleteShippingOption({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.shipping_option',
            method: 'DELETE',
            methodFqn: 'wix.ecom.delivery.v1.ShippingOptionService.DeleteShippingOption',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryV1ShippingOptionServiceUrl({
                protoPath: '/v1/shipping-options/{shippingOptionId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteShippingOption;
}
exports.deleteShippingOption = deleteShippingOption;
/**
 * Retrieves a list of ShippingOptions, given the provided [paging, filtering, and sorting][1].
 *
 * Up to 1,000 ShippingOptions can be returned per request.
 *
 * To learn how to query ShippingOptions, see [API Query Language][2].
 *
 * [1]: https://dev.wix.com/api/rest/getting-started/sorting-and-paging
 * [2]: https://dev.wix.com/api/rest/getting-started/api-query-language
 */
function queryShippingOptions(payload) {
    function __queryShippingOptions({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.shipping_option',
            method: 'POST',
            methodFqn: 'wix.ecom.delivery.v1.ShippingOptionService.QueryShippingOptions',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryV1ShippingOptionServiceUrl({
                protoPath: '/v1/shipping-options/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'shippingOptions.createdDate' },
                        { path: 'shippingOptions.updatedDate' },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'POST',
                    url: resolveWixEcomDeliveryV1ShippingOptionServiceUrl({
                        protoPath: '/v1/shipping-options/query',
                        data: payload,
                        host,
                    }),
                    data: payload,
                },
            ],
        };
        return metadata;
    }
    return __queryShippingOptions;
}
exports.queryShippingOptions = queryShippingOptions;
/** Updates extended fields of a ShippingOption without incrementing revision */
function updateExtendedFields(payload) {
    function __updateExtendedFields({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.shipping_option',
            method: 'POST',
            methodFqn: 'wix.ecom.delivery.v1.ShippingOptionService.UpdateExtendedFields',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryV1ShippingOptionServiceUrl({
                protoPath: '/v1/shipping-options/{id}/update-extended-fields',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'shippingOption.createdDate' },
                        { path: 'shippingOption.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateExtendedFields;
}
exports.updateExtendedFields = updateExtendedFields;
