// @wix/ecom_shipping-options/build/cjs/src/ecom-v1-shipping-option-shipping-options.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.LogicalOperator = exports.ConditionType = exports.onShippingOptionUpdated = exports.onShippingOptionDeleted = exports.onShippingOptionCreated = exports.updateExtendedFields = exports.queryShippingOptions = exports.deleteShippingOption = exports.updateShippingOption = exports.getShippingOption = exports.createShippingOption = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_shipping_option_shipping_options_universal_js_1 = require("./ecom-v1-shipping-option-shipping-options.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createShippingOption(httpClient) {
    return (shippingOption) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.createShippingOption)(shippingOption, 
    // @ts-ignore
    { httpClient });
}
exports.createShippingOption = createShippingOption;
function getShippingOption(httpClient) {
    return (shippingOptionId) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.getShippingOption)(shippingOptionId, 
    // @ts-ignore
    { httpClient });
}
exports.getShippingOption = getShippingOption;
function updateShippingOption(httpClient) {
    return (_id, shippingOption) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.updateShippingOption)(_id, shippingOption, 
    // @ts-ignore
    { httpClient });
}
exports.updateShippingOption = updateShippingOption;
function deleteShippingOption(httpClient) {
    return (shippingOptionId) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.deleteShippingOption)(shippingOptionId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteShippingOption = deleteShippingOption;
function queryShippingOptions(httpClient) {
    return () => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.queryShippingOptions)(
    // @ts-ignore
    { httpClient });
}
exports.queryShippingOptions = queryShippingOptions;
function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.updateExtendedFields)(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateExtendedFields = updateExtendedFields;
exports.onShippingOptionCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.shipping_option_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onShippingOptionDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.shipping_option_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onShippingOptionUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.shipping_option_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var ecom_v1_shipping_option_shipping_options_universal_js_2 = require("./ecom-v1-shipping-option-shipping-options.universal.js");
Object.defineProperty(exports, "ConditionType", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.ConditionType; } });
Object.defineProperty(exports, "LogicalOperator", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.LogicalOperator; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.WebhookIdentityType; } });
