// @wix/identity_authentication/build/cjs/src/iam-authentication-v1-authentication-authentication.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.verify = exports.logout = exports.signOn = exports.loginWithIdpConnectionTokenParams = exports.loginCallback = exports.loginWithIdpConnection = exports.changePassword = exports.loginV2 = exports.registerV2 = void 0;
const ambassadorWixIamAuthenticationV1Authentication = __importStar(require("./iam-authentication-v1-authentication-authentication.http.js"));
function registerV2() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.registerV2(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/register',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.registerV2 = registerV2;
function loginV2() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginV2(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/login',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.loginV2 = loginV2;
function changePassword() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.changePassword(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/change-password',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.changePassword = changePassword;
function loginWithIdpConnection() {
    const payload = {
        tenantId: ':tenantId',
        idpConnectionId: ':idpConnectionId',
    };
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginWithIdpConnection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/sso/login/{tenantId}/{idpConnectionId}',
        pathParams: { tenantId: 'tenantId', idpConnectionId: 'idpConnectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.loginWithIdpConnection = loginWithIdpConnection;
function loginCallback() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginCallback(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/callback',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.loginCallback = loginCallback;
function loginWithIdpConnectionTokenParams() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.loginWithIdpConnectionTokenParams(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/sso/login-with-token-params',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.loginWithIdpConnectionTokenParams = loginWithIdpConnectionTokenParams;
function signOn() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.signOn(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/sign-on',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.signOn = signOn;
function logout() {
    const payload = {};
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.logout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/logout',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.logout = logout;
function verify() {
    const payload = { factorType: ':factorType' };
    const getRequestOptions = ambassadorWixIamAuthenticationV1Authentication.verify(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/{factorType}/verify',
        pathParams: { factorType: 'factorType' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.verify = verify;
