// @wix/identity_authentication/build/cjs/src/iam-authentication-v1-authentication-authentication.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantType = exports.StatusName = exports.Status = exports.StateType = exports.Reason = exports.PrivacyStatus = exports.PhoneTag = exports.FactorType = exports.EmailTag = exports.AddressTag = exports.verify = exports.logout = exports.signOn = exports.loginWithIdpConnectionTokenParams = exports.loginCallback = exports.loginWithIdpConnection = exports.changePassword = exports.loginV2 = exports.registerV2 = exports.__metadata = void 0;
const iam_authentication_v1_authentication_authentication_universal_js_1 = require("./iam-authentication-v1-authentication-authentication.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/identity' };
function registerV2(httpClient) {
    return (loginId, options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.registerV2)(loginId, options, 
    // @ts-ignore
    { httpClient });
}
exports.registerV2 = registerV2;
function loginV2(httpClient) {
    return (loginId, options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.loginV2)(loginId, options, 
    // @ts-ignore
    { httpClient });
}
exports.loginV2 = loginV2;
function changePassword(httpClient) {
    return (newPassword) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.changePassword)(newPassword, 
    // @ts-ignore
    { httpClient });
}
exports.changePassword = changePassword;
function loginWithIdpConnection(httpClient) {
    return (identifiers, tenantType, options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.loginWithIdpConnection)(identifiers, tenantType, options, 
    // @ts-ignore
    { httpClient });
}
exports.loginWithIdpConnection = loginWithIdpConnection;
function loginCallback(httpClient) {
    return (options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.loginCallback)(options, 
    // @ts-ignore
    { httpClient });
}
exports.loginCallback = loginCallback;
function loginWithIdpConnectionTokenParams(httpClient) {
    return (options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.loginWithIdpConnectionTokenParams)(options, 
    // @ts-ignore
    { httpClient });
}
exports.loginWithIdpConnectionTokenParams = loginWithIdpConnectionTokenParams;
function signOn(httpClient) {
    return (loginId, options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.signOn)(loginId, options, 
    // @ts-ignore
    { httpClient });
}
exports.signOn = signOn;
function logout(httpClient) {
    return (options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.logout)(options, 
    // @ts-ignore
    { httpClient });
}
exports.logout = logout;
function verify(httpClient) {
    return (factorType, options) => (0, iam_authentication_v1_authentication_authentication_universal_js_1.verify)(factorType, options, 
    // @ts-ignore
    { httpClient });
}
exports.verify = verify;
var iam_authentication_v1_authentication_authentication_universal_js_2 = require("./iam-authentication-v1-authentication-authentication.universal.js");
Object.defineProperty(exports, "AddressTag", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.AddressTag; } });
Object.defineProperty(exports, "EmailTag", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.EmailTag; } });
Object.defineProperty(exports, "FactorType", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.FactorType; } });
Object.defineProperty(exports, "PhoneTag", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.PhoneTag; } });
Object.defineProperty(exports, "PrivacyStatus", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.PrivacyStatus; } });
Object.defineProperty(exports, "Reason", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.Reason; } });
Object.defineProperty(exports, "StateType", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.StateType; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.Status; } });
Object.defineProperty(exports, "StatusName", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.StatusName; } });
Object.defineProperty(exports, "TenantType", { enumerable: true, get: function () { return iam_authentication_v1_authentication_authentication_universal_js_2.TenantType; } });
