// @wix/ecom_shippo-configurations/build/es/src/ecom-v1-shippo-configuration-shippo-configurations.meta.js
import * as ambassadorWixEcomV1ShippoConfiguration from './ecom-v1-shippo-configuration-shippo-configurations.http.js';
export function createShippoConfiguration() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.createShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shippo-configurations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getShippoConfiguration() {
    const payload = { shippoConfigurationId: ':shippoConfigurationId' };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.getShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shippo-configurations/{shippoConfigurationId}',
        pathParams: { shippoConfigurationId: 'shippoConfigurationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateShippoConfiguration() {
    const payload = {
        shippoConfiguration: { id: ':shippoConfigurationId' },
    };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.updateShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/shippo-configurations/{shippoConfiguration.id}',
        pathParams: { shippoConfigurationId: 'shippoConfigurationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteShippoConfiguration() {
    const payload = { shippoConfigurationId: ':shippoConfigurationId' };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.deleteShippoConfiguration(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/shippo-configurations/{shippoConfigurationId}',
        pathParams: { shippoConfigurationId: 'shippoConfigurationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryShippoConfigurations() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.queryShippoConfigurations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/shippo-configurations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1ShippoConfiguration.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/shippo-configurations/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
