// @wix/ecom_order-payment-requests/build/cjs/src/ecom-v1-order-payment-request-order-payment-requests.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateExtendedFields = exports.getOrderPaymentRequestUrl = exports.queryOrderPaymentRequests = exports.deleteOrderPaymentRequest = exports.updateOrderPaymentRequest = exports.getOrderPaymentRequest = exports.createOrderPaymentRequest = void 0;
const ambassadorWixEcomV1OrderPaymentRequest = __importStar(require("./ecom-v1-order-payment-request-order-payment-requests.http.js"));
function createOrderPaymentRequest() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.createOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createOrderPaymentRequest = createOrderPaymentRequest;
function getOrderPaymentRequest() {
    const payload = { orderPaymentRequestId: ':orderPaymentRequestId' };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.getOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/order-payment-requests/{orderPaymentRequestId}',
        pathParams: { orderPaymentRequestId: 'orderPaymentRequestId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrderPaymentRequest = getOrderPaymentRequest;
function updateOrderPaymentRequest() {
    const payload = {
        orderPaymentRequest: { id: ':orderPaymentRequestId' },
    };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.updateOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/order-payment-requests/{orderPaymentRequest.id}',
        pathParams: { orderPaymentRequestId: 'orderPaymentRequestId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateOrderPaymentRequest = updateOrderPaymentRequest;
function deleteOrderPaymentRequest() {
    const payload = { orderPaymentRequestId: ':orderPaymentRequestId' };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.deleteOrderPaymentRequest(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/order-payment-requests/{orderPaymentRequestId}',
        pathParams: { orderPaymentRequestId: 'orderPaymentRequestId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteOrderPaymentRequest = deleteOrderPaymentRequest;
function queryOrderPaymentRequests() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.queryOrderPaymentRequests(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryOrderPaymentRequests = queryOrderPaymentRequests;
function getOrderPaymentRequestUrl() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.getOrderPaymentRequestUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests/url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrderPaymentRequestUrl = getOrderPaymentRequestUrl;
function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1OrderPaymentRequest.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-payment-requests/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateExtendedFields = updateExtendedFields;
