// @wix/ecom_order-payment-requests/build/cjs/src/ecom-v1-order-payment-request-order-payment-requests.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Status = exports.SortOrder = exports.updateExtendedFields = exports.getOrderPaymentRequestUrl = exports.queryOrderPaymentRequests = exports.deleteOrderPaymentRequest = exports.updateOrderPaymentRequest = exports.getOrderPaymentRequest = exports.createOrderPaymentRequest = exports.__metadata = void 0;
const ecom_v1_order_payment_request_order_payment_requests_universal_js_1 = require("./ecom-v1-order-payment-request-order-payment-requests.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createOrderPaymentRequest(httpClient) {
    return (options) => (0, ecom_v1_order_payment_request_order_payment_requests_universal_js_1.createOrderPaymentRequest)(options, 
    // @ts-ignore
    { httpClient });
}
exports.createOrderPaymentRequest = createOrderPaymentRequest;
function getOrderPaymentRequest(httpClient) {
    return (orderPaymentRequestId) => (0, ecom_v1_order_payment_request_order_payment_requests_universal_js_1.getOrderPaymentRequest)(orderPaymentRequestId, 
    // @ts-ignore
    { httpClient });
}
exports.getOrderPaymentRequest = getOrderPaymentRequest;
function updateOrderPaymentRequest(httpClient) {
    return (_id, orderPaymentRequest) => (0, ecom_v1_order_payment_request_order_payment_requests_universal_js_1.updateOrderPaymentRequest)(_id, orderPaymentRequest, 
    // @ts-ignore
    { httpClient });
}
exports.updateOrderPaymentRequest = updateOrderPaymentRequest;
function deleteOrderPaymentRequest(httpClient) {
    return (orderPaymentRequestId) => (0, ecom_v1_order_payment_request_order_payment_requests_universal_js_1.deleteOrderPaymentRequest)(orderPaymentRequestId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteOrderPaymentRequest = deleteOrderPaymentRequest;
function queryOrderPaymentRequests(httpClient) {
    return () => (0, ecom_v1_order_payment_request_order_payment_requests_universal_js_1.queryOrderPaymentRequests)(
    // @ts-ignore
    { httpClient });
}
exports.queryOrderPaymentRequests = queryOrderPaymentRequests;
function getOrderPaymentRequestUrl(httpClient) {
    return (orderPaymentRequestId) => (0, ecom_v1_order_payment_request_order_payment_requests_universal_js_1.getOrderPaymentRequestUrl)(orderPaymentRequestId, 
    // @ts-ignore
    { httpClient });
}
exports.getOrderPaymentRequestUrl = getOrderPaymentRequestUrl;
function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => (0, ecom_v1_order_payment_request_order_payment_requests_universal_js_1.updateExtendedFields)(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateExtendedFields = updateExtendedFields;
var ecom_v1_order_payment_request_order_payment_requests_universal_js_2 = require("./ecom-v1-order-payment-request-order-payment-requests.universal.js");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_order_payment_request_order_payment_requests_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return ecom_v1_order_payment_request_order_payment_requests_universal_js_2.Status; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_order_payment_request_order_payment_requests_universal_js_2.WebhookIdentityType; } });
