// @wix/ecom_order-payment-requests/build/cjs/src/ecom-v1-order-payment-request-order-payment-requests.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateExtendedFields = exports.getOrderPaymentRequestUrl = exports.queryOrderPaymentRequests = exports.deleteOrderPaymentRequest = exports.updateOrderPaymentRequest = exports.getOrderPaymentRequest = exports.createOrderPaymentRequest = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/order-payment-request',
                destPath: '',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
        _: [
            {
                srcPath: '/ecom/v1/order-payment-requests',
                destPath: '/v1/order-payment-requests',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_order-payment-requests';
/** Creates a order payment request. */
function createOrderPaymentRequest(payload) {
    function __createOrderPaymentRequest({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'orderPaymentRequest.expirationDate' },
                    { path: 'orderPaymentRequest.createdDate' },
                    { path: 'orderPaymentRequest.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.CreateOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({ protoPath: '/v1/order-payment-requests', data: serializedData, host }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createOrderPaymentRequest;
}
exports.createOrderPaymentRequest = createOrderPaymentRequest;
/** Retrieves a order payment request. */
function getOrderPaymentRequest(payload) {
    function __getOrderPaymentRequest({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'GET',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.GetOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{orderPaymentRequestId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getOrderPaymentRequest;
}
exports.getOrderPaymentRequest = getOrderPaymentRequest;
/**
 * Updates a order payment request.
 *
 * Please note that only `UNPAID` payment requests can be updated.
 */
function updateOrderPaymentRequest(payload) {
    function __updateOrderPaymentRequest({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'orderPaymentRequest.expirationDate' },
                    { path: 'orderPaymentRequest.createdDate' },
                    { path: 'orderPaymentRequest.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'PATCH',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.UpdateOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{orderPaymentRequest.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateOrderPaymentRequest;
}
exports.updateOrderPaymentRequest = updateOrderPaymentRequest;
/**
 * Deletes a order payment request.
 *
 * Please note that only `UNPAID` payment requests can be deleted.
 */
function deleteOrderPaymentRequest(payload) {
    function __deleteOrderPaymentRequest({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'DELETE',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.DeleteOrderPaymentRequest',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{orderPaymentRequestId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteOrderPaymentRequest;
}
exports.deleteOrderPaymentRequest = deleteOrderPaymentRequest;
/**
 * Retrieves a list of Payment Requests, given the provided [paging, filtering, and sorting][1].
 *
 * Query Payment Requests runs with these defaults, which you can override:
 * - `createdDate` is sorted in DESC order
 * - `cursorPaging.limit` is 100
 *
 * To learn about working with _Query_ endpoints, see
 * [API Query Language](https://dev.wix.com/docs/rest/articles/getting-started/api-query-language), and
 * [Sorting and Paging](https://dev.wix.com/docs/rest/articles/getting-started/sorting-and-paging).
 */
function queryOrderPaymentRequests(payload) {
    function __queryOrderPaymentRequests({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.QueryOrderPaymentRequests',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({ protoPath: '/v1/order-payment-requests/query', data: payload, host }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequests.expirationDate' },
                        { path: 'orderPaymentRequests.createdDate' },
                        { path: 'orderPaymentRequests.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryOrderPaymentRequests;
}
exports.queryOrderPaymentRequests = queryOrderPaymentRequests;
/** Retrieves the order payment request page URL of a specified order payment request. */
function getOrderPaymentRequestUrl(payload) {
    function __getOrderPaymentRequestUrl({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.GetOrderPaymentRequestURL',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({ protoPath: '/v1/order-payment-requests/url', data: payload, host }),
            data: payload,
        };
        return metadata;
    }
    return __getOrderPaymentRequestUrl;
}
exports.getOrderPaymentRequestUrl = getOrderPaymentRequestUrl;
/** Updates extended fields of a order payment request */
function updateExtendedFields(payload) {
    function __updateExtendedFields({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.order_payment_request',
            method: 'POST',
            methodFqn: 'wix.ecom.order_payment_request.api.v1.OrderPaymentRequestsService.UpdateExtendedFields',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrderPaymentRequestApiV1OrderPaymentRequestsServiceUrl({
                protoPath: '/v1/order-payment-requests/{id}/update-extended-fields',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'orderPaymentRequest.expirationDate' },
                        { path: 'orderPaymentRequest.createdDate' },
                        { path: 'orderPaymentRequest.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateExtendedFields;
}
exports.updateExtendedFields = updateExtendedFields;
