// @wix/ecom_tip-settings/build/cjs/src/tips-settings-v1-tip-settings-tip-settings.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.TipType = exports.StaffDistributionMethod = exports.SortOrder = exports.deleteTipSettings = exports.createDefaultTipSettings = exports.createTipSettings = exports.updateTipSettings = exports.queryTipSettings = exports.__metadata = void 0;
const tips_settings_v1_tip_settings_tip_settings_universal_js_1 = require("./tips-settings-v1-tip-settings-tip-settings.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function queryTipSettings(httpClient) {
    return () => (0, tips_settings_v1_tip_settings_tip_settings_universal_js_1.queryTipSettings)(
    // @ts-ignore
    { httpClient });
}
exports.queryTipSettings = queryTipSettings;
function updateTipSettings(httpClient) {
    return (tipSettings) => (0, tips_settings_v1_tip_settings_tip_settings_universal_js_1.updateTipSettings)(tipSettings, 
    // @ts-ignore
    { httpClient });
}
exports.updateTipSettings = updateTipSettings;
function createTipSettings(httpClient) {
    return (tipSettings) => (0, tips_settings_v1_tip_settings_tip_settings_universal_js_1.createTipSettings)(tipSettings, 
    // @ts-ignore
    { httpClient });
}
exports.createTipSettings = createTipSettings;
function createDefaultTipSettings(httpClient) {
    return (tipSettings) => (0, tips_settings_v1_tip_settings_tip_settings_universal_js_1.createDefaultTipSettings)(tipSettings, 
    // @ts-ignore
    { httpClient });
}
exports.createDefaultTipSettings = createDefaultTipSettings;
function deleteTipSettings(httpClient) {
    return (tipSettingsId) => (0, tips_settings_v1_tip_settings_tip_settings_universal_js_1.deleteTipSettings)(tipSettingsId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteTipSettings = deleteTipSettings;
var tips_settings_v1_tip_settings_tip_settings_universal_js_2 = require("./tips-settings-v1-tip-settings-tip-settings.universal.js");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return tips_settings_v1_tip_settings_tip_settings_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "StaffDistributionMethod", { enumerable: true, get: function () { return tips_settings_v1_tip_settings_tip_settings_universal_js_2.StaffDistributionMethod; } });
Object.defineProperty(exports, "TipType", { enumerable: true, get: function () { return tips_settings_v1_tip_settings_tip_settings_universal_js_2.TipType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return tips_settings_v1_tip_settings_tip_settings_universal_js_2.WebhookIdentityType; } });
