// @wix/ecom_tip-settings/build/cjs/src/tips-settings-v1-tip-settings-tip-settings.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteTipSettings = exports.createDefaultTipSettings = exports.createTipSettings = exports.updateTipSettings = exports.queryTipSettings = void 0;
const ambassadorWixTipsSettingsV1TipSettings = __importStar(require("./tips-settings-v1-tip-settings-tip-settings.http.js"));
function queryTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.queryTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tip-settings/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryTipSettings = queryTipSettings;
function updateTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.updateTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tip-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateTipSettings = updateTipSettings;
function createTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.createTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tip-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createTipSettings = createTipSettings;
function createDefaultTipSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.createDefaultTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tip-settings/create-default',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDefaultTipSettings = createDefaultTipSettings;
function deleteTipSettings() {
    const payload = { tipSettingsId: ':tipSettingsId' };
    const getRequestOptions = ambassadorWixTipsSettingsV1TipSettings.deleteTipSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tip-settings/{tipSettingsId}',
        pathParams: { tipSettingsId: 'tipSettingsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteTipSettings = deleteTipSettings;
