// @wix/ecom_additional-fees/build/es/src/interfaces-ecommerce-v1-additional-fees-provider.public.js
import { ServicePluginDefinition } from '@wix/sdk-types';
import { transformRESTAddressToSDKAddress } from '@wix/sdk-runtime/transformations/address';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
export const provideHandlers = ServicePluginDefinition('ECOM_ADDITIONAL_FEES', [
    {
        name: 'calculateAdditionalFees',
        primaryHttpMappingPath: '/v1/calculate-additional-fees',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = transformPaths(payload, [
                    {
                        transformFn: transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.shippingAddress' },
                            {
                                path: 'request.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address',
                            },
                        ],
                    },
                    {
                        transformFn: transformRESTFloatToSDKFloat,
                        paths: [{ path: 'request.lineItems.physicalProperties.weight' }],
                    },
                ]);
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
]);
