// @wix/ecom_pickup-locations/build/es/src/ecom-v1-pickup-location-pickup-locations.public.js
import { addDeliveryRegion as universalAddDeliveryRegion, bulkCreatePickupLocation as universalBulkCreatePickupLocation, bulkDeletePickupLocation as universalBulkDeletePickupLocation, bulkUpdatePickupLocation as universalBulkUpdatePickupLocation, createPickupLocation as universalCreatePickupLocation, deletePickupLocation as universalDeletePickupLocation, getPickupLocation as universalGetPickupLocation, queryPickupLocation as universalQueryPickupLocation, removeDeliveryRegion as universalRemoveDeliveryRegion, updatePickupLocation as universalUpdatePickupLocation, } from './ecom-v1-pickup-location-pickup-locations.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function createPickupLocation(httpClient) {
    return (pickupLocation) => universalCreatePickupLocation(pickupLocation, 
    // @ts-ignore
    { httpClient });
}
export function getPickupLocation(httpClient) {
    return (pickupLocationId) => universalGetPickupLocation(pickupLocationId, 
    // @ts-ignore
    { httpClient });
}
export function updatePickupLocation(httpClient) {
    return (_id, pickupLocation) => universalUpdatePickupLocation(_id, pickupLocation, 
    // @ts-ignore
    { httpClient });
}
export function deletePickupLocation(httpClient) {
    return (pickupLocationId) => universalDeletePickupLocation(pickupLocationId, 
    // @ts-ignore
    { httpClient });
}
export function queryPickupLocation(httpClient) {
    return () => universalQueryPickupLocation(
    // @ts-ignore
    { httpClient });
}
export function addDeliveryRegion(httpClient) {
    return (pickupLocationId, deliveryRegionId, options) => universalAddDeliveryRegion(pickupLocationId, deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function removeDeliveryRegion(httpClient) {
    return (pickupLocationId, deliveryRegionId, options) => universalRemoveDeliveryRegion(pickupLocationId, deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkCreatePickupLocation(httpClient) {
    return (pickupLocations) => universalBulkCreatePickupLocation(pickupLocations, 
    // @ts-ignore
    { httpClient });
}
export function bulkUpdatePickupLocation(httpClient) {
    return (pickupLocations) => universalBulkUpdatePickupLocation(pickupLocations, 
    // @ts-ignore
    { httpClient });
}
export function bulkDeletePickupLocation(httpClient) {
    return (pickupLocationIds) => universalBulkDeletePickupLocation(pickupLocationIds, 
    // @ts-ignore
    { httpClient });
}
export { ConditionType, LogicalOperator, SortOrder, WebhookIdentityType, } from './ecom-v1-pickup-location-pickup-locations.universal.js';
