// @wix/ecom_pickup-locations/build/es/src/ecom-v1-pickup-location-pickup-locations.meta.js
import * as ambassadorWixEcomV1PickupLocation from './ecom-v1-pickup-location-pickup-locations.http.js';
export function createPickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.createPickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getPickupLocation() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.getPickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/pickup-locations/{pickupLocationId}',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updatePickupLocation() {
    const payload = { pickupLocation: { id: ':pickupLocationId' } };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.updatePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/pickup-locations/{pickupLocation.id}',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deletePickupLocation() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.deletePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/pickup-locations/{pickupLocationId}',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryPickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.queryPickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addDeliveryRegion() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.addDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations/{pickupLocationId}/add-delivery-region',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeDeliveryRegion() {
    const payload = { pickupLocationId: ':pickupLocationId' };
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.removeDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/pickup-locations/{pickupLocationId}/remove-delivery-region',
        pathParams: { pickupLocationId: 'pickupLocationId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreatePickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.bulkCreatePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/pickup-locations/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdatePickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.bulkUpdatePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/bulk/pickup-locations/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeletePickupLocation() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1PickupLocation.bulkDeletePickupLocation(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/bulk/pickup-locations/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
