// @wix/ecom/build/cjs/service-plugins.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.validations = exports.paymentSettings = exports.discounts = exports.customTriggers = exports.catalog = exports.additionalFees = exports.shippingRates = exports.recommendationsProvider = exports.memberships = exports.giftVouchersProvider = void 0;
exports.giftVouchersProvider = __importStar(require("@wix/ecom_gift-vouchers-provider"));
exports.memberships = __importStar(require("@wix/ecom_memberships"));
exports.recommendationsProvider = __importStar(require("@wix/ecom_recommendations-provider"));
exports.shippingRates = __importStar(require("@wix/ecom_shipping-rates"));
exports.additionalFees = __importStar(require("@wix/ecom_additional-fees"));
exports.catalog = __importStar(require("@wix/ecom_catalog"));
exports.customTriggers = __importStar(require("@wix/ecom_custom-triggers"));
exports.discounts = __importStar(require("@wix/ecom_discounts"));
exports.paymentSettings = __importStar(require("@wix/ecom_payment-settings"));
exports.validations = __importStar(require("@wix/ecom_validations"));
