// @wix/ecom_tips/build/cjs/src/tips-v1-tip-tips.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sumTipDistributionAmounts = exports.queryTipDistributions = exports.addTipDistributionsToOrder = exports.removeTipFromOrder = exports.setTipOnOrder = exports.updateTipChoice = exports.getTipChoice = exports.getTipByDraftOrderId = exports.getTipByOrderId = exports.previewTip = void 0;
const ambassadorWixTipsV1Tip = __importStar(require("./tips-v1-tip-tips.http.js"));
function previewTip() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.previewTip(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/preview',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.previewTip = previewTip;
function getTipByOrderId() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixTipsV1Tip.getTipByOrderId(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tips/order/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTipByOrderId = getTipByOrderId;
function getTipByDraftOrderId() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixTipsV1Tip.getTipByDraftOrderId(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tips/draft-order/{draftOrderId}',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTipByDraftOrderId = getTipByDraftOrderId;
function getTipChoice() {
    const payload = { purchaseFlowId: ':purchaseFlowId' };
    const getRequestOptions = ambassadorWixTipsV1Tip.getTipChoice(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tips/choice/{purchaseFlowId}',
        pathParams: { purchaseFlowId: 'purchaseFlowId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTipChoice = getTipChoice;
function updateTipChoice() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.updateTipChoice(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tips/choice',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateTipChoice = updateTipChoice;
function setTipOnOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.setTipOnOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/set',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setTipOnOrder = setTipOnOrder;
function removeTipFromOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.removeTipFromOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/remove',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeTipFromOrder = removeTipFromOrder;
function addTipDistributionsToOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.addTipDistributionsToOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/tip-distributions/add',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addTipDistributionsToOrder = addTipDistributionsToOrder;
function queryTipDistributions() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.queryTipDistributions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/tip-distributions/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryTipDistributions = queryTipDistributions;
function sumTipDistributionAmounts() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.sumTipDistributionAmounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/tip-distributions/sum-amounts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.sumTipDistributionAmounts = sumTipDistributionAmounts;
