// @wix/ecom_tips/build/cjs/src/tips-v1-tip-tips.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.TipType = exports.Source = exports.SortOrder = exports.QueryBy = exports.IdentityType = exports.sumTipDistributionAmounts = exports.queryTipDistributions = exports.addTipDistributionsToOrder = exports.removeTipFromOrder = exports.setTipOnOrder = exports.updateTipChoice = exports.getTipChoice = exports.getTipByDraftOrderId = exports.getTipByOrderId = exports.previewTip = exports.__metadata = void 0;
const tips_v1_tip_tips_universal_js_1 = require("./tips-v1-tip-tips.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function previewTip(httpClient) {
    return (lineItemInfo, options) => (0, tips_v1_tip_tips_universal_js_1.previewTip)(lineItemInfo, options, 
    // @ts-ignore
    { httpClient });
}
exports.previewTip = previewTip;
function getTipByOrderId(httpClient) {
    return (orderId) => (0, tips_v1_tip_tips_universal_js_1.getTipByOrderId)(orderId, 
    // @ts-ignore
    { httpClient });
}
exports.getTipByOrderId = getTipByOrderId;
function getTipByDraftOrderId(httpClient) {
    return (draftOrderId) => (0, tips_v1_tip_tips_universal_js_1.getTipByDraftOrderId)(draftOrderId, 
    // @ts-ignore
    { httpClient });
}
exports.getTipByDraftOrderId = getTipByDraftOrderId;
function getTipChoice(httpClient) {
    return (purchaseFlowId) => (0, tips_v1_tip_tips_universal_js_1.getTipChoice)(purchaseFlowId, 
    // @ts-ignore
    { httpClient });
}
exports.getTipChoice = getTipChoice;
function updateTipChoice(httpClient) {
    return (tipChoice, options) => (0, tips_v1_tip_tips_universal_js_1.updateTipChoice)(tipChoice, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateTipChoice = updateTipChoice;
function setTipOnOrder(httpClient) {
    return (options) => (0, tips_v1_tip_tips_universal_js_1.setTipOnOrder)(options, 
    // @ts-ignore
    { httpClient });
}
exports.setTipOnOrder = setTipOnOrder;
function removeTipFromOrder(httpClient) {
    return (options) => (0, tips_v1_tip_tips_universal_js_1.removeTipFromOrder)(options, 
    // @ts-ignore
    { httpClient });
}
exports.removeTipFromOrder = removeTipFromOrder;
function addTipDistributionsToOrder(httpClient) {
    return (options) => (0, tips_v1_tip_tips_universal_js_1.addTipDistributionsToOrder)(options, 
    // @ts-ignore
    { httpClient });
}
exports.addTipDistributionsToOrder = addTipDistributionsToOrder;
function queryTipDistributions(httpClient) {
    return () => (0, tips_v1_tip_tips_universal_js_1.queryTipDistributions)(
    // @ts-ignore
    { httpClient });
}
exports.queryTipDistributions = queryTipDistributions;
function sumTipDistributionAmounts(httpClient) {
    return (filter) => (0, tips_v1_tip_tips_universal_js_1.sumTipDistributionAmounts)(filter, 
    // @ts-ignore
    { httpClient });
}
exports.sumTipDistributionAmounts = sumTipDistributionAmounts;
var tips_v1_tip_tips_universal_js_2 = require("./tips-v1-tip-tips.universal.js");
Object.defineProperty(exports, "IdentityType", { enumerable: true, get: function () { return tips_v1_tip_tips_universal_js_2.IdentityType; } });
Object.defineProperty(exports, "QueryBy", { enumerable: true, get: function () { return tips_v1_tip_tips_universal_js_2.QueryBy; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return tips_v1_tip_tips_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Source", { enumerable: true, get: function () { return tips_v1_tip_tips_universal_js_2.Source; } });
Object.defineProperty(exports, "TipType", { enumerable: true, get: function () { return tips_v1_tip_tips_universal_js_2.TipType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return tips_v1_tip_tips_universal_js_2.WebhookIdentityType; } });
