// @wix/ecom_catalog/build/es/src/interfaces-ecommerce-v1-catalog-provider.public.js
import { ServicePluginDefinition } from '@wix/sdk-types';
import { transformSDKImageToRESTImage } from '@wix/sdk-runtime/transformations/image';
import { transformSDKPageURLV2ToRESTPageURLV2 } from '@wix/sdk-runtime/transformations/page-url-v2';
import { transformSDKFloatToRESTFloat } from '@wix/sdk-runtime/transformations/float';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
export const provideHandlers = ServicePluginDefinition('ECOM_CATALOG', [
    {
        name: 'getCatalogItems',
        primaryHttpMappingPath: '/get-catalog-items',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = transformPaths(payload, [
                    {
                        transformFn: transformSDKPageURLV2ToRESTPageURLV2,
                        paths: [{ path: 'catalogItems.data.url' }],
                    },
                    {
                        transformFn: transformSDKImageToRESTImage,
                        paths: [{ path: 'catalogItems.data.media' }],
                    },
                    {
                        transformFn: transformSDKFloatToRESTFloat,
                        paths: [{ path: 'catalogItems.data.physicalProperties.weight' }],
                    },
                    {
                        transformFn: transformSDKTimestampToRESTTimestamp,
                        paths: [
                            {
                                path: 'catalogItems.data.subscriptionOptionInfo.subscriptionSettings.startDate',
                            },
                            { path: 'catalogItems.data.serviceProperties.scheduledDate' },
                        ],
                    },
                ]);
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
]);
