// @wix/ecom_discount-rules/build/cjs/src/ecom-discounts-v1-discount-rule-discount-rules.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryDiscountRules = exports.deleteDiscountRule = exports.updateDiscountRule = exports.getDiscountRule = exports.createDiscountRule = void 0;
const ambassadorWixEcomDiscountsV1DiscountRule = __importStar(require("./ecom-discounts-v1-discount-rule-discount-rules.http.js"));
function createDiscountRule() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.createDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/discount-rules',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDiscountRule = createDiscountRule;
function getDiscountRule() {
    const payload = { discountRuleId: ':discountRuleId' };
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.getDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/discount-rules/{discountRuleId}',
        pathParams: { discountRuleId: 'discountRuleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getDiscountRule = getDiscountRule;
function updateDiscountRule() {
    const payload = { discountRule: { id: ':discountRuleId' } };
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.updateDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/discount-rules/{discountRule.id}',
        pathParams: { discountRuleId: 'discountRuleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDiscountRule = updateDiscountRule;
function deleteDiscountRule() {
    const payload = { discountRuleId: ':discountRuleId' };
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.deleteDiscountRule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/discount-rules/{discountRuleId}',
        pathParams: { discountRuleId: 'discountRuleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteDiscountRule = deleteDiscountRule;
function queryDiscountRules() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomDiscountsV1DiscountRule.queryDiscountRules(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/discount-rules/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryDiscountRules = queryDiscountRules;
