// @wix/ecom_discount-rules/build/cjs/src/ecom-discounts-v1-discount-rule-discount-rules.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Type = exports.TriggerType = exports.Status = exports.SortOrder = exports.ScopeType = exports.DiscountType = exports.AppliedSubjectType = exports.onDiscountRuleUpdated = exports.onDiscountRuleDeleted = exports.onDiscountRuleCreated = exports.queryDiscountRules = exports.deleteDiscountRule = exports.updateDiscountRule = exports.getDiscountRule = exports.createDiscountRule = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_discounts_v1_discount_rule_discount_rules_universal_js_1 = require("./ecom-discounts-v1-discount-rule-discount-rules.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createDiscountRule(httpClient) {
    return (discountRule) => (0, ecom_discounts_v1_discount_rule_discount_rules_universal_js_1.createDiscountRule)(discountRule, 
    // @ts-ignore
    { httpClient });
}
exports.createDiscountRule = createDiscountRule;
function getDiscountRule(httpClient) {
    return (discountRuleId) => (0, ecom_discounts_v1_discount_rule_discount_rules_universal_js_1.getDiscountRule)(discountRuleId, 
    // @ts-ignore
    { httpClient });
}
exports.getDiscountRule = getDiscountRule;
function updateDiscountRule(httpClient) {
    return (_id, discountRule) => (0, ecom_discounts_v1_discount_rule_discount_rules_universal_js_1.updateDiscountRule)(_id, discountRule, 
    // @ts-ignore
    { httpClient });
}
exports.updateDiscountRule = updateDiscountRule;
function deleteDiscountRule(httpClient) {
    return (discountRuleId) => (0, ecom_discounts_v1_discount_rule_discount_rules_universal_js_1.deleteDiscountRule)(discountRuleId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteDiscountRule = deleteDiscountRule;
function queryDiscountRules(httpClient) {
    return () => (0, ecom_discounts_v1_discount_rule_discount_rules_universal_js_1.queryDiscountRules)(
    // @ts-ignore
    { httpClient });
}
exports.queryDiscountRules = queryDiscountRules;
exports.onDiscountRuleCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.discounts.v1.discount_rule_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.activeTimeInfo.start' },
            { path: 'entity.activeTimeInfo.end' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'entity.discounts.values.percentage' }],
    },
])))();
exports.onDiscountRuleDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.discounts.v1.discount_rule_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'undefined.activeTimeInfo.start' },
            { path: 'undefined.activeTimeInfo.end' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'undefined.discounts.values.percentage' }],
    },
])))();
exports.onDiscountRuleUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.discounts.v1.discount_rule_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.activeTimeInfo.start' },
            { path: 'entity.activeTimeInfo.end' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [{ path: 'entity.discounts.values.percentage' }],
    },
])))();
var ecom_discounts_v1_discount_rule_discount_rules_universal_js_2 = require("./ecom-discounts-v1-discount-rule-discount-rules.universal.js");
Object.defineProperty(exports, "AppliedSubjectType", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.AppliedSubjectType; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "ScopeType", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.ScopeType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.Status; } });
Object.defineProperty(exports, "TriggerType", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.TriggerType; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.Type; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_discounts_v1_discount_rule_discount_rules_universal_js_2.WebhookIdentityType; } });
