// @wix/ecom_discount-rules/build/cjs/src/ecom-discounts-v1-discount-rule-discount-rules.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryDiscountRules = exports.deleteDiscountRule = exports.updateDiscountRule = exports.getDiscountRule = exports.createDiscountRule = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const float_2 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixEcomDiscountsDiscountRuleServiceUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/discount-rules-service',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/discount-rules',
                destPath: '/v1/discount-rules',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/discount-rules',
                destPath: '/v1/discount-rules',
            },
            {
                srcPath: '/v1/discount-rules',
                destPath: '/v1/discount-rules',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_discount-rules';
/**
 * Creates a new discount rule.
 *
 *
 * The `createDiscountRule()` function returns a Promise that resolves to the new discount rule when it's created.
 */
function createDiscountRule(payload) {
    function __createDiscountRule({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'discountRule.createdDate' },
                    { path: 'discountRule.updatedDate' },
                    { path: 'discountRule.activeTimeInfo.start' },
                    { path: 'discountRule.activeTimeInfo.end' },
                ],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [{ path: 'discountRule.discounts.values.percentage' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.discounts.v1.discount_rule',
            method: 'POST',
            methodFqn: 'com.wix.ecom.discounts.DiscountRuleService.CreateDiscountRule',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
                protoPath: '/v1/discount-rules',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'discountRule.createdDate' },
                        { path: 'discountRule.updatedDate' },
                        { path: 'discountRule.activeTimeInfo.start' },
                        { path: 'discountRule.activeTimeInfo.end' },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [{ path: 'discountRule.discounts.values.percentage' }],
                },
            ]),
        };
        return metadata;
    }
    return __createDiscountRule;
}
exports.createDiscountRule = createDiscountRule;
/**
 * Retrieves a discount rule.
 *
 *
 * The `getDiscountRule()` function returns a Promise that resolves when the specified discount rule is retrieved.
 */
function getDiscountRule(payload) {
    function __getDiscountRule({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.discounts.v1.discount_rule',
            method: 'GET',
            methodFqn: 'com.wix.ecom.discounts.DiscountRuleService.GetDiscountRule',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
                protoPath: '/v1/discount-rules/{discountRuleId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'discountRule.createdDate' },
                        { path: 'discountRule.updatedDate' },
                        { path: 'discountRule.activeTimeInfo.start' },
                        { path: 'discountRule.activeTimeInfo.end' },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [{ path: 'discountRule.discounts.values.percentage' }],
                },
            ]),
        };
        return metadata;
    }
    return __getDiscountRule;
}
exports.getDiscountRule = getDiscountRule;
/**
 * Updates a discount rule's properties.
 *
 *
 * The `updateDiscountRule()` function returns a Promise that resolves when the specified discount rule's properties are updated.
 *
 * Each time the discount rule is updated, `revision` increments by 1. The existing `revision` must be included when updating the discount rule. This ensures you're working with the latest discount rule information, and it prevents unintended overwrites.
 */
function updateDiscountRule(payload) {
    function __updateDiscountRule({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'mask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'discountRule.createdDate' },
                    { path: 'discountRule.updatedDate' },
                    { path: 'discountRule.activeTimeInfo.start' },
                    { path: 'discountRule.activeTimeInfo.end' },
                ],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [{ path: 'discountRule.discounts.values.percentage' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.discounts.v1.discount_rule',
            method: 'PATCH',
            methodFqn: 'com.wix.ecom.discounts.DiscountRuleService.UpdateDiscountRule',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
                protoPath: '/v1/discount-rules/{discountRule.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'discountRule.createdDate' },
                        { path: 'discountRule.updatedDate' },
                        { path: 'discountRule.activeTimeInfo.start' },
                        { path: 'discountRule.activeTimeInfo.end' },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [{ path: 'discountRule.discounts.values.percentage' }],
                },
            ]),
        };
        return metadata;
    }
    return __updateDiscountRule;
}
exports.updateDiscountRule = updateDiscountRule;
/**
 * Deletes a discount rule.
 *
 *
 * The `deleteDiscountRule()` function returns a Promise that resolves when the specified discount rule is deleted.
 */
function deleteDiscountRule(payload) {
    function __deleteDiscountRule({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.discounts.v1.discount_rule',
            method: 'DELETE',
            methodFqn: 'com.wix.ecom.discounts.DiscountRuleService.DeleteDiscountRule',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
                protoPath: '/v1/discount-rules/{discountRuleId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteDiscountRule;
}
exports.deleteDiscountRule = deleteDiscountRule;
/**
 * Creates a query to retrieve a list of discount rules.
 *
 *
 * The `queryDiscountRules()` function builds a query to retrieve a list of up to 100 discount rules, and returns a [`DiscountRulesQueryBuilder`](#discountrulesquerybuilder) object.
 *
 * The returned object contains the query definition which is typically used to run the query using the [`find()`](/discount-rules/discount-rules-query-builder/find) function.
 *
 * You can refine the query by chaining `DiscountRulesQueryBuilder` functions onto the query. `DiscountRulesQueryBuilder` functions enable you to sort, filter, and control the results queryDiscountRules() returns.
 *
 * By default, `queryDiscountRules()` sorts results by [`ascending("_id")`](/discount-rules/discount-rules-query-builder/ascending) by default. This can be overridden.
 *
 * To learn how to query posts, refer to the table below.
 *
 * The following `DiscountRulesQueryBuilder` functions are supported for the `queryDiscountRules()` function. For a full description of the discount rule object, see the object returned for the [`items`](/discount-rules/discount-rules-query-result/items) property in the `DiscountRulesQueryResult`.
 */
function queryDiscountRules(payload) {
    function __queryDiscountRules({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.discounts.v1.discount_rule',
            method: 'POST',
            methodFqn: 'com.wix.ecom.discounts.DiscountRuleService.QueryDiscountRules',
            packageName: PACKAGE_NAME,
            url: resolveComWixEcomDiscountsDiscountRuleServiceUrl({
                protoPath: '/v1/discount-rules/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'discountRules.createdDate' },
                        { path: 'discountRules.updatedDate' },
                        { path: 'discountRules.activeTimeInfo.start' },
                        { path: 'discountRules.activeTimeInfo.end' },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [{ path: 'discountRules.discounts.values.percentage' }],
                },
            ]),
        };
        return metadata;
    }
    return __queryDiscountRules;
}
exports.queryDiscountRules = queryDiscountRules;
