// @wix/ecom_recommendations-provider/build/es/src/service-plugins-error-classes.js
export class AlgorithmNotSupportedWixError extends Error {
    constructor(data = {}) {
        super('AlgorithmNotSupportedError');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'ALGORITHM_NOT_SUPPORTED';
        this.name = 'AlgorithmNotSupportedError';
        this.errorSchemaName =
            'com.wix.ecom.recommendations.spi.RecommendationAlgorithmNotSupported';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'AlgorithmNotSupportedError',
            applicationCode: 'ALGORITHM_NOT_SUPPORTED',
        };
        this.data = data;
    }
}
/** @hidden */
AlgorithmNotSupportedWixError.__type = 'wix_spi_error';
export class ItemAppIdNotSupportedWixError extends Error {
    constructor(data = {}) {
        super('ItemAppIdNotSupportedError');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'ITEM_APP_ID_NOT_SUPPORTED';
        this.name = 'ItemAppIdNotSupportedError';
        this.errorSchemaName =
            'com.wix.ecom.recommendations.spi.ItemAppIdNotSupportedError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'ItemAppIdNotSupportedError',
            applicationCode: 'ITEM_APP_ID_NOT_SUPPORTED',
        };
        this.data = data;
    }
}
/** @hidden */
ItemAppIdNotSupportedWixError.__type = 'wix_spi_error';
