// @wix/ecom_orders-settings/build/cjs/src/ecom-v1-orders-settings-orders-settings.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.InventoryUpdateTrigger = void 0;
var InventoryUpdateTrigger;
(function (InventoryUpdateTrigger) {
    InventoryUpdateTrigger["UNKNOWN_INVENTORY_UPDATE_TRIGGER"] = "UNKNOWN_INVENTORY_UPDATE_TRIGGER";
    /** Stock quantities will update only after the payment is approved. */
    InventoryUpdateTrigger["ON_ORDER_PAID"] = "ON_ORDER_PAID";
    /** Stock quantities will update while the payment is being processed. If the payment did not go through, items will restock. */
    InventoryUpdateTrigger["ON_ORDER_PLACED"] = "ON_ORDER_PLACED";
})(InventoryUpdateTrigger || (exports.InventoryUpdateTrigger = InventoryUpdateTrigger = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
