// @wix/ecom_orders-settings/build/cjs/src/ecom-v1-orders-settings-orders-settings.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateOrdersSettings = exports.getOrdersSettings = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/orders-settings',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/ecom/v1/orders-settings',
                destPath: '/v1/orders-settings',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/orders-settings',
                destPath: '/v1/orders-settings',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_orders-settings';
/**
 * Retrieves the sites' order settings.
 *
 *
 * The `getOrdersSettings()` function returns a Promise that resolves to orders settings.
 */
function getOrdersSettings(payload) {
    function __getOrdersSettings({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.orders_settings',
            method: 'GET',
            methodFqn: 'wix.ecom.orders_settings.api.v1.OrdersSettingsService.GetOrdersSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl({
                protoPath: '/v1/orders-settings',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'ordersSettings.createdDate' },
                        { path: 'ordersSettings.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getOrdersSettings;
}
exports.getOrdersSettings = getOrdersSettings;
/**
 * Updates the sites' orders settings.
 *
 *
 * The `updateOrdersSettings()` function returns a Promise that resolves to the newly updated orders settings.
 */
function updateOrdersSettings(payload) {
    function __updateOrdersSettings({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'ordersSettings.createdDate' },
                    { path: 'ordersSettings.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.orders_settings',
            method: 'PATCH',
            methodFqn: 'wix.ecom.orders_settings.api.v1.OrdersSettingsService.UpdateOrdersSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomOrdersSettingsApiV1OrdersSettingsServiceUrl({
                protoPath: '/v1/orders-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'ordersSettings.createdDate' },
                        { path: 'ordersSettings.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateOrdersSettings;
}
exports.updateOrdersSettings = updateOrdersSettings;
