// @wix/ecom_orders-settings/build/cjs/src/ecom-v1-orders-settings-orders-settings.universal.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateOrdersSettings = exports.getOrdersSettings = exports.WebhookIdentityType = exports.InventoryUpdateTrigger = void 0;
const transform_error_1 = require("@wix/sdk-runtime/transform-error");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const ambassadorWixEcomV1OrdersSettings = __importStar(require("./ecom-v1-orders-settings-orders-settings.http.js"));
var InventoryUpdateTrigger;
(function (InventoryUpdateTrigger) {
    InventoryUpdateTrigger["UNKNOWN_INVENTORY_UPDATE_TRIGGER"] = "UNKNOWN_INVENTORY_UPDATE_TRIGGER";
    /** Stock quantities will update only after the payment is approved. */
    InventoryUpdateTrigger["ON_ORDER_PAID"] = "ON_ORDER_PAID";
    /** Stock quantities will update while the payment is being processed. If the payment did not go through, items will restock. */
    InventoryUpdateTrigger["ON_ORDER_PLACED"] = "ON_ORDER_PLACED";
})(InventoryUpdateTrigger || (exports.InventoryUpdateTrigger = InventoryUpdateTrigger = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
/**
 * Retrieves the sites' order settings.
 *
 *
 * The `getOrdersSettings()` function returns a Promise that resolves to orders settings.
 * @public
 * @documentationMaturity preview
 * @permissionId ECOM.READ_ORDERS_SETTINGS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Read eCommerce - all read permissions
 * @permissionScopeId SCOPE.DC-ECOM-MEGA.READ-ECOM
 * @permissionScope Read Orders
 * @permissionScopeId SCOPE.DC-STORES.READ-ORDERS
 * @permissionScope Read Stores - all read permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.READ-STORES
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage eCommerce - all permissions
 * @permissionScopeId SCOPE.DC-ECOM-MEGA.MANAGE-ECOM
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @returns The requested orders settings.
 * @fqn wix.ecom.orders_settings.api.v1.OrdersSettingsService.GetOrdersSettings
 */
async function getOrdersSettings() {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[0];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({});
    const reqOpts = ambassadorWixEcomV1OrdersSettings.getOrdersSettings(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {},
            singleArgumentUnchanged: false,
        }, []);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.getOrdersSettings = getOrdersSettings;
/**
 * Updates the sites' orders settings.
 *
 *
 * The `updateOrdersSettings()` function returns a Promise that resolves to the newly updated orders settings.
 * @public
 * @documentationMaturity preview
 * @requiredField ordersSettings
 * @param ordersSettings - Orders settings to update.
 * @permissionId ECOM.UPDATE_ORDERS_SETTINGS
 * @permissionScope Manage Stores - all permissions
 * @permissionScopeId SCOPE.DC-STORES-MEGA.MANAGE-STORES
 * @permissionScope Manage Restaurants - all permissions
 * @permissionScopeId SCOPE.RESTAURANTS.MEGA-SCOPES
 * @permissionScope Manage eCommerce - all permissions
 * @permissionScopeId SCOPE.DC-ECOM-MEGA.MANAGE-ECOM
 * @permissionScope Manage Orders
 * @permissionScopeId SCOPE.DC-STORES.MANAGE-ORDERS
 * @applicableIdentity APP
 * @returns The updated orders settings.
 * @fqn wix.ecom.orders_settings.api.v1.OrdersSettingsService.UpdateOrdersSettings
 */
async function updateOrdersSettings(ordersSettings) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        ordersSettings: ordersSettings,
    });
    const reqOpts = ambassadorWixEcomV1OrdersSettings.updateOrdersSettings(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { ordersSettings: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['ordersSettings']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.updateOrdersSettings = updateOrdersSettings;
