// @wix/ecom_orders-settings/build/cjs/src/ecom-v1-orders-settings-orders-settings.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.InventoryUpdateTrigger = exports.onOrdersSettingsUpdated = exports.updateOrdersSettings = exports.getOrdersSettings = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_orders_settings_orders_settings_universal_js_1 = require("./ecom-v1-orders-settings-orders-settings.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function getOrdersSettings(httpClient) {
    return () => (0, ecom_v1_orders_settings_orders_settings_universal_js_1.getOrdersSettings)(
    // @ts-ignore
    { httpClient });
}
exports.getOrdersSettings = getOrdersSettings;
function updateOrdersSettings(httpClient) {
    return (ordersSettings) => (0, ecom_v1_orders_settings_orders_settings_universal_js_1.updateOrdersSettings)(ordersSettings, 
    // @ts-ignore
    { httpClient });
}
exports.updateOrdersSettings = updateOrdersSettings;
exports.onOrdersSettingsUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.orders_settings_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var ecom_v1_orders_settings_orders_settings_universal_js_2 = require("./ecom-v1-orders-settings-orders-settings.universal.js");
Object.defineProperty(exports, "InventoryUpdateTrigger", { enumerable: true, get: function () { return ecom_v1_orders_settings_orders_settings_universal_js_2.InventoryUpdateTrigger; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_orders_settings_orders_settings_universal_js_2.WebhookIdentityType; } });
