// @wix/ecom_payment-settings/build/cjs/src/interfaces-ecommerce-v1-payment-settings-provider.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('ECOM_PAYMENT_SETTINGS', [
    {
        name: 'getPaymentSettings',
        primaryHttpMappingPath: '/v1/payment-settings',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                        paths: [
                            { path: 'request.order.createdDate' },
                            { path: 'request.order.updatedDate' },
                            { path: 'request.order.purchasedDate' },
                            { path: 'request.order.lineItems.digitalFile.expirationDate' },
                            {
                                path: 'request.order.lineItems.subscriptionInfo.subscriptionSettings.startDate',
                            },
                            { path: 'request.order.shippingInfo.logistics.deliverByDate' },
                            {
                                path: 'request.order.shippingInfo.logistics.deliveryTimeSlot.from',
                            },
                            {
                                path: 'request.order.shippingInfo.logistics.deliveryTimeSlot.to',
                            },
                            { path: 'request.order.activities.createdDate' },
                        ],
                    },
                    {
                        transformFn: image_1.transformRESTImageToSDKImage,
                        paths: [{ path: 'request.order.lineItems.image' }],
                    },
                    {
                        transformFn: float_1.transformRESTFloatToSDKFloat,
                        paths: [
                            { path: 'request.order.lineItems.physicalProperties.weight' },
                        ],
                    },
                    {
                        transformFn: address_1.transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.order.billingInfo.address' },
                            {
                                path: 'request.order.shippingInfo.logistics.shippingDestination.address',
                            },
                            {
                                path: 'request.order.shippingInfo.logistics.pickupDetails.address',
                            },
                            { path: 'request.order.recipientInfo.address' },
                        ],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'getPaymentSettingsForCheckout',
        primaryHttpMappingPath: '/v1/payment-settings-for-checkout',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                        paths: [
                            { path: 'request.checkout.createdDate' },
                            { path: 'request.checkout.updatedDate' },
                            {
                                path: 'request.checkout.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
                            },
                            {
                                path: 'request.checkout.lineItems.serviceProperties.scheduledDate',
                            },
                            {
                                path: 'request.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.from',
                            },
                            {
                                path: 'request.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.to',
                            },
                            {
                                path: 'request.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.from',
                            },
                            {
                                path: 'request.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.to',
                            },
                            {
                                path: 'request.checkout.membershipOptions.eligibleMemberships.expirationDate',
                            },
                            {
                                path: 'request.checkout.membershipOptions.invalidMemberships.membership.expirationDate',
                            },
                        ],
                    },
                    {
                        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
                        paths: [{ path: 'request.checkout.lineItems.url' }],
                    },
                    {
                        transformFn: image_1.transformRESTImageToSDKImage,
                        paths: [
                            { path: 'request.checkout.lineItems.media' },
                            {
                                path: 'request.checkout.lineItems.catalogOverrideFields.image',
                            },
                        ],
                    },
                    {
                        transformFn: float_1.transformRESTFloatToSDKFloat,
                        paths: [
                            { path: 'request.checkout.lineItems.physicalProperties.weight' },
                            {
                                path: 'request.checkout.lineItems.catalogOverrideFields.physicalProperties.weight',
                            },
                            {
                                path: 'request.checkout.appliedDiscounts.merchantDiscount.percentage',
                            },
                        ],
                    },
                    {
                        transformFn: address_1.transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.checkout.billingInfo.address' },
                            {
                                path: 'request.checkout.shippingInfo.shippingDestination.address',
                            },
                            {
                                path: 'request.checkout.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address',
                            },
                            {
                                path: 'request.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address',
                            },
                        ],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
