// @wix/ecom_validations/build/cjs/src/interfaces-ecommerce-v1-validations-provider.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('ECOM_VALIDATIONS', [
    {
        name: 'getValidationViolations',
        primaryHttpMappingPath: '/v1/get-violations',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: float_1.transformRESTFloatToSDKFloat,
                        paths: [
                            {
                                path: 'request.validationInfo.lineItems.physicalProperties.weight',
                            },
                        ],
                    },
                    {
                        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                        paths: [
                            {
                                path: 'request.validationInfo.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
                            },
                        ],
                    },
                    {
                        transformFn: address_1.transformRESTAddressToSDKAddress,
                        paths: [
                            { path: 'request.validationInfo.billingInfo.address' },
                            { path: 'request.validationInfo.shippingAddress.address' },
                        ],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
