// @wix/ecom_subscription-contracts/build/es/src/ecom-subscription-contracts-v1-subscription-contract-subscription-contracts.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl(opts) {
    const domainToMappings = {};
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_subscription-contracts';
/** Retrieves a SubscriptionContract. */
export function getSubscriptionContract(payload) {
    function __getSubscriptionContract({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.subscription_contracts.v1.subscription_contract',
            method: 'GET',
            methodFqn: 'wix.ecom.subscription_contracts.v1.SubscriptionContracts.GetSubscriptionContract',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
                protoPath: '/v1/subscription-contracts/{subscriptionContractId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'subscriptionContract.createdDate' },
                        { path: 'subscriptionContract.updatedDate' },
                        { path: 'subscriptionContract.subscriptionSettings.startDate' },
                        {
                            path: 'subscriptionContract.lineItems.image.urlExpirationDate',
                        },
                        {
                            path: 'subscriptionContract.lineItems.digitalFile.expirationDate',
                        },
                        {
                            path: 'subscriptionContract.lineItems.subscriptionInfo.subscriptionSettings.startDate',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.deliverByDate',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.from',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.deliveryTimeSlot.to',
                        },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'subscriptionContract.lineItems.image.focalPoint.x' },
                        { path: 'subscriptionContract.lineItems.image.focalPoint.y' },
                        {
                            path: 'subscriptionContract.lineItems.physicalProperties.weight',
                        },
                        {
                            path: 'subscriptionContract.billingInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.billingInfo.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.shippingDestination.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.shippingInfo.logistics.pickupDetails.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContract.recipientInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContract.recipientInfo.address.geocode.longitude',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getSubscriptionContract;
}
/**
 * Retrieves a list of SubscriptionContracts, given the provided [paging, filtering, and sorting][1].
 *
 * Up to 300 SubscriptionContracts can be returned per request.
 *
 * To learn how to query SubscriptionContracts, see [API Query Language][2].
 *
 * [1]: https://dev.wix.com/api/rest/getting-started/sorting-and-paging
 * [2]: https://dev.wix.com/api/rest/getting-started/api-query-language
 */
export function querySubscriptionContracts(payload) {
    function __querySubscriptionContracts({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.subscription_contracts.v1.subscription_contract',
            method: 'POST',
            methodFqn: 'wix.ecom.subscription_contracts.v1.SubscriptionContracts.QuerySubscriptionContracts',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
                protoPath: '/v1/subscription-contracts/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'subscriptionContracts.createdDate' },
                        { path: 'subscriptionContracts.updatedDate' },
                        { path: 'subscriptionContracts.subscriptionSettings.startDate' },
                        {
                            path: 'subscriptionContracts.lineItems.image.urlExpirationDate',
                        },
                        {
                            path: 'subscriptionContracts.lineItems.digitalFile.expirationDate',
                        },
                        {
                            path: 'subscriptionContracts.lineItems.subscriptionInfo.subscriptionSettings.startDate',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.deliverByDate',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.from',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.deliveryTimeSlot.to',
                        },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'subscriptionContracts.lineItems.image.focalPoint.x' },
                        { path: 'subscriptionContracts.lineItems.image.focalPoint.y' },
                        {
                            path: 'subscriptionContracts.lineItems.physicalProperties.weight',
                        },
                        {
                            path: 'subscriptionContracts.billingInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.billingInfo.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.shippingDestination.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.shippingInfo.logistics.pickupDetails.address.geocode.longitude',
                        },
                        {
                            path: 'subscriptionContracts.recipientInfo.address.geocode.latitude',
                        },
                        {
                            path: 'subscriptionContracts.recipientInfo.address.geocode.longitude',
                        },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'POST',
                    url: resolveWixEcomSubscriptionContractsV1SubscriptionContractsUrl({
                        protoPath: '/v1/subscription-contracts/query',
                        data: payload,
                        host,
                    }),
                    data: payload,
                },
            ],
        };
        return metadata;
    }
    return __querySubscriptionContracts;
}
