// @wix/ecom_checkout/build/cjs/src/ecom-v1-checkout-checkout.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateLineItemsQuantity = exports.markCheckoutAsCompleted = exports.createOrder = exports.removeLineItems = exports.addToCheckout = exports.removeOverrideCheckoutUrl = exports.removeGiftCard = exports.removeCoupon = exports.updateCheckout = exports.getCheckoutUrl = exports.getCheckoutByCartId = exports.getCheckout = exports.createCheckout = void 0;
const ambassadorWixEcomV1Checkout = __importStar(require("./ecom-v1-checkout-checkout.http.js"));
function createCheckout() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Checkout.createCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCheckout = createCheckout;
function getCheckout() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.getCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkouts/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCheckout = getCheckout;
function getCheckoutByCartId() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.getCheckoutByCartId(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkouts/by-cart-id/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCheckoutByCartId = getCheckoutByCartId;
function getCheckoutUrl() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.getCheckoutUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkouts/{id}/checkout-url',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCheckoutUrl = getCheckoutUrl;
function updateCheckout() {
    const payload = { checkout: { id: ':checkoutId' } };
    const getRequestOptions = ambassadorWixEcomV1Checkout.updateCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/checkouts/{checkout.id}',
        pathParams: { checkoutId: 'checkoutId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCheckout = updateCheckout;
function removeCoupon() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeCoupon(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-coupon',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeCoupon = removeCoupon;
function removeGiftCard() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeGiftCard(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-gift-card',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeGiftCard = removeGiftCard;
function removeOverrideCheckoutUrl() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeOverrideCheckoutUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-override-checkout-url',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeOverrideCheckoutUrl = removeOverrideCheckoutUrl;
function addToCheckout() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.addToCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/add-to-checkout',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addToCheckout = addToCheckout;
function removeLineItems() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.removeLineItems(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/remove-line-items',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeLineItems = removeLineItems;
function createOrder() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.createOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/create-order',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createOrder = createOrder;
function markCheckoutAsCompleted() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.markCheckoutAsCompleted(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/mark-as-completed',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.markCheckoutAsCompleted = markCheckoutAsCompleted;
function updateLineItemsQuantity() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1Checkout.updateLineItemsQuantity(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkouts/{id}/update-line-items-quantity',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateLineItemsQuantity = updateLineItemsQuantity;
