// @wix/ecom_checkout/build/cjs/src/ecom-v1-checkout-checkout.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeightUnit = exports.WebhookIdentityType = exports.VatType = exports.TaxableAddressType = exports.SuggestedFix = exports.SubscriptionFrequency = exports.Severity = exports.RuleType = exports.RateType = exports.PickupMethod = exports.PaymentOptionType = exports.NameInOther = exports.NameInLineItem = exports.MediaItemType = exports.ManualCalculationReason = exports.LineItemType = exports.JurisdictionType = exports.ItemTypeItemType = exports.ItemAvailabilityStatus = exports.IdentityType = exports.GetCheckoutPaymentSettingsResponsePaymentOption = exports.FileType = exports.FallbackReason = exports.DiscountType = exports.DiscountDiscountType = exports.DescriptionLineType = exports.CommonVatType = exports.ChargeType = exports.ChannelType = exports.ChannelInfoChannelType = exports.AdditionalFeeSource = exports.onCheckoutUpdated = exports.onCheckoutCreated = exports.onCheckoutCompleted = exports.updateLineItemsQuantity = exports.markCheckoutAsCompleted = exports.createOrder = exports.removeLineItems = exports.addToCheckout = exports.removeOverrideCheckoutUrl = exports.removeGiftCard = exports.removeCoupon = exports.updateCheckout = exports.getCheckoutUrl = exports.getCheckoutByCartId = exports.getCheckout = exports.createCheckout = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_checkout_checkout_universal_js_1 = require("./ecom-v1-checkout-checkout.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createCheckout(httpClient) {
    return (options) => (0, ecom_v1_checkout_checkout_universal_js_1.createCheckout)(options, 
    // @ts-ignore
    { httpClient });
}
exports.createCheckout = createCheckout;
function getCheckout(httpClient) {
    return (_id) => (0, ecom_v1_checkout_checkout_universal_js_1.getCheckout)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getCheckout = getCheckout;
function getCheckoutByCartId(httpClient) {
    return (_id) => (0, ecom_v1_checkout_checkout_universal_js_1.getCheckoutByCartId)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getCheckoutByCartId = getCheckoutByCartId;
function getCheckoutUrl(httpClient) {
    return (_id) => (0, ecom_v1_checkout_checkout_universal_js_1.getCheckoutUrl)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getCheckoutUrl = getCheckoutUrl;
function updateCheckout(httpClient) {
    return (_id, checkout, options) => (0, ecom_v1_checkout_checkout_universal_js_1.updateCheckout)(_id, checkout, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateCheckout = updateCheckout;
function removeCoupon(httpClient) {
    return (_id) => (0, ecom_v1_checkout_checkout_universal_js_1.removeCoupon)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.removeCoupon = removeCoupon;
function removeGiftCard(httpClient) {
    return (_id) => (0, ecom_v1_checkout_checkout_universal_js_1.removeGiftCard)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.removeGiftCard = removeGiftCard;
function removeOverrideCheckoutUrl(httpClient) {
    return (_id) => (0, ecom_v1_checkout_checkout_universal_js_1.removeOverrideCheckoutUrl)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.removeOverrideCheckoutUrl = removeOverrideCheckoutUrl;
function addToCheckout(httpClient) {
    return (_id, options) => (0, ecom_v1_checkout_checkout_universal_js_1.addToCheckout)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.addToCheckout = addToCheckout;
function removeLineItems(httpClient) {
    return (_id, lineItemIds) => (0, ecom_v1_checkout_checkout_universal_js_1.removeLineItems)(_id, lineItemIds, 
    // @ts-ignore
    { httpClient });
}
exports.removeLineItems = removeLineItems;
function createOrder(httpClient) {
    return (_id, options) => (0, ecom_v1_checkout_checkout_universal_js_1.createOrder)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.createOrder = createOrder;
function markCheckoutAsCompleted(httpClient) {
    return (_id) => (0, ecom_v1_checkout_checkout_universal_js_1.markCheckoutAsCompleted)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.markCheckoutAsCompleted = markCheckoutAsCompleted;
function updateLineItemsQuantity(httpClient) {
    return (_id, lineItems) => (0, ecom_v1_checkout_checkout_universal_js_1.updateLineItemsQuantity)(_id, lineItems, 
    // @ts-ignore
    { httpClient });
}
exports.updateLineItemsQuantity = updateLineItemsQuantity;
exports.onCheckoutCompleted = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_completed', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.checkout.createdDate' },
            { path: 'data.checkout.updatedDate' },
            {
                path: 'data.checkout.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'data.checkout.lineItems.serviceProperties.scheduledDate' },
            {
                path: 'data.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'data.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'data.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'data.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'data.checkout.membershipOptions.eligibleMemberships.expirationDate',
            },
            {
                path: 'data.checkout.membershipOptions.invalidMemberships.membership.expirationDate',
            },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'data.checkout.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'data.checkout.lineItems.media' },
            { path: 'data.checkout.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'data.checkout.lineItems.physicalProperties.weight' },
            {
                path: 'data.checkout.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            {
                path: 'data.checkout.appliedDiscounts.merchantDiscount.percentage',
            },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'data.checkout.billingInfo.address' },
            { path: 'data.checkout.shippingInfo.shippingDestination.address' },
            {
                path: 'data.checkout.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address',
            },
            {
                path: 'data.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address',
            },
        ],
    },
])))();
exports.onCheckoutCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            {
                path: 'entity.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.lineItems.serviceProperties.scheduledDate' },
            {
                path: 'entity.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'entity.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'entity.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'entity.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'entity.membershipOptions.eligibleMemberships.expirationDate',
            },
            {
                path: 'entity.membershipOptions.invalidMemberships.membership.expirationDate',
            },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.media' },
            { path: 'entity.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.lineItems.physicalProperties.weight' },
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            { path: 'entity.appliedDiscounts.merchantDiscount.percentage' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.billingInfo.address' },
            { path: 'entity.shippingInfo.shippingDestination.address' },
            {
                path: 'entity.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address',
            },
            {
                path: 'entity.shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address',
            },
        ],
    },
])))();
exports.onCheckoutUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            {
                path: 'entity.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.lineItems.serviceProperties.scheduledDate' },
            {
                path: 'entity.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'entity.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'entity.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'entity.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'entity.membershipOptions.eligibleMemberships.expirationDate',
            },
            {
                path: 'entity.membershipOptions.invalidMemberships.membership.expirationDate',
            },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.lineItems.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.media' },
            { path: 'entity.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.lineItems.physicalProperties.weight' },
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            { path: 'entity.appliedDiscounts.merchantDiscount.percentage' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.billingInfo.address' },
            { path: 'entity.shippingInfo.shippingDestination.address' },
            {
                path: 'entity.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address',
            },
            {
                path: 'entity.shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address',
            },
        ],
    },
])))();
var ecom_v1_checkout_checkout_universal_js_2 = require("./ecom-v1-checkout-checkout.universal.js");
Object.defineProperty(exports, "AdditionalFeeSource", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.AdditionalFeeSource; } });
Object.defineProperty(exports, "ChannelInfoChannelType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.ChannelInfoChannelType; } });
Object.defineProperty(exports, "ChannelType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.ChannelType; } });
Object.defineProperty(exports, "ChargeType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.ChargeType; } });
Object.defineProperty(exports, "CommonVatType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.CommonVatType; } });
Object.defineProperty(exports, "DescriptionLineType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.DescriptionLineType; } });
Object.defineProperty(exports, "DiscountDiscountType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.DiscountDiscountType; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "FallbackReason", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.FallbackReason; } });
Object.defineProperty(exports, "FileType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.FileType; } });
Object.defineProperty(exports, "GetCheckoutPaymentSettingsResponsePaymentOption", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.GetCheckoutPaymentSettingsResponsePaymentOption; } });
Object.defineProperty(exports, "IdentityType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.IdentityType; } });
Object.defineProperty(exports, "ItemAvailabilityStatus", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.ItemAvailabilityStatus; } });
Object.defineProperty(exports, "ItemTypeItemType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.ItemTypeItemType; } });
Object.defineProperty(exports, "JurisdictionType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.JurisdictionType; } });
Object.defineProperty(exports, "LineItemType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.LineItemType; } });
Object.defineProperty(exports, "ManualCalculationReason", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.ManualCalculationReason; } });
Object.defineProperty(exports, "MediaItemType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.MediaItemType; } });
Object.defineProperty(exports, "NameInLineItem", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.NameInLineItem; } });
Object.defineProperty(exports, "NameInOther", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.NameInOther; } });
Object.defineProperty(exports, "PaymentOptionType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.PaymentOptionType; } });
Object.defineProperty(exports, "PickupMethod", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.PickupMethod; } });
Object.defineProperty(exports, "RateType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.RateType; } });
Object.defineProperty(exports, "RuleType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.RuleType; } });
Object.defineProperty(exports, "Severity", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.Severity; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "SuggestedFix", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.SuggestedFix; } });
Object.defineProperty(exports, "TaxableAddressType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.TaxableAddressType; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.VatType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return ecom_v1_checkout_checkout_universal_js_2.WeightUnit; } });
