// @wix/ecom_tips/build/es/src/tips-v1-tip-tips.meta.js
import * as ambassadorWixTipsV1Tip from './tips-v1-tip-tips.http.js';
export function previewTip() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.previewTip(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/preview',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTipByOrderId() {
    const payload = { orderId: ':orderId' };
    const getRequestOptions = ambassadorWixTipsV1Tip.getTipByOrderId(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tips/order/{orderId}',
        pathParams: { orderId: 'orderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTipByDraftOrderId() {
    const payload = { draftOrderId: ':draftOrderId' };
    const getRequestOptions = ambassadorWixTipsV1Tip.getTipByDraftOrderId(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tips/draft-order/{draftOrderId}',
        pathParams: { draftOrderId: 'draftOrderId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTipChoice() {
    const payload = { purchaseFlowId: ':purchaseFlowId' };
    const getRequestOptions = ambassadorWixTipsV1Tip.getTipChoice(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tips/choice/{purchaseFlowId}',
        pathParams: { purchaseFlowId: 'purchaseFlowId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateTipChoice() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.updateTipChoice(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tips/choice',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setTipOnOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.setTipOnOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/set',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeTipFromOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.removeTipFromOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/remove',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addTipDistributionsToOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.addTipDistributionsToOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/tip-distributions/add',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryTipDistributions() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.queryTipDistributions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/tip-distributions/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function sumTipDistributionAmounts() {
    const payload = {};
    const getRequestOptions = ambassadorWixTipsV1Tip.sumTipDistributionAmounts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tips/tip-distributions/sum-amounts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
