// @wix/ecom_order-billing/build/cjs/src/ecom-v1-order-billing-order-billing.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.captureAuthorizedPayments = exports.voidAuthorizedPayments = exports.authorizeChargeWithSavedPaymentMethod = exports.refundPayments = exports.calculateRefund = exports.getOrderRefundability = void 0;
const ambassadorWixEcomV1OrderBilling = __importStar(require("./ecom-v1-order-billing-order-billing.http.js"));
function getOrderRefundability() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.getOrderRefundability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/get-order-refundability',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getOrderRefundability = getOrderRefundability;
function calculateRefund() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.calculateRefund(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/calculate-refund',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.calculateRefund = calculateRefund;
function refundPayments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.refundPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/refund-payments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.refundPayments = refundPayments;
function authorizeChargeWithSavedPaymentMethod() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.authorizeChargeWithSavedPaymentMethod(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/authorize-charge-with-saved-payment-method',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.authorizeChargeWithSavedPaymentMethod = authorizeChargeWithSavedPaymentMethod;
function voidAuthorizedPayments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.voidAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/void-authorized-payments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.voidAuthorizedPayments = voidAuthorizedPayments;
function captureAuthorizedPayments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.captureAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/capture-authorized-payments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.captureAuthorizedPayments = captureAuthorizedPayments;
