// @wix/ecom_order-billing/build/cjs/src/ecom-v1-order-billing-order-billing.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionStatus = exports.ScheduledAction = exports.RuleType = exports.RefundStatus = exports.Reason = exports.NonRefundableReason = exports.MembershipPaymentStatus = exports.ManuallyRefundableReason = exports.DurationUnit = exports.AuthorizationVoidStatus = exports.AuthorizationCaptureStatus = exports.ActionType = exports.captureAuthorizedPayments = exports.voidAuthorizedPayments = exports.authorizeChargeWithSavedPaymentMethod = exports.refundPayments = exports.calculateRefund = exports.getOrderRefundability = exports.__metadata = void 0;
const ecom_v1_order_billing_order_billing_universal_js_1 = require("./ecom-v1-order-billing-order-billing.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function getOrderRefundability(httpClient) {
    return (orderId) => (0, ecom_v1_order_billing_order_billing_universal_js_1.getOrderRefundability)(orderId, 
    // @ts-ignore
    { httpClient });
}
exports.getOrderRefundability = getOrderRefundability;
function calculateRefund(httpClient) {
    return (orderId, options) => (0, ecom_v1_order_billing_order_billing_universal_js_1.calculateRefund)(orderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.calculateRefund = calculateRefund;
function refundPayments(httpClient) {
    return (orderId, options) => (0, ecom_v1_order_billing_order_billing_universal_js_1.refundPayments)(orderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.refundPayments = refundPayments;
function authorizeChargeWithSavedPaymentMethod(httpClient) {
    return (orderId, options) => (0, ecom_v1_order_billing_order_billing_universal_js_1.authorizeChargeWithSavedPaymentMethod)(orderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.authorizeChargeWithSavedPaymentMethod = authorizeChargeWithSavedPaymentMethod;
function voidAuthorizedPayments(httpClient) {
    return (orderId, options) => (0, ecom_v1_order_billing_order_billing_universal_js_1.voidAuthorizedPayments)(orderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.voidAuthorizedPayments = voidAuthorizedPayments;
function captureAuthorizedPayments(httpClient) {
    return (orderId, options) => (0, ecom_v1_order_billing_order_billing_universal_js_1.captureAuthorizedPayments)(orderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.captureAuthorizedPayments = captureAuthorizedPayments;
var ecom_v1_order_billing_order_billing_universal_js_2 = require("./ecom-v1-order-billing-order-billing.universal.js");
Object.defineProperty(exports, "ActionType", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.ActionType; } });
Object.defineProperty(exports, "AuthorizationCaptureStatus", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.AuthorizationCaptureStatus; } });
Object.defineProperty(exports, "AuthorizationVoidStatus", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.AuthorizationVoidStatus; } });
Object.defineProperty(exports, "DurationUnit", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.DurationUnit; } });
Object.defineProperty(exports, "ManuallyRefundableReason", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.ManuallyRefundableReason; } });
Object.defineProperty(exports, "MembershipPaymentStatus", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.MembershipPaymentStatus; } });
Object.defineProperty(exports, "NonRefundableReason", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.NonRefundableReason; } });
Object.defineProperty(exports, "Reason", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.Reason; } });
Object.defineProperty(exports, "RefundStatus", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.RefundStatus; } });
Object.defineProperty(exports, "RuleType", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.RuleType; } });
Object.defineProperty(exports, "ScheduledAction", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.ScheduledAction; } });
Object.defineProperty(exports, "TransactionStatus", { enumerable: true, get: function () { return ecom_v1_order_billing_order_billing_universal_js_2.TransactionStatus; } });
