// @wix/ecom_delivery-profile/build/cjs/src/ecom-v1-delivery-profile-delivery-profile.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateExtendedFields = exports.updateCarrierSettings = exports.listDeliveryCarriers = exports.listDeliveryDestinations = exports.listInstalledDeliveryCarriers = exports.listDeliveryCarrierDetails = exports.updateDeliveryCarrier = exports.removeDeliveryCarrier = exports.addDeliveryCarrier = exports.removeDeliveryRegion = exports.updateDeliveryRegion = exports.addDeliveryRegion = exports.queryDeliveryProfiles = exports.deleteDeliveryProfile = exports.updateDeliveryProfile = exports.getDeliveryProfile = exports.createDeliveryProfile = void 0;
const ambassadorWixEcomV1DeliveryProfile = __importStar(require("./ecom-v1-delivery-profile-delivery-profile.http.js"));
function createDeliveryProfile() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.createDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDeliveryProfile = createDeliveryProfile;
function getDeliveryProfile() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.getDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/{deliveryProfileId}',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getDeliveryProfile = getDeliveryProfile;
function updateDeliveryProfile() {
    const payload = { deliveryProfile: { id: ':deliveryProfileId' } };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/delivery-profiles/{deliveryProfile.id}',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDeliveryProfile = updateDeliveryProfile;
function deleteDeliveryProfile() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.deleteDeliveryProfile(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/delivery-profiles/{deliveryProfileId}',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteDeliveryProfile = deleteDeliveryProfile;
function queryDeliveryProfiles() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.queryDeliveryProfiles(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryDeliveryProfiles = queryDeliveryProfiles;
function addDeliveryRegion() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.addDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addDeliveryRegion = addDeliveryRegion;
function updateDeliveryRegion() {
    const payload = {
        deliveryProfileId: ':deliveryProfileId',
        deliveryRegion: { id: ':deliveryRegionId' },
    };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegion.id}',
        pathParams: {
            deliveryProfileId: 'deliveryProfileId',
            deliveryRegionId: 'deliveryRegionId',
        },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDeliveryRegion = updateDeliveryRegion;
function removeDeliveryRegion() {
    const payload = {
        deliveryProfileId: ':deliveryProfileId',
        deliveryRegionId: ':deliveryRegionId',
    };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.removeDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegionId}',
        pathParams: {
            deliveryProfileId: 'deliveryProfileId',
            deliveryRegionId: 'deliveryRegionId',
        },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeDeliveryRegion = removeDeliveryRegion;
function addDeliveryCarrier() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.addDeliveryCarrier(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/add-delivery-carrier',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addDeliveryCarrier = addDeliveryCarrier;
function removeDeliveryCarrier() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.removeDeliveryCarrier(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/remove-delivery-carrier',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeDeliveryCarrier = removeDeliveryCarrier;
function updateDeliveryCarrier() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateDeliveryCarrier(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/delivery-profiles/update-delivery-carrier',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDeliveryCarrier = updateDeliveryCarrier;
function listDeliveryCarrierDetails() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listDeliveryCarrierDetails(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-carriers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDeliveryCarrierDetails = listDeliveryCarrierDetails;
function listInstalledDeliveryCarriers() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listInstalledDeliveryCarriers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/installed-carriers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listInstalledDeliveryCarriers = listInstalledDeliveryCarriers;
function listDeliveryDestinations() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listDeliveryDestinations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/delivery-profiles/list-delivery-destinations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDeliveryDestinations = listDeliveryDestinations;
function listDeliveryCarriers() {
    const payload = { deliveryProfileId: ':deliveryProfileId' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.listDeliveryCarriers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/{deliveryProfileId}/delivery-carriers',
        pathParams: { deliveryProfileId: 'deliveryProfileId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDeliveryCarriers = listDeliveryCarriers;
function updateCarrierSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateCarrierSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-carriers/update-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCarrierSettings = updateCarrierSettings;
function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixEcomV1DeliveryProfile.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/delivery-profiles/{id}/update-extended-fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateExtendedFields = updateExtendedFields;
