// @wix/ecom_delivery-profile/build/cjs/src/ecom-v1-delivery-profile-delivery-profile.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateExtendedFields = exports.updateCarrierSettings = exports.listDeliveryCarriers = exports.listDeliveryDestinations = exports.listInstalledDeliveryCarriers = exports.listDeliveryCarrierDetails = exports.updateDeliveryCarrier = exports.removeDeliveryCarrier = exports.addDeliveryCarrier = exports.removeDeliveryRegion = exports.updateDeliveryRegion = exports.addDeliveryRegion = exports.queryDeliveryProfiles = exports.deleteDeliveryProfile = exports.updateDeliveryProfile = exports.getDeliveryProfile = exports.createDeliveryProfile = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/ecom/v1/list-delivery-carriers',
                destPath: '/v1/list-delivery-carriers',
            },
            {
                srcPath: '/ecom/v1/list-delivery-carrier-details',
                destPath: '/v1/list-delivery-carrier-details',
            },
            {
                srcPath: '/ecom/v1/delivery-profiles',
                destPath: '/v1/delivery-profiles',
            },
            {
                srcPath: '/ecom/v1/delivery-carriers',
                destPath: '/v1/delivery-carriers',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/delivery-profiles-service',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_delivery-profile';
/**
 * Creates a DeliveryProfile.
 *
 * The request body must include the DeliveryProfiles name. DeliveryRegions are optional.
 */
function createDeliveryProfile(payload) {
    function __createDeliveryProfile({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'deliveryProfile.createdDate' },
                    { path: 'deliveryProfile.updatedDate' },
                    { path: 'deliveryProfile.deliveryRegions.createdDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.CreateDeliveryProfile',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createDeliveryProfile;
}
exports.createDeliveryProfile = createDeliveryProfile;
/** Retrieves a DeliveryProfile. */
function getDeliveryProfile(payload) {
    function __getDeliveryProfile({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'GET',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.GetDeliveryProfile',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{deliveryProfileId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getDeliveryProfile;
}
exports.getDeliveryProfile = getDeliveryProfile;
/**
 * Updates a DeliveryProfile.
 *
 *
 * Each time the DeliveryProfile is updated,
 * `revision` increments by 1.
 * The current `revision` must be passed when updating the DeliveryProfile.
 * This ensures you're working with the latest DeliveryProfile
 * and prevents unintended overwrites.
 *
 * This method does not allow updating the DeliveryRegions in this DeliveryProfile.
 * Use AddDeliveryRegion, UpdateDeliveryRegion and RemoveDeliveryRegion for these functionalities.
 */
function updateDeliveryProfile(payload) {
    function __updateDeliveryProfile({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'deliveryProfile.createdDate' },
                    { path: 'deliveryProfile.updatedDate' },
                    { path: 'deliveryProfile.deliveryRegions.createdDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'PATCH',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateDeliveryProfile',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{deliveryProfile.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateDeliveryProfile;
}
exports.updateDeliveryProfile = updateDeliveryProfile;
/**
 * Deletes a DeliveryProfile.
 *
 * Deleting a DeliveryProfile permanently removes them from the DeliveryProfile List.
 */
function deleteDeliveryProfile(payload) {
    function __deleteDeliveryProfile({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'DELETE',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.DeleteDeliveryProfile',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{deliveryProfileId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteDeliveryProfile;
}
exports.deleteDeliveryProfile = deleteDeliveryProfile;
/**
 * Retrieves a list of DeliveryProfiles, given the provided [paging, filtering, and sorting][1].
 *
 * Up to 1,000 DeliveryProfiles can be returned per request.
 *
 * To learn how to query DeliveryProfiles, see [API Query Language][2].
 *
 * [1]: https://dev.wix.com/api/rest/getting-started/sorting-and-paging
 * [2]: https://dev.wix.com/api/rest/getting-started/api-query-language
 */
function queryDeliveryProfiles(payload) {
    function __queryDeliveryProfiles({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.QueryDeliveryProfiles',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfiles.createdDate' },
                        { path: 'deliveryProfiles.updatedDate' },
                        { path: 'deliveryProfiles.deliveryRegions.createdDate' },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'POST',
                    url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                        protoPath: '/v1/delivery-profiles/query',
                        data: payload,
                        host,
                    }),
                    data: payload,
                },
            ],
        };
        return metadata;
    }
    return __queryDeliveryProfiles;
}
exports.queryDeliveryProfiles = queryDeliveryProfiles;
/** Creates a new DeliveryRegion in an existing DeliveryProfile. */
function addDeliveryRegion(payload) {
    function __addDeliveryRegion({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'deliveryRegion.createdDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.AddDeliveryRegion',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __addDeliveryRegion;
}
exports.addDeliveryRegion = addDeliveryRegion;
/**
 * Update a DeliveryRegion, supports partial update
 * Pass the latest `revision` for a successful update
 *
 * This method does not allow setting or updating the delivery delivery_carriers in this delivery_region.
 * Use AddDeliveryCarrier, UpdateDeliveryCarrier and RemoveDeliveryCarrier for these functionalities.
 */
function updateDeliveryRegion(payload) {
    function __updateDeliveryRegion({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'deliveryRegion.createdDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'PATCH',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateDeliveryRegion',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegion.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateDeliveryRegion;
}
exports.updateDeliveryRegion = updateDeliveryRegion;
/** Delete a DeliveryRegion */
function removeDeliveryRegion(payload) {
    function __removeDeliveryRegion({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'DELETE',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.RemoveDeliveryRegion',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{deliveryProfileId}/delivery-region/{deliveryRegionId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __removeDeliveryRegion;
}
exports.removeDeliveryRegion = removeDeliveryRegion;
/** Add a delivery_carrier to a delivery region */
function addDeliveryCarrier(payload) {
    function __addDeliveryCarrier({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.AddDeliveryCarrier',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/add-delivery-carrier',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __addDeliveryCarrier;
}
exports.addDeliveryCarrier = addDeliveryCarrier;
/** Remove a delivery_carrier from a delivery region */
function removeDeliveryCarrier(payload) {
    function __removeDeliveryCarrier({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.RemoveDeliveryCarrier',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/remove-delivery-carrier',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __removeDeliveryCarrier;
}
exports.removeDeliveryCarrier = removeDeliveryCarrier;
/** Update a delivery carrier in a delivery region */
function updateDeliveryCarrier(payload) {
    function __updateDeliveryCarrier({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'PATCH',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateDeliveryCarrier',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/update-delivery-carrier',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateDeliveryCarrier;
}
exports.updateDeliveryCarrier = updateDeliveryCarrier;
/**
 * List all installed delivery carriers in order to set their ids in delivery rules
 * @deprecated It has been replaced with ListInstalledDeliveryCarriers(), and will be removed on 2025-04-01.
 */
function listDeliveryCarrierDetails(payload) {
    function __listDeliveryCarrierDetails({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'GET',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.ListDeliveryCarrierDetails',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-carriers',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __listDeliveryCarrierDetails;
}
exports.listDeliveryCarrierDetails = listDeliveryCarrierDetails;
function listInstalledDeliveryCarriers(payload) {
    function __listInstalledDeliveryCarriers({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'GET',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.ListInstalledDeliveryCarriers',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/installed-carriers',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __listInstalledDeliveryCarriers;
}
exports.listInstalledDeliveryCarriers = listInstalledDeliveryCarriers;
/**
 * Retrieves a list of all delivery destinations configured on a site.
 * Sorted by `countryCode` in ascending order.
 */
function listDeliveryDestinations(payload) {
    function __listDeliveryDestinations({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'GET',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.ListDeliveryDestinations',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/list-delivery-destinations',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __listDeliveryDestinations;
}
exports.listDeliveryDestinations = listDeliveryDestinations;
/**
 * Get delivery carrier settings for a delivery profile and delivery region.
 * These are returned in a table format for the dashboard.
 */
function listDeliveryCarriers(payload) {
    function __listDeliveryCarriers({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.ListDeliveryCarriers',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{deliveryProfileId}/delivery-carriers',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __listDeliveryCarriers;
}
exports.listDeliveryCarriers = listDeliveryCarriers;
/** Update carrier's external active settings */
function updateCarrierSettings(payload) {
    function __updateCarrierSettings({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateCarrierSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-carriers/update-settings',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __updateCarrierSettings;
}
exports.updateCarrierSettings = updateCarrierSettings;
/** Updates extended fields of a DeliveryProfile without incrementing revision */
function updateExtendedFields(payload) {
    function __updateExtendedFields({ host }) {
        const metadata = {
            entityFqdn: 'wix.ecom.v1.delivery_profile',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryProfiles.UpdateExtendedFields',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryProfilesUrl({
                protoPath: '/v1/delivery-profiles/{id}/update-extended-fields',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliveryProfile.createdDate' },
                        { path: 'deliveryProfile.updatedDate' },
                        { path: 'deliveryProfile.deliveryRegions.createdDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateExtendedFields;
}
exports.updateExtendedFields = updateExtendedFields;
