// @wix/ecom_delivery-profile/build/cjs/src/ecom-v1-delivery-profile-delivery-profile.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.State = exports.SortOrder = exports.SiteCreatedContext = exports.Namespace = exports.DestinationScope = exports.DeleteStatus = exports.ChargeType = exports.updateExtendedFields = exports.updateCarrierSettings = exports.listDeliveryCarriers = exports.listDeliveryDestinations = exports.listInstalledDeliveryCarriers = exports.listDeliveryCarrierDetails = exports.updateDeliveryCarrier = exports.removeDeliveryCarrier = exports.addDeliveryCarrier = exports.removeDeliveryRegion = exports.updateDeliveryRegion = exports.addDeliveryRegion = exports.queryDeliveryProfiles = exports.deleteDeliveryProfile = exports.updateDeliveryProfile = exports.getDeliveryProfile = exports.createDeliveryProfile = exports.__metadata = void 0;
const ecom_v1_delivery_profile_delivery_profile_universal_js_1 = require("./ecom-v1-delivery-profile-delivery-profile.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createDeliveryProfile(httpClient) {
    return (deliveryProfile) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.createDeliveryProfile)(deliveryProfile, 
    // @ts-ignore
    { httpClient });
}
exports.createDeliveryProfile = createDeliveryProfile;
function getDeliveryProfile(httpClient) {
    return (deliveryProfileId) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.getDeliveryProfile)(deliveryProfileId, 
    // @ts-ignore
    { httpClient });
}
exports.getDeliveryProfile = getDeliveryProfile;
function updateDeliveryProfile(httpClient) {
    return (_id, deliveryProfile) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.updateDeliveryProfile)(_id, deliveryProfile, 
    // @ts-ignore
    { httpClient });
}
exports.updateDeliveryProfile = updateDeliveryProfile;
function deleteDeliveryProfile(httpClient) {
    return (deliveryProfileId) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.deleteDeliveryProfile)(deliveryProfileId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteDeliveryProfile = deleteDeliveryProfile;
function queryDeliveryProfiles(httpClient) {
    return () => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.queryDeliveryProfiles)(
    // @ts-ignore
    { httpClient });
}
exports.queryDeliveryProfiles = queryDeliveryProfiles;
function addDeliveryRegion(httpClient) {
    return (deliveryProfileId, deliveryRegion, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.addDeliveryRegion)(deliveryProfileId, deliveryRegion, options, 
    // @ts-ignore
    { httpClient });
}
exports.addDeliveryRegion = addDeliveryRegion;
function updateDeliveryRegion(httpClient) {
    return (identifiers, deliveryRegion, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.updateDeliveryRegion)(identifiers, deliveryRegion, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateDeliveryRegion = updateDeliveryRegion;
function removeDeliveryRegion(httpClient) {
    return (identifiers, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.removeDeliveryRegion)(identifiers, options, 
    // @ts-ignore
    { httpClient });
}
exports.removeDeliveryRegion = removeDeliveryRegion;
function addDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.addDeliveryCarrier)(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.addDeliveryCarrier = addDeliveryCarrier;
function removeDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.removeDeliveryCarrier)(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.removeDeliveryCarrier = removeDeliveryCarrier;
function updateDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.updateDeliveryCarrier)(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateDeliveryCarrier = updateDeliveryCarrier;
function listDeliveryCarrierDetails(httpClient) {
    return () => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.listDeliveryCarrierDetails)(
    // @ts-ignore
    { httpClient });
}
exports.listDeliveryCarrierDetails = listDeliveryCarrierDetails;
function listInstalledDeliveryCarriers(httpClient) {
    return () => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.listInstalledDeliveryCarriers)(
    // @ts-ignore
    { httpClient });
}
exports.listInstalledDeliveryCarriers = listInstalledDeliveryCarriers;
function listDeliveryDestinations(httpClient) {
    return () => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.listDeliveryDestinations)(
    // @ts-ignore
    { httpClient });
}
exports.listDeliveryDestinations = listDeliveryDestinations;
function listDeliveryCarriers(httpClient) {
    return (deliveryProfileId, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.listDeliveryCarriers)(deliveryProfileId, options, 
    // @ts-ignore
    { httpClient });
}
exports.listDeliveryCarriers = listDeliveryCarriers;
function updateCarrierSettings(httpClient) {
    return (carrierId, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.updateCarrierSettings)(carrierId, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateCarrierSettings = updateCarrierSettings;
function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => (0, ecom_v1_delivery_profile_delivery_profile_universal_js_1.updateExtendedFields)(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateExtendedFields = updateExtendedFields;
var ecom_v1_delivery_profile_delivery_profile_universal_js_2 = require("./ecom-v1-delivery-profile-delivery-profile.universal.js");
Object.defineProperty(exports, "ChargeType", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.ChargeType; } });
Object.defineProperty(exports, "DeleteStatus", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.DeleteStatus; } });
Object.defineProperty(exports, "DestinationScope", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.DestinationScope; } });
Object.defineProperty(exports, "Namespace", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.Namespace; } });
Object.defineProperty(exports, "SiteCreatedContext", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.SiteCreatedContext; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "State", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.State; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_delivery_profile_delivery_profile_universal_js_2.WebhookIdentityType; } });
