// @wix/ecom_discounts/build/cjs/src/service-plugins-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityType = exports.DiscountSource = exports.Type = exports.TargetType = void 0;
var TargetType;
(function (TargetType) {
    TargetType["UNKNOWN_TARGET"] = "UNKNOWN_TARGET";
    TargetType["GLOBAL"] = "GLOBAL";
    TargetType["SHIPPING"] = "SHIPPING";
    TargetType["LINE_ITEMS"] = "LINE_ITEMS";
})(TargetType || (exports.TargetType = TargetType = {}));
var Type;
(function (Type) {
    Type["UNKNOWN_RATE"] = "UNKNOWN_RATE";
    Type["FIXED_AMOUNT"] = "FIXED_AMOUNT";
    Type["PERCENTAGE"] = "PERCENTAGE";
})(Type || (exports.Type = Type = {}));
var DiscountSource;
(function (DiscountSource) {
    DiscountSource["UNKNOWN_SOURCE"] = "UNKNOWN_SOURCE";
    DiscountSource["COUPON"] = "COUPON";
    DiscountSource["AUTOMATIC"] = "AUTOMATIC";
    DiscountSource["CUSTOM"] = "CUSTOM";
})(DiscountSource || (exports.DiscountSource = DiscountSource = {}));
var IdentityType;
(function (IdentityType) {
    IdentityType["UNKNOWN"] = "UNKNOWN";
    IdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    IdentityType["MEMBER"] = "MEMBER";
    IdentityType["WIX_USER"] = "WIX_USER";
    IdentityType["APP"] = "APP";
})(IdentityType || (exports.IdentityType = IdentityType = {}));
