// @wix/ecom_discounts/build/cjs/src/service-plugins-error-classes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CouponDoesNotExistWixError = exports.CouponIsNotActiveYetWixError = exports.CouponHasExpiredWixError = exports.CouponUsageExceededWixError = exports.CouponIsDisabledWixError = exports.CouponLimitPerCustomerExceededWixError = void 0;
class CouponLimitPerCustomerExceededWixError extends Error {
    constructor() {
        super('CouponLimitPerCustomerExceeded');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED';
        this.name = 'CouponLimitPerCustomerExceeded';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponLimitPerCustomerExceeded',
            applicationCode: 'ERROR_COUPON_LIMIT_PER_CUSTOMER_EXCEEDED',
        };
    }
}
exports.CouponLimitPerCustomerExceededWixError = CouponLimitPerCustomerExceededWixError;
/** @hidden */
CouponLimitPerCustomerExceededWixError.__type = 'wix_spi_error';
class CouponIsDisabledWixError extends Error {
    constructor() {
        super('CouponIsDisabled');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_IS_DISABLED';
        this.name = 'CouponIsDisabled';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponIsDisabled',
            applicationCode: 'ERROR_COUPON_IS_DISABLED',
        };
    }
}
exports.CouponIsDisabledWixError = CouponIsDisabledWixError;
/** @hidden */
CouponIsDisabledWixError.__type = 'wix_spi_error';
class CouponUsageExceededWixError extends Error {
    constructor() {
        super('CouponUsageExceeded');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_USAGE_EXCEEDED';
        this.name = 'CouponUsageExceeded';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponUsageExceeded',
            applicationCode: 'ERROR_COUPON_USAGE_EXCEEDED',
        };
    }
}
exports.CouponUsageExceededWixError = CouponUsageExceededWixError;
/** @hidden */
CouponUsageExceededWixError.__type = 'wix_spi_error';
class CouponHasExpiredWixError extends Error {
    constructor() {
        super('CouponHasExpired');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_HAS_EXPIRED';
        this.name = 'CouponHasExpired';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponHasExpired',
            applicationCode: 'ERROR_COUPON_HAS_EXPIRED',
        };
    }
}
exports.CouponHasExpiredWixError = CouponHasExpiredWixError;
/** @hidden */
CouponHasExpiredWixError.__type = 'wix_spi_error';
class CouponIsNotActiveYetWixError extends Error {
    constructor() {
        super('CouponIsNotActiveYet');
        this.httpCode = 428;
        this.statusCode = 'FAILED_PRECONDITION';
        this.applicationCode = 'ERROR_COUPON_IS_NOT_ACTIVE_YET';
        this.name = 'CouponIsNotActiveYet';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponIsNotActiveYet',
            applicationCode: 'ERROR_COUPON_IS_NOT_ACTIVE_YET',
        };
    }
}
exports.CouponIsNotActiveYetWixError = CouponIsNotActiveYetWixError;
/** @hidden */
CouponIsNotActiveYetWixError.__type = 'wix_spi_error';
class CouponDoesNotExistWixError extends Error {
    constructor() {
        super('CouponDoesNotExist');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'ERROR_COUPON_DOES_NOT_EXIST';
        this.name = 'CouponDoesNotExist';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CouponDoesNotExist',
            applicationCode: 'ERROR_COUPON_DOES_NOT_EXIST',
        };
    }
}
exports.CouponDoesNotExistWixError = CouponDoesNotExistWixError;
/** @hidden */
CouponDoesNotExistWixError.__type = 'wix_spi_error';
