// @wix/ecom_checkout-templates/build/cjs/src/ecom-v1-checkout-template-checkout-templates.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAndRedirectToCheckout = exports.createCheckoutFromTemplate = exports.queryCheckoutTemplates = exports.deleteCheckoutTemplate = exports.updateCheckoutTemplate = exports.getCheckoutTemplate = exports.createCheckoutTemplate = void 0;
const ambassadorWixEcomV1CheckoutTemplate = __importStar(require("./ecom-v1-checkout-template-checkout-templates.http.js"));
function createCheckoutTemplate() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.createCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkout-templates',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCheckoutTemplate = createCheckoutTemplate;
function getCheckoutTemplate() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.getCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkout-templates/{checkoutTemplateId}',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCheckoutTemplate = getCheckoutTemplate;
function updateCheckoutTemplate() {
    const payload = { checkoutTemplate: { id: ':checkoutTemplateId' } };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.updateCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/checkout-templates/{checkoutTemplate.id}',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCheckoutTemplate = updateCheckoutTemplate;
function deleteCheckoutTemplate() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.deleteCheckoutTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/checkout-templates/{checkoutTemplateId}',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCheckoutTemplate = deleteCheckoutTemplate;
function queryCheckoutTemplates() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.queryCheckoutTemplates(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkout-templates/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryCheckoutTemplates = queryCheckoutTemplates;
function createCheckoutFromTemplate() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.createCheckoutFromTemplate(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/checkout-templates/{checkoutTemplateId}/create-checkout-from-template',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCheckoutFromTemplate = createCheckoutFromTemplate;
function createAndRedirectToCheckout() {
    const payload = { checkoutTemplateId: ':checkoutTemplateId' };
    const getRequestOptions = ambassadorWixEcomV1CheckoutTemplate.createAndRedirectToCheckout(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/checkout-templates/{checkoutTemplateId}/create-and-redirect-to-checkout',
        pathParams: { checkoutTemplateId: 'checkoutTemplateId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createAndRedirectToCheckout = createAndRedirectToCheckout;
