// @wix/ecom_checkout-templates/build/cjs/src/ecom-v1-checkout-template-checkout-templates.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeightUnit = exports.WebhookIdentityType = exports.VatType = exports.TaxableAddressType = exports.SuggestedFix = exports.SubscriptionFrequency = exports.Status = exports.SortOrder = exports.Severity = exports.RuleType = exports.RateType = exports.PickupMethod = exports.PaymentOptionType = exports.NameInOther = exports.NameInLineItem = exports.ManualCalculationReason = exports.JurisdictionType = exports.ItemTypeItemType = exports.ItemAvailabilityStatus = exports.FileType = exports.FallbackReason = exports.DiscountType = exports.DescriptionLineType = exports.ChargeType = exports.ChannelType = exports.AdditionalFeeSource = exports.onCheckoutTemplateUsed = exports.onCheckoutTemplateUpdated = exports.onCheckoutTemplateDeleted = exports.onCheckoutTemplateCreated = exports.createAndRedirectToCheckout = exports.createCheckoutFromTemplate = exports.queryCheckoutTemplates = exports.deleteCheckoutTemplate = exports.updateCheckoutTemplate = exports.getCheckoutTemplate = exports.createCheckoutTemplate = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_checkout_template_checkout_templates_universal_js_1 = require("./ecom-v1-checkout-template-checkout-templates.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createCheckoutTemplate(httpClient) {
    return (checkoutTemplate) => (0, ecom_v1_checkout_template_checkout_templates_universal_js_1.createCheckoutTemplate)(checkoutTemplate, 
    // @ts-ignore
    { httpClient });
}
exports.createCheckoutTemplate = createCheckoutTemplate;
function getCheckoutTemplate(httpClient) {
    return (checkoutTemplateId) => (0, ecom_v1_checkout_template_checkout_templates_universal_js_1.getCheckoutTemplate)(checkoutTemplateId, 
    // @ts-ignore
    { httpClient });
}
exports.getCheckoutTemplate = getCheckoutTemplate;
function updateCheckoutTemplate(httpClient) {
    return (_id, checkoutTemplate) => (0, ecom_v1_checkout_template_checkout_templates_universal_js_1.updateCheckoutTemplate)(_id, checkoutTemplate, 
    // @ts-ignore
    { httpClient });
}
exports.updateCheckoutTemplate = updateCheckoutTemplate;
function deleteCheckoutTemplate(httpClient) {
    return (checkoutTemplateId) => (0, ecom_v1_checkout_template_checkout_templates_universal_js_1.deleteCheckoutTemplate)(checkoutTemplateId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCheckoutTemplate = deleteCheckoutTemplate;
function queryCheckoutTemplates(httpClient) {
    return () => (0, ecom_v1_checkout_template_checkout_templates_universal_js_1.queryCheckoutTemplates)(
    // @ts-ignore
    { httpClient });
}
exports.queryCheckoutTemplates = queryCheckoutTemplates;
function createCheckoutFromTemplate(httpClient) {
    return (checkoutTemplateId, siteId) => (0, ecom_v1_checkout_template_checkout_templates_universal_js_1.createCheckoutFromTemplate)(checkoutTemplateId, siteId, 
    // @ts-ignore
    { httpClient });
}
exports.createCheckoutFromTemplate = createCheckoutFromTemplate;
function createAndRedirectToCheckout(httpClient) {
    return (checkoutTemplateId, siteId) => (0, ecom_v1_checkout_template_checkout_templates_universal_js_1.createAndRedirectToCheckout)(checkoutTemplateId, siteId, 
    // @ts-ignore
    { httpClient });
}
exports.createAndRedirectToCheckout = createAndRedirectToCheckout;
exports.onCheckoutTemplateCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_template_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.catalogOverrideFields.image' },
            { path: 'entity.customLineItems.media' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            { path: 'entity.customLineItems.physicalProperties.weight' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.customLineItems.url' }],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            {
                path: 'entity.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.customLineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onCheckoutTemplateDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_template_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'undefined.lineItems.catalogOverrideFields.image' },
            { path: 'undefined.customLineItems.media' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'undefined.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            { path: 'undefined.customLineItems.physicalProperties.weight' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'undefined.customLineItems.url' }],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            {
                path: 'undefined.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            {
                path: 'undefined.customLineItems.serviceProperties.scheduledDate',
            },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onCheckoutTemplateUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_template_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.lineItems.catalogOverrideFields.image' },
            { path: 'entity.customLineItems.media' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            { path: 'entity.customLineItems.physicalProperties.weight' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [{ path: 'entity.customLineItems.url' }],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            {
                path: 'entity.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'entity.customLineItems.serviceProperties.scheduledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onCheckoutTemplateUsed = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.checkout_template_used', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            {
                path: 'data.checkoutTemplate.lineItems.catalogOverrideFields.image',
            },
            { path: 'data.checkoutTemplate.customLineItems.media' },
            { path: 'data.checkout.lineItems.media' },
            { path: 'data.checkout.lineItems.catalogOverrideFields.image' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'data.checkoutTemplate.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            {
                path: 'data.checkoutTemplate.customLineItems.physicalProperties.weight',
            },
            { path: 'data.checkout.lineItems.physicalProperties.weight' },
            {
                path: 'data.checkout.lineItems.catalogOverrideFields.physicalProperties.weight',
            },
            {
                path: 'data.checkout.appliedDiscounts.merchantDiscount.percentage',
            },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [
            { path: 'data.checkoutTemplate.customLineItems.url' },
            { path: 'data.checkout.lineItems.url' },
        ],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            {
                path: 'data.checkoutTemplate.customLineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            {
                path: 'data.checkoutTemplate.customLineItems.serviceProperties.scheduledDate',
            },
            { path: 'data.checkout.createdDate' },
            { path: 'data.checkout.updatedDate' },
            {
                path: 'data.checkout.lineItems.subscriptionOptionInfo.subscriptionSettings.startDate',
            },
            { path: 'data.checkout.lineItems.serviceProperties.scheduledDate' },
            {
                path: 'data.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'data.checkout.shippingInfo.selectedCarrierServiceOption.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'data.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.from',
            },
            {
                path: 'data.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.deliveryTimeSlot.to',
            },
            {
                path: 'data.checkout.membershipOptions.eligibleMemberships.expirationDate',
            },
            {
                path: 'data.checkout.membershipOptions.invalidMemberships.membership.expirationDate',
            },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'data.checkout.billingInfo.address' },
            { path: 'data.checkout.shippingInfo.shippingDestination.address' },
            {
                path: 'data.checkout.shippingInfo.selectedCarrierServiceOption.logistics.pickupDetails.address',
            },
            {
                path: 'data.checkout.shippingInfo.carrierServiceOptions.shippingOptions.logistics.pickupDetails.address',
            },
        ],
    },
])))();
var ecom_v1_checkout_template_checkout_templates_universal_js_2 = require("./ecom-v1-checkout-template-checkout-templates.universal.js");
Object.defineProperty(exports, "AdditionalFeeSource", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.AdditionalFeeSource; } });
Object.defineProperty(exports, "ChannelType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.ChannelType; } });
Object.defineProperty(exports, "ChargeType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.ChargeType; } });
Object.defineProperty(exports, "DescriptionLineType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.DescriptionLineType; } });
Object.defineProperty(exports, "DiscountType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.DiscountType; } });
Object.defineProperty(exports, "FallbackReason", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.FallbackReason; } });
Object.defineProperty(exports, "FileType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.FileType; } });
Object.defineProperty(exports, "ItemAvailabilityStatus", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.ItemAvailabilityStatus; } });
Object.defineProperty(exports, "ItemTypeItemType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.ItemTypeItemType; } });
Object.defineProperty(exports, "JurisdictionType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.JurisdictionType; } });
Object.defineProperty(exports, "ManualCalculationReason", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.ManualCalculationReason; } });
Object.defineProperty(exports, "NameInLineItem", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.NameInLineItem; } });
Object.defineProperty(exports, "NameInOther", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.NameInOther; } });
Object.defineProperty(exports, "PaymentOptionType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.PaymentOptionType; } });
Object.defineProperty(exports, "PickupMethod", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.PickupMethod; } });
Object.defineProperty(exports, "RateType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.RateType; } });
Object.defineProperty(exports, "RuleType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.RuleType; } });
Object.defineProperty(exports, "Severity", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.Severity; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.Status; } });
Object.defineProperty(exports, "SubscriptionFrequency", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.SubscriptionFrequency; } });
Object.defineProperty(exports, "SuggestedFix", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.SuggestedFix; } });
Object.defineProperty(exports, "TaxableAddressType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.TaxableAddressType; } });
Object.defineProperty(exports, "VatType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.VatType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WeightUnit", { enumerable: true, get: function () { return ecom_v1_checkout_template_checkout_templates_universal_js_2.WeightUnit; } });
