// @wix/ecom_tippable-staff/build/es/src/tips-staff-v1-staff-tippable-staff.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixTipsStaffV1TipStaffServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'bo._base_domain_': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'wixbo.ai': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'wix-bo.com': [
            {
                srcPath: '/_api/tips-staff-service',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/tips-staff',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_tippable-staff';
/**
 * Retrieves a list of up to 100 staff who are eligible to receive tips, given
 * the provided filtering.
 */
export function listTippableStaff(payload) {
    function __listTippableStaff({ host }) {
        const metadata = {
            entityFqdn: 'wix.tips.staff.v1.staff',
            method: 'GET',
            methodFqn: 'wix.tips.staff.v1.TipStaffService.ListTippableStaff',
            packageName: PACKAGE_NAME,
            url: resolveWixTipsStaffV1TipStaffServiceUrl({
                protoPath: '/v1/eligible-staff',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __listTippableStaff;
}
