// @wix/ecom_local-delivery-options/build/es/src/ecom-v1-local-delivery-option-local-delivery-options.public.js
import { addDeliveryRegion as universalAddDeliveryRegion, bulkCreateLocalDeliveryOption as universalBulkCreateLocalDeliveryOption, bulkDeleteLocalDeliveryOption as universalBulkDeleteLocalDeliveryOption, bulkUpdateLocalDeliveryOption as universalBulkUpdateLocalDeliveryOption, createLocalDeliveryOption as universalCreateLocalDeliveryOption, deleteLocalDeliveryOption as universalDeleteLocalDeliveryOption, getLocalDeliveryOption as universalGetLocalDeliveryOption, listLocalDeliveryOptions as universalListLocalDeliveryOptions, queryLocalDeliveryOptions as universalQueryLocalDeliveryOptions, removeDeliveryRegion as universalRemoveDeliveryRegion, updateLocalDeliveryOption as universalUpdateLocalDeliveryOption, } from './ecom-v1-local-delivery-option-local-delivery-options.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function createLocalDeliveryOption(httpClient) {
    return (localDeliveryOption) => universalCreateLocalDeliveryOption(localDeliveryOption, 
    // @ts-ignore
    { httpClient });
}
export function getLocalDeliveryOption(httpClient) {
    return (localDeliveryOptionId, options) => universalGetLocalDeliveryOption(localDeliveryOptionId, options, 
    // @ts-ignore
    { httpClient });
}
export function listLocalDeliveryOptions(httpClient) {
    return (externalId) => universalListLocalDeliveryOptions(externalId, 
    // @ts-ignore
    { httpClient });
}
export function queryLocalDeliveryOptions(httpClient) {
    return () => universalQueryLocalDeliveryOptions(
    // @ts-ignore
    { httpClient });
}
export function updateLocalDeliveryOption(httpClient) {
    return (_id, localDeliveryOption) => universalUpdateLocalDeliveryOption(_id, localDeliveryOption, 
    // @ts-ignore
    { httpClient });
}
export function deleteLocalDeliveryOption(httpClient) {
    return (localDeliveryOptionId, options) => universalDeleteLocalDeliveryOption(localDeliveryOptionId, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkCreateLocalDeliveryOption(httpClient) {
    return (localDeliveryOptions) => universalBulkCreateLocalDeliveryOption(localDeliveryOptions, 
    // @ts-ignore
    { httpClient });
}
export function bulkUpdateLocalDeliveryOption(httpClient) {
    return (localDeliveryOptions) => universalBulkUpdateLocalDeliveryOption(localDeliveryOptions, 
    // @ts-ignore
    { httpClient });
}
export function bulkDeleteLocalDeliveryOption(httpClient) {
    return (ids) => universalBulkDeleteLocalDeliveryOption(ids, 
    // @ts-ignore
    { httpClient });
}
export function addDeliveryRegion(httpClient) {
    return (deliveryRegionId, options) => universalAddDeliveryRegion(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function removeDeliveryRegion(httpClient) {
    return (deliveryRegionId, options) => universalRemoveDeliveryRegion(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export { ConditionType, DayOfWeek, LengthUnit, LogicalOperator, SortOrder, WebhookIdentityType, } from './ecom-v1-local-delivery-option-local-delivery-options.universal.js';
