// @wix/ecom_local-delivery-options/build/es/src/ecom-v1-local-delivery-option-local-delivery-options.meta.js
import * as ambassadorWixEcomV1LocalDeliveryOption from './ecom-v1-local-delivery-option-local-delivery-options.http.js';
export function createLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.createLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getLocalDeliveryOption() {
    const payload = { localDeliveryOptionId: ':localDeliveryOptionId' };
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.getLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/local-delivery-options/{localDeliveryOptionId}',
        pathParams: { localDeliveryOptionId: 'localDeliveryOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listLocalDeliveryOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.listLocalDeliveryOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/localDeliveryOptions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryLocalDeliveryOptions() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.queryLocalDeliveryOptions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateLocalDeliveryOption() {
    const payload = {
        localDeliveryOption: { id: ':localDeliveryOptionId' },
    };
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.updateLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/local-delivery-options/{localDeliveryOption.id}',
        pathParams: { localDeliveryOptionId: 'localDeliveryOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteLocalDeliveryOption() {
    const payload = { localDeliveryOptionId: ':localDeliveryOptionId' };
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.deleteLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/local-delivery-options/{localDeliveryOptionId}',
        pathParams: { localDeliveryOptionId: 'localDeliveryOptionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.bulkCreateLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/local-delivery-options/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.bulkUpdateLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/bulk/local-delivery-options/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteLocalDeliveryOption() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.bulkDeleteLocalDeliveryOption(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/bulk/local-delivery-options/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addDeliveryRegion() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.addDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options/add-delivery-region',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeDeliveryRegion() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1LocalDeliveryOption.removeDeliveryRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/local-delivery-options/remove-delivery-region',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
