// @wix/ecom_delivery-solutions/build/cjs/src/gateways-ecom-v1-delivery-rates-gateway-delivery-solutions.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDeliverySolutions = void 0;
const float_1 = require("@wix/sdk-runtime/transformations/float");
const float_2 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
function resolveWixEcomDeliveryprofileV1DeliveryRatesUrl(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/delivery-rates',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/delivery-rates',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_delivery-solutions';
/** Get Delivery Solutions for the given line items from all relevant delivery carriers */
function getDeliverySolutions(payload) {
    function __getDeliverySolutions({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'lineItems.physicalProperties.weight' },
                    { path: 'destination.geocode.latitude' },
                    { path: 'destination.geocode.longitude' },
                    { path: 'origin.geocode.latitude' },
                    { path: 'origin.geocode.longitude' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.gateways.ecom.v1.delivery_rates_gateway',
            method: 'POST',
            methodFqn: 'wix.ecom.deliveryprofile.v1.DeliveryRates.GetDeliverySolutions',
            packageName: PACKAGE_NAME,
            url: resolveWixEcomDeliveryprofileV1DeliveryRatesUrl({
                protoPath: '/v1/get-delivery-solutions',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'deliverySolutions.logistics.pickupDetails.address.geocode.latitude',
                        },
                        {
                            path: 'deliverySolutions.logistics.pickupDetails.address.geocode.longitude',
                        },
                    ],
                },
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'deliverySolutions.logistics.deliveryTimeSlot.from' },
                        { path: 'deliverySolutions.logistics.deliveryTimeSlot.to' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getDeliverySolutions;
}
exports.getDeliverySolutions = getDeliverySolutions;
