// @wix/ecom_current-cart/build/es/src/ecom-v1-cart-current-cart.meta.js
import * as ambassadorWixEcomV1Cart from './ecom-v1-cart-current-cart.http.js';
export function getCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.currentCartGetCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/carts/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.updateCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/carts/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addToCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.addToCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/add-to-cart',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeLineItemsFromCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.removeLineItemsFromCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/remove-line-items',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createCheckoutFromCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.createCheckoutFromCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/create-checkout',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeCouponFromCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.removeCouponFromCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/remove-coupon',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCurrentCartLineItemQuantity() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.updateCurrentCartLineItemQuantity(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/update-line-items-quantity',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function estimateCurrentCartTotals() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.estimateCurrentCartTotals(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/carts/current/estimate-totals',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCurrentCart() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1Cart.deleteCurrentCart(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/carts/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
