// @wix/ecom_order-billing/build/es/src/ecom-v1-order-billing-order-billing.meta.js
import * as ambassadorWixEcomV1OrderBilling from './ecom-v1-order-billing-order-billing.http.js';
export function getOrderRefundability() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.getOrderRefundability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/get-order-refundability',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function calculateRefund() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.calculateRefund(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/calculate-refund',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function refundPayments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.refundPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/refund-payments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function authorizeChargeWithSavedPaymentMethod() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.authorizeChargeWithSavedPaymentMethod(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/authorize-charge-with-saved-payment-method',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function voidAuthorizedPayments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.voidAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/void-authorized-payments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function captureAuthorizedPayments() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1OrderBilling.captureAuthorizedPayments(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/order-billing/capture-authorized-payments',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
