// @wix/ecom_gift-vouchers/build/es/src/ecom-v1-gift-card-gift-vouchers.meta.js
import * as ambassadorWixEcomV1GiftCard from './ecom-v1-gift-card-gift-vouchers.http.js';
export function getGiftCard() {
    const payload = { code: ':code' };
    const getRequestOptions = ambassadorWixEcomV1GiftCard.getGiftCard(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/gift-cards/{code}',
        pathParams: { code: 'code' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function redeemGiftCard() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1GiftCard.redeemGiftCard(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/gift-cards/redeem',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function voidTransaction() {
    const payload = {};
    const getRequestOptions = ambassadorWixEcomV1GiftCard.voidTransaction(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/gift-cards/void',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
