// @wix/identity_recovery/build/cjs/src/iam-recovery-v1-recovery-recovery.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantType = exports.StatusName = exports.Status = exports.StateType = exports.Reason = exports.PrivacyStatus = exports.PhoneTag = exports.FactorType = exports.EmailTag = exports.AddressTag = exports.recover = exports.sendActivationEmail = exports.sendRecoveryEmail = exports.__metadata = void 0;
const iam_recovery_v1_recovery_recovery_universal_js_1 = require("./iam-recovery-v1-recovery-recovery.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/identity' };
function sendRecoveryEmail(httpClient) {
    return (email, options) => (0, iam_recovery_v1_recovery_recovery_universal_js_1.sendRecoveryEmail)(email, options, 
    // @ts-ignore
    { httpClient });
}
exports.sendRecoveryEmail = sendRecoveryEmail;
function sendActivationEmail(httpClient) {
    return (identityId, options) => (0, iam_recovery_v1_recovery_recovery_universal_js_1.sendActivationEmail)(identityId, options, 
    // @ts-ignore
    { httpClient });
}
exports.sendActivationEmail = sendActivationEmail;
function recover(httpClient) {
    return (recoveryToken, options) => (0, iam_recovery_v1_recovery_recovery_universal_js_1.recover)(recoveryToken, options, 
    // @ts-ignore
    { httpClient });
}
exports.recover = recover;
var iam_recovery_v1_recovery_recovery_universal_js_2 = require("./iam-recovery-v1-recovery-recovery.universal.js");
Object.defineProperty(exports, "AddressTag", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.AddressTag; } });
Object.defineProperty(exports, "EmailTag", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.EmailTag; } });
Object.defineProperty(exports, "FactorType", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.FactorType; } });
Object.defineProperty(exports, "PhoneTag", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.PhoneTag; } });
Object.defineProperty(exports, "PrivacyStatus", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.PrivacyStatus; } });
Object.defineProperty(exports, "Reason", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.Reason; } });
Object.defineProperty(exports, "StateType", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.StateType; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.Status; } });
Object.defineProperty(exports, "StatusName", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.StatusName; } });
Object.defineProperty(exports, "TenantType", { enumerable: true, get: function () { return iam_recovery_v1_recovery_recovery_universal_js_2.TenantType; } });
