// @wix/ecom_back-in-stock-notifications/build/cjs/src/ecom-v1-back-in-stock-notification-request-back-in-stock-notifications.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Status = exports.SortOrder = exports.onBackInStockNotificationRequestDeleted = exports.onBackInStockNotificationRequestUpdated = exports.onBackInStockNotificationRequestCreated = exports.reportItemsBackInStock = exports.getBackInStockNotificationRequestsCountByCatalogReferences = exports.queryBackInStockNotificationRequests = exports.markAsNotificationSent = exports.deleteBackInStockNotificationRequest = exports.getBackInStockNotificationRequest = exports.createBackInStockNotificationRequest = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1 = require("./ecom-v1-back-in-stock-notification-request-back-in-stock-notifications.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createBackInStockNotificationRequest(httpClient) {
    return (request, itemDetails) => (0, ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1.createBackInStockNotificationRequest)(request, itemDetails, 
    // @ts-ignore
    { httpClient });
}
exports.createBackInStockNotificationRequest = createBackInStockNotificationRequest;
function getBackInStockNotificationRequest(httpClient) {
    return (_id) => (0, ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1.getBackInStockNotificationRequest)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.getBackInStockNotificationRequest = getBackInStockNotificationRequest;
function deleteBackInStockNotificationRequest(httpClient) {
    return (_id) => (0, ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1.deleteBackInStockNotificationRequest)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.deleteBackInStockNotificationRequest = deleteBackInStockNotificationRequest;
function markAsNotificationSent(httpClient) {
    return (_id) => (0, ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1.markAsNotificationSent)(_id, 
    // @ts-ignore
    { httpClient });
}
exports.markAsNotificationSent = markAsNotificationSent;
function queryBackInStockNotificationRequests(httpClient) {
    return () => (0, ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1.queryBackInStockNotificationRequests)(
    // @ts-ignore
    { httpClient });
}
exports.queryBackInStockNotificationRequests = queryBackInStockNotificationRequests;
function getBackInStockNotificationRequestsCountByCatalogReferences(httpClient) {
    return (catalogReferences) => (0, ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1.getBackInStockNotificationRequestsCountByCatalogReferences)(catalogReferences, 
    // @ts-ignore
    { httpClient });
}
exports.getBackInStockNotificationRequestsCountByCatalogReferences = getBackInStockNotificationRequestsCountByCatalogReferences;
function reportItemsBackInStock(httpClient) {
    return (itemDetails, options) => (0, ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_1.reportItemsBackInStock)(itemDetails, options, 
    // @ts-ignore
    { httpClient });
}
exports.reportItemsBackInStock = reportItemsBackInStock;
exports.onBackInStockNotificationRequestCreated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.back_in_stock_notification_request_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onBackInStockNotificationRequestUpdated = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.back_in_stock_notification_request_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onBackInStockNotificationRequestDeleted = (0, sdk_types_1.EventDefinition)('wix.ecom.v1.back_in_stock_notification_request_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_2 = require("./ecom-v1-back-in-stock-notification-request-back-in-stock-notifications.universal.js");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_2.Status; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_back_in_stock_notification_request_back_in_stock_notifications_universal_js_2.WebhookIdentityType; } });
